{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "deletable": true,
    "editable": true
   },
   "source": [
    "<img src=\"images/JHI_STRAP_Web.png\" style=\"width: 150px; float: right;\">\n",
    "# 04 - Using NCBI `BLAST+` Service  with Biopython"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "deletable": true,
    "editable": true
   },
   "source": [
    "## Table of Contents\n",
    "\n",
    "1. [Introduction](#introduction)\n",
    "2. [Python imports](#imports)\n",
    "3. [Running and analysing a remote BLASTX search](#blastx)\n",
    "  1. [Run `BLASTX`](#runblastx)\n",
    "  2. [Load `BLASTX` results](#loadresults)\n",
    "  3. [Exercise 01](#ex01)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "deletable": true,
    "editable": true
   },
   "source": [
    "## Introduction\n",
    "\n",
    "This notebook presents examples of methods for using `BLAST` programmatically, with the webservice provided by NCBI. All calculations will be run on NCBI's servers, using NCBI's databases (not your local `BLAST` installation), but you will be controlling the search using Python code in this notebook.\n",
    "\n",
    "The advantages to using a programmatic interface for remote BLAST queries are the same as those listed in [notebook 03](03-programming_for_blast):\n",
    "\n",
    "* It is easy to set up repeatable searches for many sequences, or collections of sequences\n",
    "* It is easy to read in the search results and conduct downstream analyses that add value to your search\n",
    "\n",
    "Where it is be practical to submit a large number of simultaneous queries via a web form (because it is tiresome to point-and-click over and over again), this can be handled programmatically instead. You have the opportunity to change custom options to help refine your query, compared to the website interface. If you need to repeat a query, it can be trivial to get the same settings every time, if you use a programmatic approach."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "deletable": true,
    "editable": true
   },
   "source": [
    "<a id=\"imports\"></a>\n",
    "## Python imports\n",
    "\n",
    "To interact with the NCBI's `BLAST` service, we will again use the free `Biopython` programming tools. These provide an interface to interact with NCBI's `BLAST` server, run jobs, and to retrieve the output files.\n",
    "\n",
    "We import these tools, and some standard library packages for working with files (`os`) below."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Populating the interactive namespace from numpy and matplotlib\n"
     ]
    }
   ],
   "source": [
    "# Show plots as part of the notebook\n",
    "%pylab inline\n",
    "\n",
    "# Standard library packages\n",
    "import os\n",
    "\n",
    "# Import Biopython tools for running remote BLAST searches\n",
    "from Bio.Blast import NCBIWWW\n",
    "\n",
    "# Import Biopython SeqIO module to handle reading sequence data\n",
    "from Bio import SeqIO"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "deletable": true,
    "editable": true
   },
   "source": [
    "<a id=\"blastx\"></a>\n",
    "## Running and analysing a remote BLASTX search\n",
    "\n",
    "As in [notebook 03](03-programming_for_blast.ipynb), our first worked example will be to run a `BLASTX` search, querying a nucleotide sequence against a protein database, to identify potential homologues. What is different about this search is that we will be conducting it at NCBI, and using a different database.\n",
    "\n",
    "We will use `Biopython` in the code blocks below to first perform the same `BLASTX` search you carried out in the exercise from [notebook 01](01-blast_at_NCBI_website.ipynb) - query a lantibiotic biosynthesis protein from *Kitasatospora* against other *Kitasatospora* species, and write the results to file. Then we will parse the results into a `pandas` dataframe, and finally plot some summary statistics using `seaborn`."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "deletable": true,
    "editable": true
   },
   "source": [
    "<a id=\"runblastx\"></a>\n",
    "### Run `BLASTX`\n",
    "\n",
    "Running a remote `BLAST` search with `Biopython` is, in some ways, simpler than running a local `BLAST` query. The key steps are:\n",
    "\n",
    "1. Read the query sequence(s) from a source (possibly a local file, but maybe a remote database)\n",
    "2. Run a remote job with the `NCBIWWW.qblast()` method, specifying your query sequence, database, and `BLAST` program\n",
    "3. Parse the output you get back from NCBI\n",
    "\n",
    "To run the remote job, we need the same kind of information as if we were running `BLAST` via the web interface - these arguments are compulsory:\n",
    "\n",
    "* the `BLAST` program to use\n",
    "* query sequence(s) to search with\n",
    "* the database to search in\n",
    "\n",
    "but we can provide some extra choices when we run the remote job, including restricting the remote search on the basis of taxonomy, just as we did in the exercise from notebook 01 (we'll do this later)."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "deletable": true,
    "editable": true
   },
   "source": [
    "<div class=\"alert-success\">\n",
    "<b>Our first task is to obtain a query sequence for the search</b>\n",
    "</div>\n",
    "\n",
    "We use the same penicillin-binding protein from notebook 01, and read it from a local file with `Biopython's` `SeqIO()` module.\n",
    "\n",
    "When data such as biological sequences are read in, their metadata - information on database IDs, and other features - follows them. `Biopython` does a nice job of showing us this information if we look at it with the `print()` function:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ID: lcl|LISX01000001.1_cds_OKJ16671.1_31\n",
      "Name: lcl|LISX01000001.1_cds_OKJ16671.1_31\n",
      "Description: lcl|LISX01000001.1_cds_OKJ16671.1_31 [locus_tag=AMK19_00175] [protein=penicillin-binding protein] [protein_id=OKJ16671.1] [location=39730..41184]\n",
      "Number of features: 0\n",
      "Seq('GTGAACAAGCCGATCCGCCGGGTGTCGATCTTCTGCCTGGTCCTGATCCTGGCC...TAG', SingleLetterAlphabet())\n"
     ]
    }
   ],
   "source": [
    "# Define path to data directory\n",
    "datadir = os.path.join(\"data\", \"kitasatospora\")\n",
    "\n",
    "# Load sequence of penicillin-binding protein, and inspect the information\n",
    "seqfile = os.path.join(datadir, \"k_sp_CB01950_penicillin.fasta\")\n",
    "seq = SeqIO.read(seqfile, \"fasta\")\n",
    "print(seq)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "deletable": true,
    "editable": true
   },
   "source": [
    "<div class=\"alert-warning\">\n",
    "<b>However, the remote `BLAST` server requires us to present our sequence in `FASTA` format!</b>\n",
    "</div>\n",
    "\n",
    "One of the clever things about `Biopython`'s sequence objects - and a big advantage of using programmatic approaches - is that we can readily convert our sequence information into a number of different formats. To do this, we can use the sequence's `.format()` method to produce a `FASTA`-formatted string.\n",
    "\n",
    "Doing this does not change the original sequence or its information in any way, but it creates a new presentation of that data, which we can use as our query:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">lcl|LISX01000001.1_cds_OKJ16671.1_31 [locus_tag=AMK19_00175] [protein=penicillin-binding protein] [protein_id=OKJ16671.1] [location=39730..41184]\n",
      "GTGAACAAGCCGATCCGCCGGGTGTCGATCTTCTGCCTGGTCCTGATCCTGGCCCTGATG\n",
      "CTCCGGGTGAACTGGGTGCAGGGCGTTCAGGCGTCGACGTGGGCCAACAACCCGCACAAC\n",
      "GACCGCACCAAGTACGACAAGTACGCCTACCCGCGCGGCAACATCATCGTCGGCGGCCAG\n",
      "GCCGTCACCAAGTCCGACTTCGTCAACGGGCTGCGCTACAAGTACAAGCGCTCCTGGGTG\n",
      "GACGGGCCGATGTACGCGCCGGTCACCGGCTACTCCTCGCAGACGTACGACGCCAGCCAG\n",
      "CTGGAGAAGCTGGAGGACGGCATCCTCTCCGGCACCGACTCGCGGCTGTTCTTCCGCAAC\n",
      "ACCCTGGACATGCTGACCGGCAAGCCCAAGCAGGGCGGCGACGTGGTCACCACCATCGAC\n",
      "CCCAAGGTGCAGAAGGCCGGCTTCGAGGGGCTCGGCAACAAGAAGGGCGCCGCGGTCGCC\n",
      "ATCGACCCGAAGACCGGGGCGATCCTCGGGCTGGTCTCCACCCCGTCCTACGACCCGGGC\n",
      "ACCTTCGCGGGCGGCACCAAGGACGACGAGAAGGCCTGGACGGCACTCGACAGCGACCCG\n",
      "AACAAGCCGATGCTGAACCGGGCGCTGCGCGAGACCTACCCGCCCGGCTCGACCTTCAAG\n",
      "CTGGTCACCGCGGCGACCGCGTTCGAGACCGGCAAGTACCAGAGCCCGTCGGACGTCACC\n",
      "GACACCCCGGACCAGTACATCCTGCCCGGCACCAGCACCCCGCTGATCAACGCCAGCCCC\n",
      "ACCGAGGACTGCGGGAACGCCACCGTGCAACACGCGATGGACCTGTCCTGCAACACGGTG\n",
      "TTCGGCAAGATGGGCGCCGAGCTGGGCGGCGACAAGCTGCGGGCGCAGGCCGAGAAGTTC\n",
      "GGCTTCAACAAGGAAGTGACCATCCCGATCCGGGCCGACGCCAGCCACTTCCCGTCCGGC\n",
      "TCCAAGCCCGACGGCACCGCGATGGACGCGATCGGCCAGCACGACACCCGGGCCACCCCG\n",
      "CTGCAGATGGCCATGGTGGCCGCCGCGATCGCCAACAACGGCTCGCTGATGGAGCCCTAC\n",
      "CTGGTCGCCCAGGAGGGCTCGGCCGGGAACGTGATCTCCACCCACACCGAGCACCAGCTG\n",
      "TCGCAGGCGGTCTCGCCGGCCACCGCGCAGAAGCTCCAGCAGCTGATGGAGTCGGTGGTG\n",
      "CAGCACGGCACCGGCACGAACGCGAAGATCCCGGGCGTGACGGTCGGCGGCAAGACCGGC\n",
      "ACCGCCCAGCACGGCCAGGACAACTCGGGTCTGCCGTTCGCCTGGTTCGTCTCCTACGCC\n",
      "AAGGGCCAGGACGGCAAGCAGGTCGCCGTCGCGGTGGTGGTCGAGGACGGCTCCAGCGAG\n",
      "GCGCAGCAGATCTCCGGCGGCTCGCTGGCCGCGCCGATCGCCAAGGCGATGATGCAGGCG\n",
      "GCGCTCGGGAAGTAG\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# We need the sequence as a string, so use the .format() method\n",
    "print(seq.format(\"fasta\"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "deletable": true,
    "editable": true
   },
   "source": [
    "<div class=\"alert-success\">\n",
    "<b>We are now almost ready to build our `BLAST` query</b>\n",
    "</div>\n",
    "\n",
    "The last two things we need to do are to consider the database we're going to search against, and the format we want the data returned in.\n",
    "\n",
    "* We are going to query against the `pdbaa` database (which contains amino acid sequences for the set of proteins with known structures), as this is small, and should return a result fairly quickly. This is a useful way to identify protein structures that may be useful for threading or other analyses (structural analyses will be covered in session 03).\n",
    "* We are going to request `\"Text\"` format output, as this will be easy to read.\n",
    "\n",
    "<div class=\"alert-warning\">\n",
    "<b>The search will return a special kind of object called a 'handle', which we still need to do some work with, to turn into a file, or read the results from.</b>\n",
    "</div>\n",
    "\n",
    "<div class=\"alert-danger\">\n",
    "<b>The remote search may take some time.</b>\n",
    "</div>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [],
   "source": [
    "# Run a remote BLASTX search, with the penicillin-binding protein as a query,\n",
    "# against the pdbaa database, getting text output\n",
    "result_handle = NCBIWWW.qblast(\"blastx\", \"pdbaa\", seq.format(\"fasta\"),\n",
    "                               format_type=\"Text\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "deletable": true,
    "editable": true
   },
   "source": [
    "Once the search has completed, you can `read()` the contents of the handle into a variable called `output`, and then inspect the contents using the `print()` function."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<p><!--\n",
      "QBlastInfoBegin\n",
      "\tStatus=READY\n",
      "QBlastInfoEnd\n",
      "--><p>\n",
      "<PRE>\n",
      "BLASTX 2.6.1+\n",
      "Reference: Stephen F. Altschul, Thomas L. Madden, Alejandro\n",
      "A. Schaffer, Jinghui Zhang, Zheng Zhang, Webb Miller, and\n",
      "David J. Lipman (1997), \"Gapped BLAST and PSI-BLAST: a new\n",
      "generation of protein database search programs\", Nucleic\n",
      "Acids Res. 25:3389-3402.\n",
      "\n",
      "\n",
      "RID: B8A9RJP9015\n",
      "\n",
      "\n",
      "Database: PDB protein database\n",
      "           89,751 sequences; 22,414,899 total letters\n",
      "Query= lcl|LISX01000001.1_cds_OKJ16671.1_31 [locus_tag=AMK19_00175]\n",
      "[protein=penicillin-binding protein] [protein_id=OKJ16671.1]\n",
      "[location=39730..41184]\n",
      "\n",
      "Length=1455\n",
      "\n",
      "\n",
      "                                                                   Score     E\n",
      "Sequences producing significant alignments:                       (Bits)  Value\n",
      "\n",
      "3UN7_A  Chain A, Crystal Structure Of Pbpa From Mycobacterium ...  297     1e-95\n",
      "3LO7_A  Chain A, Crystal Structure Of Pbpa From Mycobacterium ...  295     1e-94\n",
      "4MNR_A  Chain A, Crystal Structure Of D,d-transpeptidase Domai...  216     2e-64\n",
      "4RA7_A  Chain A, Structure Of A Putative Peptidoglycan Glycosy...  128     6e-32\n",
      "4N1X_A  Chain A, Structure Of A Putative Peptidoglycan Glycosy...  128     8e-32\n",
      "4JBF_A  Chain A, Crystal Structure Of Peptidoglycan Glycosyltr...  128     8e-32\n",
      "5E31_A  Chain A, 2.3 Angstrom Crystal Structure Of The Monomer...  115     4e-27\n",
      "4CPK_A  Chain A, Crystal Structure Of Pbp2a Double Clinical Mu...  114     1e-26\n",
      "1VQQ_A  Chain A, Structure Of Penicillin Binding Protein 2a Fr...  114     1e-26\n",
      "4BL2_A  Chain A, Crystal Structure Of Pbp2a Clinical Mutant E1...  114     1e-26\n",
      "3ZG5_A  Chain A, Crystal Structure Of Pbp2a From Mrsa In Compl...  114     1e-26\n",
      "4BL3_A  Chain A, Crystal Structure Of Pbp2a Clinical Mutant N1...  112     4e-26\n",
      "1MWS_A  Chain A, Structure Of Nitrocefin Acyl-Penicillin Bindi...  112     5e-26\n",
      "1MWR_A  Chain A, Structure Of Semet Penicillin Binding Protein...  108     7e-25\n",
      "3UE3_A  Chain A, Crystal Structure Of Acinetobacter Baumanni Pbp3  93.2    8e-20\n",
      "5TRO_A  Chain A, 1.8 Angstrom Resolution Crystal Structure Of ...  91.7    3e-19\n",
      "3EQV_A  Chain A, Crystal Structure Of Penicillin-Binding Prote...  87.8    4e-18\n",
      "3EQU_A  Chain A, Crystal Structure Of Penicillin-Binding Prote...  87.0    8e-18\n",
      "4L0L_A  Chain A, Crystal Structure Of P.aeruginosa Pbp3 In Com...  78.2    5e-15\n",
      "3PBN_A  Chain A, Crystal Structure Of Apo Pbp3 From Pseudomona...  77.8    6e-15\n",
      "4BJP_A  Chain A, Crystal Structure Of E. Coli Penicillin Bindi...  77.8    7e-15\n",
      "4FSF_A  Chain A, Crystal Structure Of Pseudomonas Aeruginosa P...  77.8    7e-15\n",
      "4WEK_A  Chain A, Crystal Structure Of Pseudomonas Aeruginosa P...  77.8    7e-15\n",
      "3OC2_A  Chain A, Crystal Structure Of Penicillin-binding Prote...  77.8    8e-15\n",
      "4ZTK_A  Chain A, Transpeptidase Domain Of Ftsi4 D,d-transpepti...  64.3    5e-11\n",
      "4OVD_A  Chain A, Crystal Structure Of A Putative Peptidoglycan...  65.5    6e-11\n",
      "4U3T_A  Chain A, Crystal Structure Of The Transpeptidase Domai...  60.1    2e-09\n",
      "3ZG8_B  Chain B, Crystal Structure Of Penicillin Binding Prote...  60.5    2e-09\n",
      "2WAF_A  Chain A, Penicillin-binding Protein 2b (pbp-2b) From S...  58.2    1e-08\n",
      "2V2F_F  Chain F, Crystal Structure Of Pbp1a From Drug-resistan...  57.8    1e-08\n",
      "2C5W_B  Chain B, Penicillin-Binding Protein 1a (Pbp-1a) Acyl-E...  55.8    5e-08\n",
      "2ZC5_B  Chain B, Penicillin-Binding Protein 1a (Pbp 1a) Acyl-E...  55.1    9e-08\n",
      "2C6W_B  Chain B, Penicillin-Binding Protein 1a (Pbp-1a) From S...  54.7    1e-07\n",
      "3ZG7_B  Chain B, Crystal Structure Of Penicillin-binding Prote...  53.9    3e-07\n",
      "2WAD_A  Chain A, Penicillin-binding Protein 2b (pbp-2b) From S...  53.5    4e-07\n",
      "2WAE_A  Chain A, Penicillin-binding Protein 2b (pbp-2b) From S...  53.1    5e-07\n",
      "3UDF_A  Chain A, Crystal Structure Of Apo Pbp1a From Acinetoba...  50.8    3e-06\n",
      "1K25_A  Chain A, Pbp2x From A Highly Penicillin-Resistant Stre...  49.7    5e-06\n",
      "5U47_A  Chain A, 1.95 Angstrom Resolution Crystal Structure Of...  48.9    1e-05\n",
      "1RP5_A  Chain A, Pbp2x From Streptococcus Pneumoniae Strain 52...  47.4    3e-05\n",
      "4OON_A  Chain A, Crystal Structure Of Pbp1a In Complex With Co...  46.6    6e-05\n",
      "1PYY_A  Chain A, Double Mutant Pbp2x T338aM339F FROM STREPTOCO...  43.9    3e-04\n",
      "1PMD_A  Chain A, Penicillin-Binding Protein 2x (Pbp-2x)            43.5    5e-04\n",
      "1QME_A  Chain A, Penicillin-binding Protein 2x (pbp-2x)            43.1    6e-04\n",
      "4YE5_A  Chain A, The Crystal Structure Of A Peptidoglycan Synt...  42.7    7e-04\n",
      "3DWK_A  Chain A, Identification Of Dynamic Structural Motifs I...  42.4    0.001\n",
      "2Z2L_B  Chain B, Penicillin-Binding Protein 2x (Pbp2x) From St...  40.4    0.004\n",
      "2OLU_A  Chain A, Structural Insight Into The Transglycosylatio...  40.0    0.005\n",
      "2Y2G_A  Chain A, Penicillin-Binding Protein 1b (Pbp-1b) In Com...  39.7    0.008\n",
      "5FGZ_A  Chain A, E. Coli Pbp1b In Complex With Fpi-1465            39.7    0.009\n",
      "\n",
      "ALIGNMENTS\n",
      ">3UN7_A Chain A, Crystal Structure Of Pbpa From Mycobacterium Tuberculosis\n",
      " 3UN7_B Chain B, Crystal Structure Of Pbpa From Mycobacterium Tuberculosis\n",
      " 3UPN_A Chain A, Structure Of Penicillin-Binding Protein A From M. Tuberculosis: \n",
      "Imipenem Acyl-Enzyme Complex\n",
      " 3UPN_B Chain B, Structure Of Penicillin-Binding Protein A From M. Tuberculosis: \n",
      "Imipenem Acyl-Enzyme Complex\n",
      " 3UPO_A Chain A, Structure Of Penicillin-Binding Protein A From M. Tuberculosis: \n",
      "Penicillin G Acyl-Enzyme Complex\n",
      " 3UPO_B Chain B, Structure Of Penicillin-Binding Protein A From M. Tuberculosis: \n",
      "Penicillin G Acyl-Enzyme Complex\n",
      " 3UPP_A Chain A, Structure Of Penicillin-Binding Protein A From M. Tuberculosis: \n",
      "Ceftrixaone Acyl-Enzyme Complex\n",
      " 3UPP_B Chain B, Structure Of Penicillin-Binding Protein A From M. Tuberculosis: \n",
      "Ceftrixaone Acyl-Enzyme Complex\n",
      "Length=462\n",
      "\n",
      " Score = 297 bits (760),  Expect = 1e-95, Method: Compositional matrix adjust.\n",
      " Identities = 192/461 (42%), Positives = 265/461 (57%), Gaps = 25/461 (5%)\n",
      " Frame = +1\n",
      "\n",
      "Query  109   NPHNDRTKYDKYAYPRGNIIVGGQAVTKSDFVNGLRYKYKRSWVDGPMYAPVTGYSSQTY  288\n",
      "             +P N R   D+Y+  RG I  GGQ +  S   +G R+++ R + +  +YAPVTG+ S  Y\n",
      "Sbjct  8     DPRNQRVLLDEYSRQRGQITAGGQLLAYSVATDG-RFRFLRVYPNPEVYAPVTGFYSLRY  66\n",
      "\n",
      "Query  289   DASQLEKLEDGILSGTDSRLFFRNTLDMLTGKPKQGGDVVTTIDPKVQKAGFEGL-----  453\n",
      "              ++ LE+ ED IL+G+D RLF R   D  TG+  +GG+V TTI+P++Q+AG++ +     \n",
      "Sbjct  67    SSTALERAEDPILNGSDRRLFGRRLADFFTGRDPRGGNVDTTINPRIQQAGWDAMQQGCY  126\n",
      "\n",
      "Query  454   GNKKGAAVAIDPKTGAILGLVSTPSYDPGTFAGGTKDDE-KAWTALDSDPNKPMLNRALR  630\n",
      "             G  KGA VA++P TG IL LVS+PSYDP   A    + + +AW  L  +P  P+ NRA+ \n",
      "Sbjct  127   GPCKGAVVALEPSTGKILALVSSPSYDPNLLASHNPEVQAQAWQRLGDNPASPLTNRAIS  186\n",
      "\n",
      "Query  631   ETYPPGSTFKLVTAATAFETGKYQSPSDVTDTPDQYILPGTSTPLIN--ASPTEDCGNAT  804\n",
      "             ETYPPGSTFK++T A A   G  ++   +T  P    LPG++  L N   +P  D    +\n",
      "Sbjct  187   ETYPPGSTFKVITTAAALAAGATET-EQLTAAPT-IPLPGSTAQLENYGGAPCGDEPTVS  244\n",
      "\n",
      "Query  805   VQHAMDLSCNTVFGKMGAELGGDKLRAQAEKFGFNKEVTIPIRADASHFPSGSKPDGTA-  981\n",
      "             ++ A   SCNT F ++G   G D LR+ A  FG +     P     +    G  PD  A \n",
      "Sbjct  245   LREAFVKSCNTAFVQLGIRTGADALRSMARAFGLDSPPR-PTPLQVAESTVGPIPDSAAL  303\n",
      "\n",
      "Query  982   -MDAIGQHDTRATPLQMAMVAAAIANNGSLMEPYLVAQ-EGSAGNVISTHTEHQLSQAVS  1155\n",
      "              M +IGQ D   TPL  A +AA IAN G  M PYLV   +G     IST   +Q  +AVS\n",
      "Sbjct  304   GMTSIGQKDVALTPLANAEIAATIANGGITMRPYLVGSLKGPDLANISTTVRYQQRRAVS  363\n",
      "\n",
      "Query  1156  PATAQKLQQLM---ESVVQHGTGTNAKIPGVTVGGKTGTAQHGQD-NSGLPFAWFVSYAK  1323\n",
      "             P  A KL +LM   E V Q        IPGV +  KTGTA+HG D     P AW++++A \n",
      "Sbjct  364   PQVAAKLTELMVGAEKVAQQ----KGAIPGVQIASKTGTAEHGTDPRHTPPHAWYIAFAP  419\n",
      "\n",
      "Query  1324  GQDGKQVAVAVVVEDGSSEAQQISGGSLAAPIAKAMMQAAL  1446\n",
      "              Q  K VAVAV+VE+G+      +GG+LAAPI +A+++AAL\n",
      "Sbjct  420   AQAPK-VAVAVLVENGADRLSA-TGGALAAPIGRAVIEAAL  458\n",
      "\n",
      "\n",
      ">3LO7_A Chain A, Crystal Structure Of Pbpa From Mycobacterium Tuberculosis\n",
      " 3LO7_B Chain B, Crystal Structure Of Pbpa From Mycobacterium Tuberculosis\n",
      "Length=483\n",
      "\n",
      " Score = 295 bits (755),  Expect = 1e-94, Method: Compositional matrix adjust.\n",
      " Identities = 192/461 (42%), Positives = 265/461 (57%), Gaps = 25/461 (5%)\n",
      " Frame = +1\n",
      "\n",
      "Query  109   NPHNDRTKYDKYAYPRGNIIVGGQAVTKSDFVNGLRYKYKRSWVDGPMYAPVTGYSSQTY  288\n",
      "             +P N R   D+Y+  RG I  GGQ +  S   +G R+++ R + +  +YAPVTG+ S  Y\n",
      "Sbjct  29    DPRNQRVLLDEYSRQRGQITAGGQLLAYSVATDG-RFRFLRVYPNPEVYAPVTGFYSLRY  87\n",
      "\n",
      "Query  289   DASQLEKLEDGILSGTDSRLFFRNTLDMLTGKPKQGGDVVTTIDPKVQKAGFEGL-----  453\n",
      "              ++ LE+ ED IL+G+D RLF R   D  TG+  +GG+V TTI+P++Q+AG++ +     \n",
      "Sbjct  88    SSTALERAEDPILNGSDRRLFGRRLADFFTGRDPRGGNVDTTINPRIQQAGWDAMQQGCY  147\n",
      "\n",
      "Query  454   GNKKGAAVAIDPKTGAILGLVSTPSYDPGTFAGGTKDDE-KAWTALDSDPNKPMLNRALR  630\n",
      "             G  KGA VA++P TG IL LVS+PSYDP   A    + + +AW  L  +P  P+ NRA+ \n",
      "Sbjct  148   GPCKGAVVALEPSTGKILALVSSPSYDPNLLASHNPEVQAQAWQRLGDNPASPLTNRAIS  207\n",
      "\n",
      "Query  631   ETYPPGSTFKLVTAATAFETGKYQSPSDVTDTPDQYILPGTSTPLIN--ASPTEDCGNAT  804\n",
      "             ETYPPGSTFK++T A A   G  ++   +T  P    LPG++  L N   +P  D    +\n",
      "Sbjct  208   ETYPPGSTFKVITTAAALAAGATET-EQLTAAPT-IPLPGSTAQLENYGGAPCGDEPTVS  265\n",
      "\n",
      "Query  805   VQHAMDLSCNTVFGKMGAELGGDKLRAQAEKFGFNKEVTIPIRADASHFPSGSKPDGTA-  981\n",
      "             ++ A   SCNT F ++G   G D LR+ A  FG +     P     +    G  PD  A \n",
      "Sbjct  266   LREAFVKSCNTAFVQLGIRTGADALRSMARAFGLDSPPR-PTPLQVAESTVGPIPDSAAL  324\n",
      "\n",
      "Query  982   -MDAIGQHDTRATPLQMAMVAAAIANNGSLMEPYLVAQ-EGSAGNVISTHTEHQLSQAVS  1155\n",
      "              M +IGQ D   TPL  A +AA IAN G  M PYLV   +G     IST   +Q  +AVS\n",
      "Sbjct  325   GMTSIGQKDVALTPLANAEIAATIANGGITMRPYLVGSLKGPDLANISTTVGYQQRRAVS  384\n",
      "\n",
      "Query  1156  PATAQKLQQLM---ESVVQHGTGTNAKIPGVTVGGKTGTAQHGQD-NSGLPFAWFVSYAK  1323\n",
      "             P  A KL +LM   E V Q        IPGV +  KTGTA+HG D     P AW++++A \n",
      "Sbjct  385   PQVAAKLTELMVGAEKVAQQ----KGAIPGVQIASKTGTAEHGTDPRHTPPHAWYIAFAP  440\n",
      "\n",
      "Query  1324  GQDGKQVAVAVVVEDGSSEAQQISGGSLAAPIAKAMMQAAL  1446\n",
      "              Q  K VAVAV+VE+G+      +GG+LAAPI +A+++AAL\n",
      "Sbjct  441   AQAPK-VAVAVLVENGADRLSA-TGGALAAPIGRAVIEAAL  479\n",
      "\n",
      "\n",
      ">4MNR_A Chain A, Crystal Structure Of D,d-transpeptidase Domain Of Peptidoglycan \n",
      "Glycosyltransferase From Eggerthella Lenta\n",
      "Length=457\n",
      "\n",
      " Score = 216 bits (550),  Expect = 2e-64, Method: Compositional matrix adjust.\n",
      " Identities = 167/457 (37%), Positives = 236/457 (52%), Gaps = 54/457 (12%)\n",
      " Frame = +1\n",
      "\n",
      "Query  85    VQASTWANNPHNDRTKYDKYAYPRGNI-----IVGGQAVTKSDFVNGLRYKYKRSWVDGP  249\n",
      "             VQA  + N P N+ T   +    RG I     +V  ++V + D        Y+R +  G \n",
      "Sbjct  4     VQADYYQNXPGNNHTLAKEARSERGTIATYDGVVLARSVKEEDGT------YEREYPAGD  57\n",
      "\n",
      "Query  250   MYAPVTGYSSQTYDASQLEKLEDGILSGTDSRLFFRNTLDMLTGKPKQGGDVVTTIDPKV  429\n",
      "             + + V GYSS  +  S +EK  +  L G ++   + + L+   G    G DV  T++ K+\n",
      "Sbjct  58    LASHVVGYSSPQFGNSGIEKAYNDTLKGEENFASWTDVLNSFAGIGTAGNDVTLTLNSKI  117\n",
      "\n",
      "Query  430   QKAGFEGLGNKKGAAVAIDPKTGAILGLVSTPSYDPGTFAGGTKDDEKAWTALDSDPNKP  609\n",
      "             Q+A  + L  +KGA V  DP TGAIL   S P+Y+   FA      E+A    +++P+  \n",
      "Sbjct  118   QQAAQDALAGRKGACVVXDPDTGAILAXASAPTYNAADFAAVI---EQA----NANPDDS  170\n",
      "\n",
      "Query  610   ML-NRALRETYPPGSTFKLVTAATAFETGKYQSPSDVTDTPDQYILPGTSTPLINASPTE  786\n",
      "              L +RA    Y PGSTFK+VT ATA E        DV      +  PGT   + NA+ + \n",
      "Sbjct  171   TLVDRAAGSLYAPGSTFKIVTLATALE-------DDVAGEDTVFSSPGTXE-IGNATVSN  222\n",
      "\n",
      "Query  787   ----DCGNATVQHAMDLSCNTVFGKMGAELGGDKLRAQAEKFGFNKEVTIPIRADASHFP  954\n",
      "                 + G+ T+  A +LS NTVFG++G E G DKL A AE FGFNKE+  P+    S  P\n",
      "Sbjct  223   FNKANYGSLTLAQATELSSNTVFGQLGVEXGADKLVAGAESFGFNKEIDFPLYTPESLXP  282\n",
      "\n",
      "Query  955   SGS----KPDGTAMDAIGQ--HDT-----------RATPLQMAMVAAAIANNGSLMEPYL  1083\n",
      "             S       P   A  A G+   DT           +AT L+ A V  AIAN+G + +PYL\n",
      "Sbjct  283   SAEDLQKSPWELAWAAAGEPVGDTTRPGRESPAGPQATVLEXAXVGTAIANDGVIXQPYL  342\n",
      "\n",
      "Query  1084  VAQEGSA-GNVISTHTEHQLSQAVSPATAQKLQQLMESVVQHGTGTNAKIPGVTVGGKTG  1260\n",
      "             V    +A G    + +  +L QAVS  TA +++ ++  VVQ+GTGT A IPG+ V GKTG\n",
      "Sbjct  343   VDSVNNANGERSFSASPTKLXQAVSKTTAGRVRDVLLGVVQNGTGTAAAIPGIDVAGKTG  402\n",
      "\n",
      "Query  1261  TAQHGQDNSGLPFAWFVSYAKGQDGKQVAVAVVVEDG  1371\n",
      "             TA+    N     +WFV  A  +D  +V VA+V+EDG\n",
      "Sbjct  403   TAEKENGND----SWFVGXAPAED-PRVVVAIVIEDG  434\n",
      "\n",
      "\n",
      ">4RA7_A Chain A, Structure Of A Putative Peptidoglycan Glycosyltransferase \n",
      "From Atopobium Parvulum In Complex With Nafcillin\n",
      " 4RA7_B Chain B, Structure Of A Putative Peptidoglycan Glycosyltransferase \n",
      "From Atopobium Parvulum In Complex With Nafcillin\n",
      "Length=453\n",
      "\n",
      " Score = 128 bits (321),  Expect = 6e-32, Method: Compositional matrix adjust.\n",
      " Identities = 131/434 (30%), Positives = 196/434 (45%), Gaps = 40/434 (9%)\n",
      " Frame = +1\n",
      "\n",
      "Query  112   PHNDRTKYDKYAYPRGNIIVGGQAVTKSDFVNGLRYKYKRSWVDGPMYAPVTGYSSQTYD  291\n",
      "             P N+ T   K AY +   I+    VT ++ V      Y R++      +   GY S  Y \n",
      "Sbjct  13    PTNNHT-IAKSAYVQRGAIITSDGVTLAESVKQDDGTYVRNYPHDGXASHTVGYISTQYG  71\n",
      "\n",
      "Query  292   ASQLEKLEDGILSGTDSRLFFRNTLDMLTGKPKQGGDVVTTIDPKVQKAGFEGLGNKKGA  471\n",
      "              + +E   +  L+G      +R+ L    G    G  VV TI+ + Q      L    G+\n",
      "Sbjct  72    TAGIESSXNETLTGHADHSDWRSALYSXAGINTTGSSVVLTINSQXQAVAEAALQGYSGS  131\n",
      "\n",
      "Query  472   AVAIDPKTGAILGLVSTPSYDPGTFAGGTKDDEKAWTALDSDPNKPMLNRALRETYPPGS  651\n",
      "              V  DP TGA+L   S+PSY                T ++S     +++R  +  Y PGS\n",
      "Sbjct  132   IVVXDPSTGAVLAKASSPSYTHAELG----------TIIESGTGSQLVDRTTQALYSPGS  181\n",
      "\n",
      "Query  652   TFKLVTAATAFETGKYQSPSDVTDTPDQYILPGTST---PLINASPTEDCGNATVQHAMD  822\n",
      "             +FK VT A   +T K       T     Y  PGT       I+    ED G   ++ A  \n",
      "Sbjct  182   SFKTVTLAAGIDTHK-------TTLDTTYSAPGTXEIGGGTIHNYANEDXGTIPLREAFA  234\n",
      "\n",
      "Query  823   LSCNTVFGKMGAELGGDKLRAQAEKFGFNKEVTIPIRADASHFPSGSKPD------GTAM  984\n",
      "              S NT   ++G  LG D L + A  FG+   +        S  P+ ++         +  \n",
      "Sbjct  235   RSSNTALAQLGVALGADNLVSYARAFGYGTALGQDFSTTPSLXPNPAEXTTWELAWASCG  294\n",
      "\n",
      "Query  985   DAIGQHDTRATP----LQMAMVAAAIANNGSLMEPYLVAQEGS-AGNVISTHTEHQLSQA  1149\n",
      "               +G+H + A P     Q A++AAAIAN G +  PY+V +  S  G V+ST +   L QA\n",
      "Sbjct  295   LPVGEHASPAGPQTTVXQNAVIAAAIANGGVVXNPYIVDRVLSPEGAVVSTTSPKSLGQA  354\n",
      "\n",
      "Query  1150  VSPATAQKLQQLMESVVQHGTGTNAKIPGVTVGGKTGTA--QHGQDNSGLPFAWFVSYAK  1323\n",
      "             VS  TA ++++    VV+ GTG  A++PGV + GKTGTA  ++G  NS     +F+ +A \n",
      "Sbjct  355   VSADTAAQVREAXLGVVESGTGXGARVPGVKIAGKTGTADVENGNFNS-----FFIGFAP  409\n",
      "\n",
      "Query  1324  GQDGKQVAVAVVVE  1365\n",
      "               D   + V+VV+E\n",
      "Sbjct  410   -YDHPTLVVSVVIE  422\n",
      "\n",
      "\n",
      ">4N1X_A Chain A, Structure Of A Putative Peptidoglycan Glycosyltransferase \n",
      "From Atopobium Parvulum In Complex With Penicillin G\n",
      " 4N1X_B Chain B, Structure Of A Putative Peptidoglycan Glycosyltransferase \n",
      "From Atopobium Parvulum In Complex With Penicillin G\n",
      "Length=472\n",
      "\n",
      " Score = 128 bits (321),  Expect = 8e-32, Method: Compositional matrix adjust.\n",
      " Identities = 132/445 (30%), Positives = 201/445 (45%), Gaps = 40/445 (9%)\n",
      " Frame = +1\n",
      "\n",
      "Query  79    QGVQASTWANNPHNDRTKYDKYAYPRGNIIVGGQAVTKSDFVNGLRYKYKRSWVDGPMYA  258\n",
      "             +G +   + +N  N+ T   K AY +   I+    VT ++ V      Y R++      +\n",
      "Sbjct  21    KGTENLYFQSNATNNHT-IAKSAYVQRGAIITSDGVTLAESVKQDDGTYVRNYPHDGXAS  79\n",
      "\n",
      "Query  259   PVTGYSSQTYDASQLEKLEDGILSGTDSRLFFRNTLDMLTGKPKQGGDVVTTIDPKVQKA  438\n",
      "                GY S  Y  + +E   +  L+G      +R+ L    G    G  VV TI+ + Q  \n",
      "Sbjct  80    HTVGYISTQYGTAGIESSXNETLTGHADHSDWRSALYSXAGINTTGSSVVLTINSQXQAV  139\n",
      "\n",
      "Query  439   GFEGLGNKKGAAVAIDPKTGAILGLVSTPSYDPGTFAGGTKDDEKAWTALDSDPNKPMLN  618\n",
      "                 L    G+ V  DP TGA+L   S+PSY                T ++S     +++\n",
      "Sbjct  140   AEAALQGYSGSIVVXDPSTGAVLAKASSPSYTHAELG----------TIIESGTGSQLVD  189\n",
      "\n",
      "Query  619   RALRETYPPGSTFKLVTAATAFETGKYQSPSDVTDTPDQYILPGTST---PLINASPTED  789\n",
      "             R  +  Y PGS+FK VT A   +T K       T     Y  PGT       I+    ED\n",
      "Sbjct  190   RTTQALYSPGSSFKTVTLAAGIDTHK-------TTLDTTYSAPGTXEIGGGTIHNYANED  242\n",
      "\n",
      "Query  790   CGNATVQHAMDLSCNTVFGKMGAELGGDKLRAQAEKFGFNKEVTIPIRADASHFPSGSKP  969\n",
      "              G   ++ A   S NT   ++G  LG D L + A  FG+   +        S  P+ ++ \n",
      "Sbjct  243   XGTIPLREAFARSSNTALAQLGVALGADNLVSYARAFGYGTALGQDFSTTPSLXPNPAEX  302\n",
      "\n",
      "Query  970   D------GTAMDAIGQHDTRATP----LQMAMVAAAIANNGSLMEPYLVAQEGS-AGNVI  1116\n",
      "                     +    +G+H + A P     Q A++AAAIAN G +  PY+V +  S  G V+\n",
      "Sbjct  303   TTWELAWASCGLPVGEHASPAGPQTTVXQNAVIAAAIANGGVVXNPYIVDRVLSPEGAVV  362\n",
      "\n",
      "Query  1117  STHTEHQLSQAVSPATAQKLQQLMESVVQHGTGTNAKIPGVTVGGKTGTA--QHGQDNSG  1290\n",
      "             ST +   L QAVS  TA ++++    VV+ GTG  A++PGV + GKTGTA  ++G  NS \n",
      "Sbjct  363   STTSPKSLGQAVSADTAAQVREAXLGVVESGTGXGARVPGVKIAGKTGTADVENGNFNS-  421\n",
      "\n",
      "Query  1291  LPFAWFVSYAKGQDGKQVAVAVVVE  1365\n",
      "                 +F+ +A   D   + V+VV+E\n",
      "Sbjct  422   ----FFIGFAP-YDHPTLVVSVVIE  441\n",
      "\n",
      "\n",
      ">4JBF_A Chain A, Crystal Structure Of Peptidoglycan Glycosyltransferase \n",
      "From Atopobium Parvulum Dsm 20469.\n",
      " 4JBF_B Chain B, Crystal Structure Of Peptidoglycan Glycosyltransferase \n",
      "From Atopobium Parvulum Dsm 20469.\n",
      " 4QJG_A Chain A, Structure Of A Putative Peptidoglycan Glycosyltransferase \n",
      "From Atopobium Parvulum In Complex With Penicillin V\n",
      " 4QJG_B Chain B, Structure Of A Putative Peptidoglycan Glycosyltransferase \n",
      "From Atopobium Parvulum In Complex With Penicillin V\n",
      " 4R0Q_A Chain A, Structure Of A Putative Peptidoglycan Glycosyltransferase \n",
      "From Atopobium Parvulum In Complex With Cephalothin\n",
      " 4R0Q_B Chain B, Structure Of A Putative Peptidoglycan Glycosyltransferase \n",
      "From Atopobium Parvulum In Complex With Cephalothin\n",
      " 4R1G_A Chain A, Structure Of A Putative Peptidoglycan Glycosyltransferase \n",
      "From Atopobium Parvulum In Complex With Cloxacillin\n",
      " 4R1G_B Chain B, Structure Of A Putative Peptidoglycan Glycosyltransferase \n",
      "From Atopobium Parvulum In Complex With Cloxacillin\n",
      " 4R23_A Chain A, Structure Of A Putative Peptidoglycan Glycosyltransferase \n",
      "From Atopobium Parvulum In Complex With Dicloxacillin\n",
      " 4R23_B Chain B, Structure Of A Putative Peptidoglycan Glycosyltransferase \n",
      "From Atopobium Parvulum In Complex With Dicloxacillin\n",
      " 4R3J_A Chain A, Structure Of A Putative Peptidoglycan Glycosyltransferase \n",
      "From Atopobium Parvulum In Complex With Cefapirin\n",
      " 4R3J_B Chain B, Structure Of A Putative Peptidoglycan Glycosyltransferase \n",
      "From Atopobium Parvulum In Complex With Cefapirin\n",
      "Length=482\n",
      "\n",
      " Score = 128 bits (321),  Expect = 8e-32, Method: Compositional matrix adjust.\n",
      " Identities = 131/434 (30%), Positives = 196/434 (45%), Gaps = 40/434 (9%)\n",
      " Frame = +1\n",
      "\n",
      "Query  112   PHNDRTKYDKYAYPRGNIIVGGQAVTKSDFVNGLRYKYKRSWVDGPMYAPVTGYSSQTYD  291\n",
      "             P N+ T   K AY +   I+    VT ++ V      Y R++      +   GY S  Y \n",
      "Sbjct  42    PTNNHT-IAKSAYVQRGAIITSDGVTLAESVKQDDGTYVRNYPHDGXASHTVGYISTQYG  100\n",
      "\n",
      "Query  292   ASQLEKLEDGILSGTDSRLFFRNTLDMLTGKPKQGGDVVTTIDPKVQKAGFEGLGNKKGA  471\n",
      "              + +E   +  L+G      +R+ L    G    G  VV TI+ + Q      L    G+\n",
      "Sbjct  101   TAGIESSXNETLTGHADHSDWRSALYSXAGINTTGSSVVLTINSQXQAVAEAALQGYSGS  160\n",
      "\n",
      "Query  472   AVAIDPKTGAILGLVSTPSYDPGTFAGGTKDDEKAWTALDSDPNKPMLNRALRETYPPGS  651\n",
      "              V  DP TGA+L   S+PSY                T ++S     +++R  +  Y PGS\n",
      "Sbjct  161   IVVXDPSTGAVLAKASSPSYTHAELG----------TIIESGTGSQLVDRTTQALYSPGS  210\n",
      "\n",
      "Query  652   TFKLVTAATAFETGKYQSPSDVTDTPDQYILPGTST---PLINASPTEDCGNATVQHAMD  822\n",
      "             +FK VT A   +T K       T     Y  PGT       I+    ED G   ++ A  \n",
      "Sbjct  211   SFKTVTLAAGIDTHK-------TTLDTTYSAPGTXEIGGGTIHNYANEDXGTIPLREAFA  263\n",
      "\n",
      "Query  823   LSCNTVFGKMGAELGGDKLRAQAEKFGFNKEVTIPIRADASHFPSGSKPD------GTAM  984\n",
      "              S NT   ++G  LG D L + A  FG+   +        S  P+ ++         +  \n",
      "Sbjct  264   RSSNTALAQLGVALGADNLVSYARAFGYGTALGQDFSTTPSLXPNPAEXTTWELAWASCG  323\n",
      "\n",
      "Query  985   DAIGQHDTRATP----LQMAMVAAAIANNGSLMEPYLVAQEGS-AGNVISTHTEHQLSQA  1149\n",
      "               +G+H + A P     Q A++AAAIAN G +  PY+V +  S  G V+ST +   L QA\n",
      "Sbjct  324   LPVGEHASPAGPQTTVXQNAVIAAAIANGGVVXNPYIVDRVLSPEGAVVSTTSPKSLGQA  383\n",
      "\n",
      "Query  1150  VSPATAQKLQQLMESVVQHGTGTNAKIPGVTVGGKTGTA--QHGQDNSGLPFAWFVSYAK  1323\n",
      "             VS  TA ++++    VV+ GTG  A++PGV + GKTGTA  ++G  NS     +F+ +A \n",
      "Sbjct  384   VSADTAAQVREAXLGVVESGTGXGARVPGVKIAGKTGTADVENGNFNS-----FFIGFAP  438\n",
      "\n",
      "Query  1324  GQDGKQVAVAVVVE  1365\n",
      "               D   + V+VV+E\n",
      "Sbjct  439   -YDHPTLVVSVVIE  451\n",
      "\n",
      "\n",
      ">5E31_A Chain A, 2.3 Angstrom Crystal Structure Of The Monomeric Form \n",
      "Of Penicillin Binding Protein 2 Prime From Enterococcus Faecium.\n",
      "Length=644\n",
      "\n",
      " Score = 115 bits (289),  Expect = 4e-27, Method: Compositional matrix adjust.\n",
      " Identities = 98/343 (29%), Positives = 158/343 (46%), Gaps = 29/343 (8%)\n",
      " Frame = +1\n",
      "\n",
      "Query  319   GILSGTDSRLFFRNTLDMLTGKPKQGGDVVTTIDPKVQKAGFEGLGNKKGAAVAIDPKTG  498\n",
      "             G LS TD+    +  L  +  + + G D+  TID K QK  F+ LG K G+ VA  PKTG\n",
      "Sbjct  283   GKLSITDADGVEKKVL--IEHEVQNGKDIKLTIDAKAQKTAFDSLGGKAGSTVATTPKTG  340\n",
      "\n",
      "Query  499   AILGLVSTPSYDPGTFAGGTKDDEKAWTALDSDPNKPMLNRALRETYPPGSTFKLVTAAT  678\n",
      "              +L L S+PSYDP     G   ++  + A + +P +P ++R     Y PGSTFK +TAA \n",
      "Sbjct  341   DLLALASSPSYDPNKXTNGISQED--YKAYEENPEQPFISR-FATGYAPGSTFKXITAAI  397\n",
      "\n",
      "Query  679   AFETGKYQSPSDVTDTPDQYILPGTSTPLINASPTEDCGNATVQHAMDLSCNTVFGKMGA  858\n",
      "               + G    P++V           +S      +   D     ++ A+  S N    +   \n",
      "Sbjct  398   GLDNGTI-DPNEVLTINGLKWQKDSSWGSYQVTRVSDVSQVDLKTALIYSDNIYXAQETL  456\n",
      "\n",
      "Query  859   ELGGDKLRAQAEKFGFNKEVTIPIRADASHFPSGSKPDGTAMDAI------GQHDTRATP  1020\n",
      "             + G  K R   +KF F +++ +PI  + +     S  D    D +      GQ +    P\n",
      "Sbjct  457   KXGEKKFRTGLDKFIFGEDLDLPISXNPAQI---SNEDSFNSDILLADTGYGQGELLINP  513\n",
      "\n",
      "Query  1021  LQMAMVAAAIANNGSLMEPYLVAQEGSAG--NVISTHTEHQLSQAVSPATAQKLQQLMES  1194\n",
      "             +Q A   +  ANNG+L+ P L+A + +    NVI         Q + P     L+++++ \n",
      "Sbjct  514   IQQAAXYSVFANNGTLVYPKLIADKETKDKKNVIG----ETAVQTIVP----DLREVVQD  565\n",
      "\n",
      "Query  1195  VVQHGTGTNAKIPGVTVGGKTGTAQ--HGQDNSGLPFAWFVSY  1317\n",
      "             V  +GT  +    G+ +  KTGTA+    QD  G   ++  ++\n",
      "Sbjct  566   V--NGTAHSLSALGIPLAAKTGTAEIKEKQDEKGKENSFLFAF  606\n",
      "\n",
      "\n",
      ">4CPK_A Chain A, Crystal Structure Of Pbp2a Double Clinical Mutant N146k- \n",
      "E150k From Mrsa\n",
      " 4CPK_B Chain B, Crystal Structure Of Pbp2a Double Clinical Mutant N146k- \n",
      "E150k From Mrsa\n",
      "Length=643\n",
      "\n",
      " Score = 114 bits (285),  Expect = 1e-26, Method: Compositional matrix adjust.\n",
      " Identities = 107/384 (28%), Positives = 165/384 (43%), Gaps = 59/384 (15%)\n",
      " Frame = +1\n",
      "\n",
      "Query  301   LEKLEDGILSGTDSRLFFRNTL----------DMLTGKPKQGGDVVTTIDPKVQKAGFEG  450\n",
      "             LEKL D  L   D    +R T+           ++  K K G D+  TID KVQK+ +  \n",
      "Sbjct  258   LEKLYDKKLQHEDG---YRVTIVDDNSNTIAHTLIEKKKKDGKDIQLTIDAKVQKSIYNN  314\n",
      "\n",
      "Query  451   LGNKKGAAVAIDPKTGAILGLVSTPSYDPGTFAGGTKDDEKAWTALDSDPNKPMLNRALR  630\n",
      "             + N  G+  AI P+TG +L LVSTPSYD   F  G  ++E  +  L  D  +P+LN+  +\n",
      "Sbjct  315   MKNDYGSGTAIHPQTGELLALVSTPSYDVYPFMYGMSNEE--YNKLTEDKKEPLLNK-FQ  371\n",
      "\n",
      "Query  631   ETYPPGSTFKLVTAATAFETGKYQSPSDVTDTPDQYILPG------TSTPLINASPTEDC  792\n",
      "              T  PGST K++TA            +   D    Y + G       S    N +  E  \n",
      "Sbjct  372   ITTSPGSTQKILTAMIGLN-------NKTLDDKTSYKIDGKGWQKDKSWGGYNVTRYEVV  424\n",
      "\n",
      "Query  793   -GNATVQHAMDLSCNTVFGKMGAELGGDKLRAQAEKFGFNKEVTIPIRADASHFP-----  954\n",
      "              GN  ++ A++ S N  F ++  ELG  K     +K G  +++        S +P     \n",
      "Sbjct  425   NGNIDLKQAIESSDNIFFARVALELGSKKFEKGMKKLGVGEDI-------PSDYPFYNAQ  477\n",
      "\n",
      "Query  955   -SGSKPDGTAMDA---IGQHDTRATPLQMAMVAAAIANNGSLMEPYLVAQEGSAGNVIST  1122\n",
      "              S    D   + A    GQ +    P+Q+  + +A+ NNG++  P+L+           T\n",
      "Sbjct  478   ISNKNLDNEILLADSGYGQGEILINPVQILSIYSALENNGNINAPHLLK---------DT  528\n",
      "\n",
      "Query  1123  HTEHQLSQAVSPATAQKLQQLMESVVQHGTGTNAKIPGVTVGGKTGTAQ--HGQDNSGLP  1296\n",
      "               +      +S      L   M+ VV      +       + GK+GTA+    Q  +G  \n",
      "Sbjct  529   KNKVWKKNIISKENINLLTDGMQQVVNKTHKEDIYRSYANLIGKSGTAELKMKQGETGRQ  588\n",
      "\n",
      "Query  1297  FAWFVSYAKGQDGKQVAVAVVVED  1368\n",
      "               WF+SY K  D   + +A+ V+D\n",
      "Sbjct  589   IGWFISYDK--DNPNMMMAINVKD  610\n",
      "\n",
      "\n",
      ">1VQQ_A Chain A, Structure Of Penicillin Binding Protein 2a From Methicillin \n",
      "Resistant Staphylococcus Aureus Strain 27r At 1.80 A \n",
      "Resolution.\n",
      " 1VQQ_B Chain B, Structure Of Penicillin Binding Protein 2a From Methicillin \n",
      "Resistant Staphylococcus Aureus Strain 27r At 1.80 A \n",
      "Resolution.\n",
      " 1MWU_A Chain A, Structure Of Methicillin Acyl-Penicillin Binding Protein \n",
      "2a From Methicillin Resistant Staphylococcus Aureus Strain \n",
      "27r At 2.60 A Resolution.\n",
      " 1MWU_B Chain B, Structure Of Methicillin Acyl-Penicillin Binding Protein \n",
      "2a From Methicillin Resistant Staphylococcus Aureus Strain \n",
      "27r At 2.60 A Resolution.\n",
      " 4DKI_A Chain A, Structural Insights Into The Anti- Methicillin-Resistant \n",
      "Staphylococcus Aureus (Mrsa) Activity Of Ceftobiprole\n",
      " 4DKI_B Chain B, Structural Insights Into The Anti- Methicillin-Resistant \n",
      "Staphylococcus Aureus (Mrsa) Activity Of Ceftobiprole\n",
      "Length=646\n",
      "\n",
      " Score = 114 bits (285),  Expect = 1e-26, Method: Compositional matrix adjust.\n",
      " Identities = 107/384 (28%), Positives = 165/384 (43%), Gaps = 59/384 (15%)\n",
      " Frame = +1\n",
      "\n",
      "Query  301   LEKLEDGILSGTDSRLFFRNTL----------DMLTGKPKQGGDVVTTIDPKVQKAGFEG  450\n",
      "             LEKL D  L   D    +R T+           ++  K K G D+  TID KVQK+ +  \n",
      "Sbjct  261   LEKLYDKKLQHEDG---YRVTIVDDNSNTIAHTLIEKKKKDGKDIQLTIDAKVQKSIYNN  317\n",
      "\n",
      "Query  451   LGNKKGAAVAIDPKTGAILGLVSTPSYDPGTFAGGTKDDEKAWTALDSDPNKPMLNRALR  630\n",
      "             + N  G+  AI P+TG +L LVSTPSYD   F  G  ++E  +  L  D  +P+LN+  +\n",
      "Sbjct  318   MKNDYGSGTAIHPQTGELLALVSTPSYDVYPFMYGMSNEE--YNKLTEDKKEPLLNK-FQ  374\n",
      "\n",
      "Query  631   ETYPPGSTFKLVTAATAFETGKYQSPSDVTDTPDQYILPG------TSTPLINASPTEDC  792\n",
      "              T  PGST K++TA            +   D    Y + G       S    N +  E  \n",
      "Sbjct  375   ITTSPGSTQKILTAMIGLN-------NKTLDDKTSYKIDGKGWQKDKSWGGYNVTRYEVV  427\n",
      "\n",
      "Query  793   -GNATVQHAMDLSCNTVFGKMGAELGGDKLRAQAEKFGFNKEVTIPIRADASHFP-----  954\n",
      "              GN  ++ A++ S N  F ++  ELG  K     +K G  +++        S +P     \n",
      "Sbjct  428   NGNIDLKQAIESSDNIFFARVALELGSKKFEKGMKKLGVGEDI-------PSDYPFYNAQ  480\n",
      "\n",
      "Query  955   -SGSKPDGTAMDA---IGQHDTRATPLQMAMVAAAIANNGSLMEPYLVAQEGSAGNVIST  1122\n",
      "              S    D   + A    GQ +    P+Q+  + +A+ NNG++  P+L+           T\n",
      "Sbjct  481   ISNKNLDNEILLADSGYGQGEILINPVQILSIYSALENNGNINAPHLLK---------DT  531\n",
      "\n",
      "Query  1123  HTEHQLSQAVSPATAQKLQQLMESVVQHGTGTNAKIPGVTVGGKTGTAQ--HGQDNSGLP  1296\n",
      "               +      +S      L   M+ VV      +       + GK+GTA+    Q  +G  \n",
      "Sbjct  532   KNKVWKKNIISKENINLLTDGMQQVVNKTHKEDIYRSYANLIGKSGTAELKMKQGETGRQ  591\n",
      "\n",
      "Query  1297  FAWFVSYAKGQDGKQVAVAVVVED  1368\n",
      "               WF+SY K  D   + +A+ V+D\n",
      "Sbjct  592   IGWFISYDK--DNPNMMMAINVKD  613\n",
      "\n",
      "\n",
      ">4BL2_A Chain A, Crystal Structure Of Pbp2a Clinical Mutant E150k From \n",
      "Mrsa\n",
      " 4BL2_B Chain B, Crystal Structure Of Pbp2a Clinical Mutant E150k From \n",
      "Mrsa\n",
      "Length=643\n",
      "\n",
      " Score = 114 bits (285),  Expect = 1e-26, Method: Compositional matrix adjust.\n",
      " Identities = 107/384 (28%), Positives = 165/384 (43%), Gaps = 59/384 (15%)\n",
      " Frame = +1\n",
      "\n",
      "Query  301   LEKLEDGILSGTDSRLFFRNTL----------DMLTGKPKQGGDVVTTIDPKVQKAGFEG  450\n",
      "             LEKL D  L   D    +R T+           ++  K K G D+  TID KVQK+ +  \n",
      "Sbjct  258   LEKLYDKKLQHEDG---YRVTIVDDNSNTIAHTLIEKKKKDGKDIQLTIDAKVQKSIYNN  314\n",
      "\n",
      "Query  451   LGNKKGAAVAIDPKTGAILGLVSTPSYDPGTFAGGTKDDEKAWTALDSDPNKPMLNRALR  630\n",
      "             + N  G+  AI P+TG +L LVSTPSYD   F  G  ++E  +  L  D  +P+LN+  +\n",
      "Sbjct  315   MKNDYGSGTAIHPQTGELLALVSTPSYDVYPFMYGMSNEE--YNKLTEDKKEPLLNK-FQ  371\n",
      "\n",
      "Query  631   ETYPPGSTFKLVTAATAFETGKYQSPSDVTDTPDQYILPG------TSTPLINASPTEDC  792\n",
      "              T  PGST K++TA            +   D    Y + G       S    N +  E  \n",
      "Sbjct  372   ITTSPGSTQKILTAMIGLN-------NKTLDDKTSYKIDGKGWQKDKSWGGYNVTRYEVV  424\n",
      "\n",
      "Query  793   -GNATVQHAMDLSCNTVFGKMGAELGGDKLRAQAEKFGFNKEVTIPIRADASHFP-----  954\n",
      "              GN  ++ A++ S N  F ++  ELG  K     +K G  +++        S +P     \n",
      "Sbjct  425   NGNIDLKQAIESSDNIFFARVALELGSKKFEKGMKKLGVGEDI-------PSDYPFYNAQ  477\n",
      "\n",
      "Query  955   -SGSKPDGTAMDA---IGQHDTRATPLQMAMVAAAIANNGSLMEPYLVAQEGSAGNVIST  1122\n",
      "              S    D   + A    GQ +    P+Q+  + +A+ NNG++  P+L+           T\n",
      "Sbjct  478   ISNKNLDNEILLADSGYGQGEILINPVQILSIYSALENNGNINAPHLLK---------DT  528\n",
      "\n",
      "Query  1123  HTEHQLSQAVSPATAQKLQQLMESVVQHGTGTNAKIPGVTVGGKTGTAQ--HGQDNSGLP  1296\n",
      "               +      +S      L   M+ VV      +       + GK+GTA+    Q  +G  \n",
      "Sbjct  529   KNKVWKKNIISKENINLLTDGMQQVVNKTHKEDIYRSYANLIGKSGTAELKMKQGETGRQ  588\n",
      "\n",
      "Query  1297  FAWFVSYAKGQDGKQVAVAVVVED  1368\n",
      "               WF+SY K  D   + +A+ V+D\n",
      "Sbjct  589   IGWFISYDK--DNPNMMMAINVKD  610\n",
      "\n",
      "\n",
      ">3ZG5_A Chain A, Crystal Structure Of Pbp2a From Mrsa In Complex With \n",
      "Peptidoglycan Analogue At Allosteric\n",
      " 3ZG5_B Chain B, Crystal Structure Of Pbp2a From Mrsa In Complex With \n",
      "Peptidoglycan Analogue At Allosteric\n",
      " 3ZG0_A Chain A, Crystal Structure Of Ceftaroline Acyl-pbp2a From Mrsa \n",
      "With Non-covalently Bound Ceftaroline And Muramic Acid At \n",
      "Allosteric Site Obtained By Cocrystallization\n",
      " 3ZG0_B Chain B, Crystal Structure Of Ceftaroline Acyl-pbp2a From Mrsa \n",
      "With Non-covalently Bound Ceftaroline And Muramic Acid At \n",
      "Allosteric Site Obtained By Cocrystallization\n",
      " 3ZFZ_A Chain A, Crystal Structure Of Ceftaroline Acyl-pbp2a From Mrsa \n",
      "With Non-covalently Bound Ceftaroline And Muramic Acid At \n",
      "Allosteric Site Obtained By Soaking\n",
      " 3ZFZ_B Chain B, Crystal Structure Of Ceftaroline Acyl-pbp2a From Mrsa \n",
      "With Non-covalently Bound Ceftaroline And Muramic Acid At \n",
      "Allosteric Site Obtained By Soaking\n",
      " 4CJN_A Chain A, Crystal Structure Of Pbp2a From Mrsa In Complex With \n",
      "Quinazolinone Ligand\n",
      " 4CJN_B Chain B, Crystal Structure Of Pbp2a From Mrsa In Complex With \n",
      "Quinazolinone Ligand\n",
      " 5M18_A Chain A, Crystal Structure Of Pbp2a From Mrsa In The Presence \n",
      "Of Cefepime Ligand\n",
      " 5M18_B Chain B, Crystal Structure Of Pbp2a From Mrsa In The Presence \n",
      "Of Cefepime Ligand\n",
      " 5M19_A Chain A, Crystal Structure Of Pbp2a From Mrsa In The Presence \n",
      "Of Oxacillin Ligand\n",
      " 5M19_B Chain B, Crystal Structure Of Pbp2a From Mrsa In The Presence \n",
      "Of Oxacillin Ligand\n",
      " 5M1A_A Chain A, Crystal Structure Of Pbp2a From Mrsa In The Presence \n",
      "Of Ceftazidime Ligand\n",
      " 5M1A_B Chain B, Crystal Structure Of Pbp2a From Mrsa In The Presence \n",
      "Of Ceftazidime Ligand\n",
      "Length=642\n",
      "\n",
      " Score = 114 bits (285),  Expect = 1e-26, Method: Compositional matrix adjust.\n",
      " Identities = 107/384 (28%), Positives = 165/384 (43%), Gaps = 59/384 (15%)\n",
      " Frame = +1\n",
      "\n",
      "Query  301   LEKLEDGILSGTDSRLFFRNTL----------DMLTGKPKQGGDVVTTIDPKVQKAGFEG  450\n",
      "             LEKL D  L   D    +R T+           ++  K K G D+  TID KVQK+ +  \n",
      "Sbjct  257   LEKLYDKKLQHEDG---YRVTIVDDNSNTIAHTLIEKKKKDGKDIQLTIDAKVQKSIYNN  313\n",
      "\n",
      "Query  451   LGNKKGAAVAIDPKTGAILGLVSTPSYDPGTFAGGTKDDEKAWTALDSDPNKPMLNRALR  630\n",
      "             + N  G+  AI P+TG +L LVSTPSYD   F  G  ++E  +  L  D  +P+LN+  +\n",
      "Sbjct  314   MKNDYGSGTAIHPQTGELLALVSTPSYDVYPFMYGMSNEE--YNKLTEDKKEPLLNK-FQ  370\n",
      "\n",
      "Query  631   ETYPPGSTFKLVTAATAFETGKYQSPSDVTDTPDQYILPG------TSTPLINASPTEDC  792\n",
      "              T  PGST K++TA            +   D    Y + G       S    N +  E  \n",
      "Sbjct  371   ITTSPGSTQKILTAMIGLN-------NKTLDDKTSYKIDGKGWQKDKSWGGYNVTRYEVV  423\n",
      "\n",
      "Query  793   -GNATVQHAMDLSCNTVFGKMGAELGGDKLRAQAEKFGFNKEVTIPIRADASHFP-----  954\n",
      "              GN  ++ A++ S N  F ++  ELG  K     +K G  +++        S +P     \n",
      "Sbjct  424   NGNIDLKQAIESSDNIFFARVALELGSKKFEKGMKKLGVGEDI-------PSDYPFYNAQ  476\n",
      "\n",
      "Query  955   -SGSKPDGTAMDA---IGQHDTRATPLQMAMVAAAIANNGSLMEPYLVAQEGSAGNVIST  1122\n",
      "              S    D   + A    GQ +    P+Q+  + +A+ NNG++  P+L+           T\n",
      "Sbjct  477   ISNKNLDNEILLADSGYGQGEILINPVQILSIYSALENNGNINAPHLLK---------DT  527\n",
      "\n",
      "Query  1123  HTEHQLSQAVSPATAQKLQQLMESVVQHGTGTNAKIPGVTVGGKTGTAQ--HGQDNSGLP  1296\n",
      "               +      +S      L   M+ VV      +       + GK+GTA+    Q  +G  \n",
      "Sbjct  528   KNKVWKKNIISKENINLLTDGMQQVVNKTHKEDIYRSYANLIGKSGTAELKMKQGETGRQ  587\n",
      "\n",
      "Query  1297  FAWFVSYAKGQDGKQVAVAVVVED  1368\n",
      "               WF+SY K  D   + +A+ V+D\n",
      "Sbjct  588   IGWFISYDK--DNPNMMMAINVKD  609\n",
      "\n",
      "\n",
      ">4BL3_A Chain A, Crystal Structure Of Pbp2a Clinical Mutant N146k From \n",
      "Mrsa\n",
      " 4BL3_B Chain B, Crystal Structure Of Pbp2a Clinical Mutant N146k From \n",
      "Mrsa\n",
      "Length=640\n",
      "\n",
      " Score = 112 bits (281),  Expect = 4e-26, Method: Compositional matrix adjust.\n",
      " Identities = 106/382 (28%), Positives = 163/382 (43%), Gaps = 58/382 (15%)\n",
      " Frame = +1\n",
      "\n",
      "Query  301   LEKLEDGILSGTDSRLFFRNTL----------DMLTGKPKQGGDVVTTIDPKVQKAGFEG  450\n",
      "             LEKL D  L   D    +R T+           ++  K K G D+  TID KVQK+ +  \n",
      "Sbjct  258   LEKLYDKKLQHEDG---YRVTIVDDNSNTIAHTLIEKKKKDGKDIQLTIDAKVQKSIYNN  314\n",
      "\n",
      "Query  451   LGNKKGAAVAIDPKTGAILGLVSTPSYDPGTFAGGTKDDEKAWTALDSDPNKPMLNRALR  630\n",
      "             + N  G+  AI P+TG +L LVSTPSYD   F  G  ++E  +  L  D  +P+LN+  +\n",
      "Sbjct  315   MKNDYGSGTAIHPQTGELLALVSTPSYDVYPFMYGMSNEE--YNKLTEDKKEPLLNK-FQ  371\n",
      "\n",
      "Query  631   ETYPPGSTFKLVTAATAFETGKYQSPSDVTDTPDQYILPG------TSTPLINASPTEDC  792\n",
      "              T  PGST K++TA            +   D    Y + G       S    N +  E  \n",
      "Sbjct  372   ITTSPGSTQKILTAMIGLN-------NKTLDDKTSYKIDGKGWQKDKSWGGYNVTRYEVV  424\n",
      "\n",
      "Query  793   -GNATVQHAMDLSCNTVFGKMGAELGGDKLRAQAEKFGFNKEVTIPIRADASHFP-----  954\n",
      "              GN  ++ A++ S N  F ++  ELG  K     +K G  +++        S +P     \n",
      "Sbjct  425   NGNIDLKQAIESSDNIFFARVALELGSKKFEKGMKKLGVGEDI-------PSDYPFYNAQ  477\n",
      "\n",
      "Query  955   -SGSKPDGTAMDA---IGQHDTRATPLQMAMVAAAIANNGSLMEPYLVAQEGSAGNVIST  1122\n",
      "              S    D   + A    GQ +    P+Q+  + +A+ NNG++  P+L+           T\n",
      "Sbjct  478   ISNKNLDNEILLADSGYGQGEILINPVQILSIYSALENNGNINAPHLLK---------DT  528\n",
      "\n",
      "Query  1123  HTEHQLSQAVSPATAQKLQQLMESVVQHGTGTNAKIPGVTVGGKTGTAQHGQDNSGLPFA  1302\n",
      "               +      +S      L   M+ VV      +       + GK+GTA+      G    \n",
      "Sbjct  529   KNKVWKKNIISKENINLLTDGMQQVVNKTHKEDIYRSYANLIGKSGTAELKM-KQGRQIG  587\n",
      "\n",
      "Query  1303  WFVSYAKGQDGKQVAVAVVVED  1368\n",
      "             WF+SY K  D   + +A+ V+D\n",
      "Sbjct  588   WFISYDK--DNPNMMMAINVKD  607\n",
      "\n",
      "\n",
      ">1MWS_A Chain A, Structure Of Nitrocefin Acyl-Penicillin Binding Protein \n",
      "2a From Methicillin Resistant Staphylococcus Aureus Strain \n",
      "27r At 2.00 A Resolution.\n",
      " 1MWS_B Chain B, Structure Of Nitrocefin Acyl-Penicillin Binding Protein \n",
      "2a From Methicillin Resistant Staphylococcus Aureus Strain \n",
      "27r At 2.00 A Resolution.\n",
      " 1MWT_A Chain A, Structure Of Penicillin G Acyl-Penicillin Binding Protein \n",
      "2a From Methicillin Resistant Staphylococcus Aureus Strain \n",
      "27r At 2.45 A Resolution.\n",
      " 1MWT_B Chain B, Structure Of Penicillin G Acyl-Penicillin Binding Protein \n",
      "2a From Methicillin Resistant Staphylococcus Aureus Strain \n",
      "27r At 2.45 A Resolution.\n",
      "Length=646\n",
      "\n",
      " Score = 112 bits (280),  Expect = 5e-26, Method: Compositional matrix adjust.\n",
      " Identities = 106/384 (28%), Positives = 164/384 (43%), Gaps = 59/384 (15%)\n",
      " Frame = +1\n",
      "\n",
      "Query  301   LEKLEDGILSGTDSRLFFRNTL----------DMLTGKPKQGGDVVTTIDPKVQKAGFEG  450\n",
      "             LEKL D  L   D    +R T+           ++  K K G D+  TID KVQK+ +  \n",
      "Sbjct  261   LEKLYDKKLQHEDG---YRVTIVDDNSNTIAHTLIEKKKKDGKDIQLTIDAKVQKSIYNN  317\n",
      "\n",
      "Query  451   LGNKKGAAVAIDPKTGAILGLVSTPSYDPGTFAGGTKDDEKAWTALDSDPNKPMLNRALR  630\n",
      "             + N  G+  AI P+TG +L LVSTPSYD   F  G  ++E  +  L  D  +P+LN+  +\n",
      "Sbjct  318   MKNDYGSGTAIHPQTGELLALVSTPSYDVYPFMYGMSNEE--YNKLTEDKKEPLLNK-FQ  374\n",
      "\n",
      "Query  631   ETYPPGSTFKLVTAATAFETGKYQSPSDVTDTPDQYILPG------TSTPLINASPTEDC  792\n",
      "              T  PG T K++TA            +   D    Y + G       S    N +  E  \n",
      "Sbjct  375   ITTSPGXTQKILTAMIGLN-------NKTLDDKTSYKIDGKGWQKDKSWGGYNVTRYEVV  427\n",
      "\n",
      "Query  793   -GNATVQHAMDLSCNTVFGKMGAELGGDKLRAQAEKFGFNKEVTIPIRADASHFP-----  954\n",
      "              GN  ++ A++ S N  F ++  ELG  K     +K G  +++        S +P     \n",
      "Sbjct  428   NGNIDLKQAIESSDNIFFARVALELGSKKFEKGMKKLGVGEDI-------PSDYPFYNAQ  480\n",
      "\n",
      "Query  955   -SGSKPDGTAMDA---IGQHDTRATPLQMAMVAAAIANNGSLMEPYLVAQEGSAGNVIST  1122\n",
      "              S    D   + A    GQ +    P+Q+  + +A+ NNG++  P+L+           T\n",
      "Sbjct  481   ISNKNLDNEILLADSGYGQGEILINPVQILSIYSALENNGNINAPHLLK---------DT  531\n",
      "\n",
      "Query  1123  HTEHQLSQAVSPATAQKLQQLMESVVQHGTGTNAKIPGVTVGGKTGTAQ--HGQDNSGLP  1296\n",
      "               +      +S      L   M+ VV      +       + GK+GTA+    Q  +G  \n",
      "Sbjct  532   KNKVWKKNIISKENINLLTDGMQQVVNKTHKEDIYRSYANLIGKSGTAELKMKQGETGRQ  591\n",
      "\n",
      "Query  1297  FAWFVSYAKGQDGKQVAVAVVVED  1368\n",
      "               WF+SY K  D   + +A+ V+D\n",
      "Sbjct  592   IGWFISYDK--DNPNMMMAINVKD  613\n",
      "\n",
      "\n",
      ">1MWR_A Chain A, Structure Of Semet Penicillin Binding Protein 2a From \n",
      "Methicillin Resistant Staphylococcus Aureus Strain 27r (Trigonal \n",
      "Form) At 2.45 A Resolution.\n",
      " 1MWR_B Chain B, Structure Of Semet Penicillin Binding Protein 2a From \n",
      "Methicillin Resistant Staphylococcus Aureus Strain 27r (Trigonal \n",
      "Form) At 2.45 A Resolution.\n",
      "Length=646\n",
      "\n",
      " Score = 108 bits (271),  Expect = 7e-25, Method: Compositional matrix adjust.\n",
      " Identities = 105/387 (27%), Positives = 165/387 (43%), Gaps = 71/387 (18%)\n",
      " Frame = +1\n",
      "\n",
      "Query  301   LEKLEDGILSGTDSRLFFRNTL----------DMLTGKPKQGGDVVTTIDPKVQKAGFEG  450\n",
      "             LEKL D  L   D    +R T+           ++  K K G D+  TID KVQK+ +  \n",
      "Sbjct  261   LEKLYDKKLQHEDG---YRVTIVDDNSNTIAHTLIEKKKKDGKDIQLTIDAKVQKSIYNN  317\n",
      "\n",
      "Query  451   LGNKKGAAVAIDPKTGAILGLVSTPSYDPGTFAGGTKDDEKAWTALDSDPNKPMLNRALR  630\n",
      "               N  G+  AI P+TG +L LVSTPSYD   F  G  ++E  +  L  D  +P+LN+  +\n",
      "Sbjct  318   XKNDYGSGTAIHPQTGELLALVSTPSYDVYPFXYGXSNEE--YNKLTEDKKEPLLNK-FQ  374\n",
      "\n",
      "Query  631   ETYPPGSTFKLVTAATAFETGKYQSPSDVTDTPDQYILPG------TSTPLINASPTEDC  792\n",
      "              T  PGST K++TA            +   D    Y + G       S    N +  E  \n",
      "Sbjct  375   ITTSPGSTQKILTAXIGLN-------NKTLDDKTSYKIDGKGWQKDKSWGGYNVTRYEVV  427\n",
      "\n",
      "Query  793   -GNATVQHAMDLSCNTVFGKMGAELGGDKLRAQAEKFGFNKEVTIPIRADASHFP-----  954\n",
      "              GN  ++ A++ S N  F ++  ELG  K     +K G  +++        S +P     \n",
      "Sbjct  428   NGNIDLKQAIESSDNIFFARVALELGSKKFEKGXKKLGVGEDI-------PSDYPFYNAQ  480\n",
      "\n",
      "Query  955   -SGSKPDGTAMDA---IGQHDTRATPLQMAMVAAAIANNGSLMEPYLVAQEGS---AGNV  1113\n",
      "              S    D   + A    GQ +    P+Q+  + +A+ NNG++  P+L+    +     N+\n",
      "Sbjct  481   ISNKNLDNEILLADSGYGQGEILINPVQILSIYSALENNGNINAPHLLKDTKNKVWKKNI  540\n",
      "\n",
      "Query  1114  ISTHTEHQLS----QAVSPATAQKLQQLMESVVQHGTGTNAKIPGVTVGGKTGTAQ--HG  1275\n",
      "             IS    + L+    Q V+    + + +   +++                GK+GTA+    \n",
      "Sbjct  541   ISKENINLLTDGXQQVVNKTHKEDIYRSYANLI----------------GKSGTAELKXK  584\n",
      "\n",
      "Query  1276  QDNSGLPFAWFVSYAKGQDGKQVAVAV  1356\n",
      "             Q  +G    WF+SY K       A+ V\n",
      "Sbjct  585   QGETGRQIGWFISYDKDNPNXXXAINV  611\n",
      "\n",
      "\n",
      ">3UE3_A Chain A, Crystal Structure Of Acinetobacter Baumanni Pbp3\n",
      "Length=554\n",
      "\n",
      " Score = 93.2 bits (230),  Expect = 8e-20, Method: Compositional matrix adjust.\n",
      " Identities = 104/372 (28%), Positives = 163/372 (44%), Gaps = 55/372 (15%)\n",
      " Frame = +1\n",
      "\n",
      "Query  388   KQGGDVVTTIDPKVQKAGFE-----GLGN--KKGAAVAIDPKTGAILGLVSTPSYDPGTF  546\n",
      "             + G ++  +ID ++Q   +      G+ N  +   A+A+D KTG IL + S PSY+P   \n",
      "Sbjct  200   ESGENITLSIDSRLQYIMYRELTAAGVANNARSATAIAVDVKTGEILAMTSWPSYNPNDK  259\n",
      "\n",
      "Query  547   AGGTKDDEKAWTALDSDPNKPMLNRALRETYPPGSTFKLVTAATAFETGKYQSPSDVTDT  726\n",
      "              G +  D              M NR   + + PGST K  T A A E+G+Y +P+ + +T\n",
      "Sbjct  260   GGLSNKDA-------------MRNRGAIDMFEPGSTMKPFTVAAALESGQY-TPNSIVNT  305\n",
      "\n",
      "Query  727   PDQYILPGTSTPLINASPTEDCGNATVQHAMDLSCNTVFGKMGAELGGDKLRAQAEKFGF  906\n",
      "                 +  G  T       T + G  T+   +  S N    K+   L  + L +   + GF\n",
      "Sbjct  306   APGTMRLGWHT----IRDTHNYGALTLTGIIVKSSNVGSAKLALSLPKEALPSFFRRAGF  361\n",
      "\n",
      "Query  907   NKEVTIPIRAD-ASHFPSGSKPDGTAMDAIGQ-HDTRATPLQMAMVAAAIANNGSLMEPY  1080\n",
      "              +   +    + A    S  K + + +  +   +   AT LQ+A   A +AN+G  M   \n",
      "Sbjct  362   GQSSDVKFPGESAGLLYSADKLNSSQLGTMAYGYGLNATILQLAQGYAMLANHGVEMP--  419\n",
      "\n",
      "Query  1081  LVAQEGSAGNVISTHTEHQL---SQAVSPATAQKLQQLMESV-VQHGTGTNAKIPGVTVG  1248\n",
      "                        +S H   Q+    Q + P  A ++  ++E V +  GT   A IPG  VG\n",
      "Sbjct  420   -----------LSLHKLDQVPEGRQVLDPKIADQVLMMLEQVTLPGGTAKQAVIPGYRVG  468\n",
      "\n",
      "Query  1249  GKTGTA------QHGQDNSGLPFAWFVSYAKGQDGKQVAVAVVVEDGSSEAQQISGGSLA  1410\n",
      "             GKTGTA      + G  NS    A F   A   D + +A+ VVVE+      +  GG +A\n",
      "Sbjct  469   GKTGTAHKLRADRKGYSNSEY-RALFAGVAPVSDPR-LAMIVVVENPQG---RYYGGLVA  523\n",
      "\n",
      "Query  1411  APIAKAMMQAAL  1446\n",
      "             AP+   +MQ +L\n",
      "Sbjct  524   APVFAKIMQESL  535\n",
      "\n",
      "\n",
      ">5TRO_A Chain A, 1.8 Angstrom Resolution Crystal Structure Of Dimerization \n",
      "And Transpeptidase Domains (residues 39-608) Of Penicillin-binding \n",
      "Protein 1 From Staphylococcus Aureus.\n",
      " 5TRO_B Chain B, 1.8 Angstrom Resolution Crystal Structure Of Dimerization \n",
      "And Transpeptidase Domains (residues 39-608) Of Penicillin-binding \n",
      "Protein 1 From Staphylococcus Aureus.\n",
      "Length=573\n",
      "\n",
      " Score = 91.7 bits (226),  Expect = 3e-19, Method: Compositional matrix adjust.\n",
      " Identities = 117/438 (27%), Positives = 178/438 (41%), Gaps = 67/438 (15%)\n",
      " Frame = +1\n",
      "\n",
      "Query  226   KRSWVDGPMYAPVTGYSSQTYDASQL------EKLEDGILSGTDSRLFFRNTL-------  366\n",
      "             +R + +G   + + G + +  D  +L      EK+ D  LSG+   L + + +       \n",
      "Sbjct  133   ERFYPNGNFASHLIGRAQKNPDTGELKGALGVEKIFDSYLSGSKGSLRYIHDIWGYIAPN  192\n",
      "\n",
      "Query  367   DMLTGKPKQGGDVVTTIDPKVQKAGFEGLGN-------KKGAAVAIDPKTGAILGLVSTP  525\n",
      "                  +PK+G DV  TID  +Q    E L         K   AV  D KTG IL     P\n",
      "Sbjct  193   TKKEKQPKRGDDVHLTIDSNIQVFVEEALDGXVERYQPKDLFAVVXDAKTGEILAYSQRP  252\n",
      "\n",
      "Query  526   SYDPGTFAGGTKDDEKAWTALDSDPNKPMLNRALRETYPPGSTFKLVTAATAFETGKYQS  705\n",
      "             +++P T     KD  K W            N   + TY PGSTFK    A A + G +  \n",
      "Sbjct  253   TFNPET----GKDFGKKWA-----------NDLYQNTYEPGSTFKSYGLAAAIQEGAFD-  296\n",
      "\n",
      "Query  706   PSDVTDTPDQYILPGTSTPLINASPTED-----CGNATVQHAMDLSCNTVFGKMGAELGG  870\n",
      "                    PD+    G     I  S   D      G          S NT+   +   +G \n",
      "Sbjct  297   -------PDKKYKSGHRD--IXGSRISDWNRVGWGEIPXSLGFTYSSNTLXXHLQDLVGA  347\n",
      "\n",
      "Query  871   DKLRAQAEKFGFNKEVTIPIRADASHFPSGSKPDGTAMDAIGQHDTRATPLQMAMVAAAI  1050\n",
      "             DK ++  E+FGF K        +A      S        + GQ  T  TP+Q     +A \n",
      "Sbjct  348   DKXKSWYERFGFGKSTKGXFDGEAPGQIGWSNELQQKTSSFGQ-STTVTPVQXLQAQSAF  406\n",
      "\n",
      "Query  1051  ANNGSLMEPYLVAQEGSAGNVISTHT----EHQLS-QAVSPATAQKLQQLMESVV--QHG  1209\n",
      "              N+G+ ++P+ V    S  N +S       + Q++ + ++  TA+K+++ ++ VV  +  \n",
      "Sbjct  407   FNDGNXLKPWFV---NSVENPVSKRQFYKGQKQIAGKPITKDTAEKVEKQLDLVVNSKKS  463\n",
      "\n",
      "Query  1210  TGTNAKIPGVTVGGKTGTAQHGQDNSG----LPFAWFVSYAKGQDGKQVAVAVVVEDGSS  1377\n",
      "                N +I G  V GKTGTAQ    N G     P  +FVS+    D  +    V+V  G S\n",
      "Sbjct  464   HAANYRIDGYEVEGKTGTAQVAAPNGGGYVKGPNPYFVSFX--GDAPKKNPKVIVYAGXS  521\n",
      "\n",
      "Query  1378  EAQQISGGSLAAPIAKAM  1431\n",
      "              AQ+    +    ++KA \n",
      "Sbjct  522   LAQKNDQEAYELGVSKAF  539\n",
      "\n",
      "\n",
      ">3EQV_A Chain A, Crystal Structure Of Penicillin-Binding Protein 2 From \n",
      "Neisseria Gonorrhoeae Containing Four Mutations Associated \n",
      "With Penicillin Resistance\n",
      " 3EQV_B Chain B, Crystal Structure Of Penicillin-Binding Protein 2 From \n",
      "Neisseria Gonorrhoeae Containing Four Mutations Associated \n",
      "With Penicillin Resistance\n",
      "Length=542\n",
      "\n",
      " Score = 87.8 bits (216),  Expect = 4e-18, Method: Compositional matrix adjust.\n",
      " Identities = 117/457 (26%), Positives = 191/457 (42%), Gaps = 81/457 (18%)\n",
      " Frame = +1\n",
      "\n",
      "Query  178   QAVTKSDFVNGLRYKYKRSWVDGPMYAPVTGYSSQTYDASQ-LE-KLEDGILSGTDSRLF  351\n",
      "             +A+   +FV     + KR +  G ++A V G++       + LE  LED +     + + \n",
      "Sbjct  113   KALGLENFV--FEKELKRHYPMGNLFAHVIGFTDIDGKGQEGLELSLEDSLYGEDGAEVV  170\n",
      "\n",
      "Query  352   FR----NTLDML----TGKPKQGGDVVTTIDPKVQKAGFEGLGN-------KKGAAVAID  486\n",
      "              R    N +D L       P+ G D++ ++D ++Q   +E L         K G  V +D\n",
      "Sbjct  171   LRDRQGNIVDSLDSPRNKAPQNGKDIILSLDQRIQTLAYEELNKAVEYHQAKAGTVVVLD  230\n",
      "\n",
      "Query  487   PKTGAILGLVSTPSYDPGTFAGGTKDDEKAWTALDSDPNKPMLNRALRETYPPGSTFKLV  666\n",
      "              +TG IL L +TP+YDP     G  D E+              NRA+ +   PGS  K  \n",
      "Sbjct  231   ARTGEILALANTPAYDPNR--PGRADSEQRR------------NRAVTDMIEPGSAIKPF  276\n",
      "\n",
      "Query  667   TAATAFETGKYQSPSDVTDTPDQYILPGTSTPLINASPTEDCG---NATVQHAMDLSCNT  837\n",
      "               A A + GK      +   P +          I  SP  D     +  V+  M  S N \n",
      "Sbjct  277   VIAKALDAGKTDLNERLNTQPYK----------IGPSPVRDTHVYPSLDVRGIMQKSSNV  326\n",
      "\n",
      "Query  838   VFGKMGAELGGDKLRAQAEKFGFNKEVTIPIRADASHFPSGS----------KPDGTAMD  987\n",
      "                K+ A  G +      E + F  E+ I +R   S FP  +          +P   A  \n",
      "Sbjct  327   GTSKLSARFGAE------EMYDFYHELGIGVRMH-SGFPGETAGLLRNWRRWRPIEQATM  379\n",
      "\n",
      "Query  988   AIGQHDTRATPLQMAMVAAAIANNGSLMEPYLVAQEGSAGNVISTHTEHQLSQAVSPATA  1167\n",
      "             + G +  + + LQ+A    A+ ++G L+ P    ++  A          Q  +     TA\n",
      "Sbjct  380   SFG-YGLQLSLLQLARAYTALTHDGVLL-PLSFEKQAVA---------PQGKRIFKEXTA  428\n",
      "\n",
      "Query  1168  QKLQQLMESVVQ-HGTGTNAKIPGVTVGGKTGTAQ---HGQDNSGLPFAWFVSYAKGQDG  1335\n",
      "             ++++ LM SV +  GTGT   + G  VG KTGTA+   +G+         F+ +A  ++ \n",
      "Sbjct  429   REVRNLMVSVTEPGGTGTAGAVDGFDVGAKTGTARKLVNGRYVDNKHVGTFIGFAPAKN-  487\n",
      "\n",
      "Query  1336  KQVAVAVVVEDGSSEAQQISGGSLAAPIAKAMMQAAL  1446\n",
      "              +V VAV +++ +  A    GG +A    K +M  +L\n",
      "Sbjct  488   PRVIVAVTIDEPT--AHGYYGGVVAGSPFKKIMGGSL  522\n",
      "\n",
      "\n",
      ">3EQU_A Chain A, Crystal Structure Of Penicillin-Binding Protein 2 From \n",
      "Neisseria Gonorrhoeae\n",
      " 3EQU_B Chain B, Crystal Structure Of Penicillin-Binding Protein 2 From \n",
      "Neisseria Gonorrhoeae\n",
      "Length=542\n",
      "\n",
      " Score = 87.0 bits (214),  Expect = 8e-18, Method: Compositional matrix adjust.\n",
      " Identities = 116/457 (25%), Positives = 190/457 (42%), Gaps = 81/457 (18%)\n",
      " Frame = +1\n",
      "\n",
      "Query  178   QAVTKSDFVNGLRYKYKRSWVDGPMYAPVTGYSSQTYDASQ-LE-KLEDGILSGTDSRLF  351\n",
      "             +A+   +FV     + KR +  G ++A V G++       + LE  LED +     + + \n",
      "Sbjct  113   KALGLENFV--FEKELKRHYPXGNLFAHVIGFTDIDGKGQEGLELSLEDSLYGEDGAEVV  170\n",
      "\n",
      "Query  352   FR----NTLDML----TGKPKQGGDVVTTIDPKVQKAGFEGLGN-------KKGAAVAID  486\n",
      "              R    N +D L       P+ G D++ ++D ++Q   +E L         K G  V +D\n",
      "Sbjct  171   LRDRQGNIVDSLDSPRNKAPQNGKDIILSLDQRIQTLAYEELNKAVEYHQAKAGTVVVLD  230\n",
      "\n",
      "Query  487   PKTGAILGLVSTPSYDPGTFAGGTKDDEKAWTALDSDPNKPMLNRALRETYPPGSTFKLV  666\n",
      "              +TG IL L +TP+YDP     G  D E+              NRA+ +   PGS  K  \n",
      "Sbjct  231   ARTGEILALANTPAYDPNR--PGRADSEQ------------RRNRAVTDXIEPGSAIKPF  276\n",
      "\n",
      "Query  667   TAATAFETGKYQSPSDVTDTPDQYILPGTSTPLINASPTEDCG---NATVQHAMDLSCNT  837\n",
      "               A A + GK      +   P +          I  SP  D     +  V+     S N \n",
      "Sbjct  277   VIAKALDAGKTDLNERLNTQPYK----------IGPSPVRDTHVYPSLDVRGIXQKSSNV  326\n",
      "\n",
      "Query  838   VFGKMGAELGGDKLRAQAEKFGFNKEVTIPIRADASHFPSGS----------KPDGTAMD  987\n",
      "                K+ A  G +      E + F  E+ I +R   S FP  +          +P   A  \n",
      "Sbjct  327   GTSKLSARFGAE------EXYDFYHELGIGVRXH-SGFPGETAGLLRNWRRWRPIEQATX  379\n",
      "\n",
      "Query  988   AIGQHDTRATPLQMAMVAAAIANNGSLMEPYLVAQEGSAGNVISTHTEHQLSQAVSPATA  1167\n",
      "             + G +  + + LQ+A    A+ ++G L+ P    ++  A          Q  +     TA\n",
      "Sbjct  380   SFG-YGLQLSLLQLARAYTALTHDGVLL-PLSFEKQAVA---------PQGKRIFKEXTA  428\n",
      "\n",
      "Query  1168  QKLQQLMESVVQ-HGTGTNAKIPGVTVGGKTGTAQ---HGQDNSGLPFAWFVSYAKGQDG  1335\n",
      "             ++++ L  SV +  GTGT   + G  VG KTGTA+   +G+       A F+ +A  ++ \n",
      "Sbjct  429   REVRNLXVSVTEPGGTGTAGAVDGFDVGAKTGTARKFVNGRYADNKHVATFIGFAPAKN-  487\n",
      "\n",
      "Query  1336  KQVAVAVVVEDGSSEAQQISGGSLAAPIAKAMMQAAL  1446\n",
      "              +V VAV +++ +  A    GG +A P  K +   +L\n",
      "Sbjct  488   PRVIVAVTIDEPT--AHGYYGGVVAGPPFKKIXGGSL  522\n",
      "\n",
      "\n",
      ">4L0L_A Chain A, Crystal Structure Of P.aeruginosa Pbp3 In Complex With \n",
      "Compound 4\n",
      "Length=537\n",
      "\n",
      " Score = 78.2 bits (191),  Expect = 5e-15, Method: Compositional matrix adjust.\n",
      " Identities = 96/382 (25%), Positives = 170/382 (45%), Gaps = 64/382 (17%)\n",
      " Frame = +1\n",
      "\n",
      "Query  373   LTGKPKQGGDVVTTIDPKVQKAGFEGLGN-------KKGAAVAIDPKTGAILGLVSTPSY  531\n",
      "             +T   K G  +  +ID ++Q      L N       K G+ V +D KTG IL + + P+Y\n",
      "Sbjct  167   VTKNAKPGKTLALSIDLRLQYLAHRELRNALLENGAKAGSLVIMDVKTGEILAMTNQPTY  226\n",
      "\n",
      "Query  532   DPGTFAGGTKDDEKAWTALDSDPNKPMLNRALRETYPPGSTFKLVTAATAFETGKYQSPS  711\n",
      "             +P       ++ + A           M NRA+ + + PGST K  + + A  +G+++ PS\n",
      "Sbjct  227   NPNN----RRNLQPA----------AMRNRAMIDVFEPGSTVKPFSMSAALASGRWK-PS  271\n",
      "\n",
      "Query  712   DVTDTPDQYILPGTSTPLINASPTEDCGNATVQHAMDL------SCNTVFGKMGAELGGD  873\n",
      "             D+ D     + PGT    I      D    + Q  +DL      S N    K+  ++G +\n",
      "Sbjct  272   DIVD-----VYPGTLQ--IGRYTIRDVSRNSRQ--LDLTGILIKSSNVGISKIAFDIGAE  322\n",
      "\n",
      "Query  874   KLRAQAEKFGFNKEVTIPIRAD-ASHFPSGSK-PDG-TAMDAIGQHDTRATPLQMAMVAA  1044\n",
      "              + +  ++ G  ++  +    +   + P+  K P   TA  A G +    T +Q+A   A\n",
      "Sbjct  323   SIYSVMQQVGLGQDTGLGFPGERVGNLPNHRKWPKAETATLAYG-YGLSVTAIQLAHAYA  381\n",
      "\n",
      "Query  1045  AIANNGSLMEPYLVAQEGSAGNVISTHTEHQLSQAVSPATAQKLQQLMESVVQHGTGT-N  1221\n",
      "             A+AN+G  +   +   +     V          Q +SP  A  +Q +++ VV+   G   \n",
      "Sbjct  382   ALANDGKSVPLSMTRVDRVPDGV----------QVISPEVASTVQGMLQQVVEAQGGVFR  431\n",
      "\n",
      "Query  1222  AKIPGVTVGGKTGTAQH-------GQDNSGLPFAWFVSYAKGQDGKQVAVAVVVEDGSSE  1380\n",
      "             A++PG    GK+GTA+         ++N+    + F  +A   D  ++A+ VV+++ S  \n",
      "Sbjct  432   AQVPGYHAAGKSGTARKVSVGTKGYRENAYR--SLFAGFAPATD-PRIAMVVVIDEPSKA  488\n",
      "\n",
      "Query  1381  AQQISGGSLAAPIAKAMMQAAL  1446\n",
      "                  GG ++AP+   +M  AL\n",
      "Sbjct  489   G--YFGGLVSAPVFSKVMAGAL  508\n",
      "\n",
      "\n",
      ">3PBN_A Chain A, Crystal Structure Of Apo Pbp3 From Pseudomonas Aeruginosa\n",
      " 3PBO_A Chain A, Crystal Structure Of Pbp3 Complexed With Ceftazidime\n",
      " 3PBQ_A Chain A, Crystal Structure Of Pbp3 Complexed With Imipenem\n",
      " 3PBR_A Chain A, Crystal Structure Of Pbp3 Complexed With Meropenem\n",
      " 3PBS_A Chain A, Crystal Structure Of Pbp3 Complexed With Aztreonam\n",
      " 3PBT_A Chain A, Crystal Structure Of Pbp3 Complexed With Mc-1\n",
      " 4OOL_A Chain A, Crystal Structure Of Pbp3 In Complex With Compound 14 \n",
      "((2e)-2-({[(2s)- 2-{[(2z)-2-(2-amino-1,3-thiazol-4-yl)-2-{[(1,5-dihydroxy-4-oxo-1,4- \n",
      "Dihydropyridin-2-yl)methoxy]imino}acetyl]amino}-3- \n",
      "Oxopropyl]oxy}imino)pentanedioic Acid)\n",
      " 4OOM_A Chain A, Crystal Structure Of Pbp3 In Complex With Bal30072 ((2z)-2-(2-amino-1, \n",
      "3-thiazol-4-yl)-2-{[(1,5-dihydroxy-4-oxo-1,4-dihydropyridin-2-yl) \n",
      "Methoxy]imino}-n-{(2s)-1-hydroxy-3-methyl-3-[(sulfooxy)amino]butan-2- \n",
      "Yl}ethanamide)\n",
      " 4WEJ_A Chain A, Crystal Structure Of Pseudomonas Aeruginosa Pbp3 With \n",
      "A R4 Substituted Allyl Monocarbam\n",
      " 4WEL_A Chain A, Crystal Structure Of Pseudomonas Aeruginosa Pbp3 With \n",
      "Smc-3176\n",
      "Length=538\n",
      "\n",
      " Score = 77.8 bits (190),  Expect = 6e-15, Method: Compositional matrix adjust.\n",
      " Identities = 96/382 (25%), Positives = 170/382 (45%), Gaps = 64/382 (17%)\n",
      " Frame = +1\n",
      "\n",
      "Query  373   LTGKPKQGGDVVTTIDPKVQKAGFEGLGN-------KKGAAVAIDPKTGAILGLVSTPSY  531\n",
      "             +T   K G  +  +ID ++Q      L N       K G+ V +D KTG IL + + P+Y\n",
      "Sbjct  168   VTKNAKPGKTLALSIDLRLQYLAHRELRNALLENGAKAGSLVIMDVKTGEILAMTNQPTY  227\n",
      "\n",
      "Query  532   DPGTFAGGTKDDEKAWTALDSDPNKPMLNRALRETYPPGSTFKLVTAATAFETGKYQSPS  711\n",
      "             +P       ++ + A           M NRA+ + + PGST K  + + A  +G+++ PS\n",
      "Sbjct  228   NPNN----RRNLQPA----------AMRNRAMIDVFEPGSTVKPFSMSAALASGRWK-PS  272\n",
      "\n",
      "Query  712   DVTDTPDQYILPGTSTPLINASPTEDCGNATVQHAMDL------SCNTVFGKMGAELGGD  873\n",
      "             D+ D     + PGT    I      D    + Q  +DL      S N    K+  ++G +\n",
      "Sbjct  273   DIVD-----VYPGTLQ--IGRYTIRDVSRNSRQ--LDLTGILIKSSNVGISKIAFDIGAE  323\n",
      "\n",
      "Query  874   KLRAQAEKFGFNKEVTIPIRAD-ASHFPSGSK-PDG-TAMDAIGQHDTRATPLQMAMVAA  1044\n",
      "              + +  ++ G  ++  +    +   + P+  K P   TA  A G +    T +Q+A   A\n",
      "Sbjct  324   SIYSVMQQVGLGQDTGLGFPGERVGNLPNHRKWPKAETATLAYG-YGLSVTAIQLAHAYA  382\n",
      "\n",
      "Query  1045  AIANNGSLMEPYLVAQEGSAGNVISTHTEHQLSQAVSPATAQKLQQLMESVVQHGTGT-N  1221\n",
      "             A+AN+G  +   +   +     V          Q +SP  A  +Q +++ VV+   G   \n",
      "Sbjct  383   ALANDGKSVPLSMTRVDRVPDGV----------QVISPEVASTVQGMLQQVVEAQGGVFR  432\n",
      "\n",
      "Query  1222  AKIPGVTVGGKTGTAQH-------GQDNSGLPFAWFVSYAKGQDGKQVAVAVVVEDGSSE  1380\n",
      "             A++PG    GK+GTA+         ++N+    + F  +A   D  ++A+ VV+++ S  \n",
      "Sbjct  433   AQVPGYHAAGKSGTARKVSVGTKGYRENAYR--SLFAGFAPATD-PRIAMVVVIDEPSKA  489\n",
      "\n",
      "Query  1381  AQQISGGSLAAPIAKAMMQAAL  1446\n",
      "                  GG ++AP+   +M  AL\n",
      "Sbjct  490   G--YFGGLVSAPVFSKVMAGAL  509\n",
      "\n",
      "\n",
      ">4BJP_A Chain A, Crystal Structure Of E. Coli Penicillin Binding Protein \n",
      "3\n",
      "Length=511\n",
      "\n",
      " Score = 77.8 bits (190),  Expect = 7e-15, Method: Compositional matrix adjust.\n",
      " Identities = 102/440 (23%), Positives = 177/440 (40%), Gaps = 70/440 (16%)\n",
      " Frame = +1\n",
      "\n",
      "Query  211   LRYKYKRSWVDGPMYAPVTGYSSQTYDASQLEKLE---DGILSGTDSRLFFRN------T  363\n",
      "             LR + +R +  G + A + G+++   D+  +E +E   D  L+G       R        \n",
      "Sbjct  95    LREESRRYYPSGEVTAHLIGFTN--VDSQGIEGVEKSFDKWLTGQPGERIVRKDRYGRVI  152\n",
      "\n",
      "Query  364   LDMLTGKPKQGGDVVTTIDPKVQKAGFEGLGN-------KKGAAVAIDPKTGAILGLVST  522\n",
      "              D+ +   +   ++  +ID ++Q   +  L N       + G+AV +D  TG +L + ++\n",
      "Sbjct  153   EDISSTDSQAAHNLALSIDERLQALVYRELNNAVAFNKAESGSAVLVDVNTGEVLAMANS  212\n",
      "\n",
      "Query  523   PSYDPGTFAGGTKDDEKAWTALDSDPNKPMLNRALRETYPPGSTFKLVTAATAFETGKYQ  702\n",
      "             PSY+P   +G               P + M NR + + + PGST K +   TA + G  +\n",
      "Sbjct  213   PSYNPNNLSG--------------TPKEAMRNRTITDVFEPGSTVKPMVVMTALQRGVVR  258\n",
      "\n",
      "Query  703   SPSDVTDTPDQYILPGTSTPLINASPTEDCGN---ATVQHAMDLSCNTVFGKMGAELGGD  873\n",
      "               S +   P +          IN    +D       T+   +  S N    K+   +   \n",
      "Sbjct  259   ENSVLNTIPYR----------INGHEIKDVARYSELTLTGVLQKSSNVGVSKLALAMPSS  308\n",
      "\n",
      "Query  874   KLRAQAEKFGFNKEVTIPIRADASHFPSGSKPDGTAMDAIGQ------HDTRATPLQMAM  1035\n",
      "              L     +FG  K   + +  +     SG  P       I +      +    TPLQ+A \n",
      "Sbjct  309   ALVDTYSRFGLGKATNLGLVGER----SGLYPQKQRWSDIERATFSFGYGLMVTPLQLAR  364\n",
      "\n",
      "Query  1036  VAAAIANNGSLMEPYLVAQEGSAGNVISTHTEHQLSQAVSPATAQKLQQLMESV-VQHGT  1212\n",
      "             V A I + G +  P  + +            E    +++     + +  +MESV +  G \n",
      "Sbjct  365   VYATIGSYG-IYRPLSITKVDP-----PVPGERVFPESI----VRTVVHMMESVALPGGG  414\n",
      "\n",
      "Query  1213  GTNAKIPGVTVGGKTGTAQH-GQDNSGLPFAWFVSYAKG-QDGKQVAVAVVVEDGSSEAQ  1386\n",
      "             G  A I G  +  KTGTA+  G D  G     +++Y  G     Q   A+VV     +A \n",
      "Sbjct  415   GVKAAIKGYRIAIKTGTAKKVGPD--GRYINKYIAYTAGVAPASQPRFALVVVINDPQAG  472\n",
      "\n",
      "Query  1387  QISGGSLAAPIAKAMMQAAL  1446\n",
      "             +  GG+++AP+  A+M   L\n",
      "Sbjct  473   KYYGGAVSAPVFGAIMGGVL  492\n",
      "\n",
      "\n",
      ">4FSF_A Chain A, Crystal Structure Of Pseudomonas Aeruginosa Pbp3 Complexed \n",
      "With Compound 14\n",
      "Length=536\n",
      "\n",
      " Score = 77.8 bits (190),  Expect = 7e-15, Method: Compositional matrix adjust.\n",
      " Identities = 96/382 (25%), Positives = 170/382 (45%), Gaps = 64/382 (17%)\n",
      " Frame = +1\n",
      "\n",
      "Query  373   LTGKPKQGGDVVTTIDPKVQKAGFEGLGN-------KKGAAVAIDPKTGAILGLVSTPSY  531\n",
      "             +T   K G  +  +ID ++Q      L N       K G+ V +D KTG IL + + P+Y\n",
      "Sbjct  166   VTKNAKPGKTLALSIDLRLQYLAHRELRNALLENGAKAGSLVIMDVKTGEILAMTNQPTY  225\n",
      "\n",
      "Query  532   DPGTFAGGTKDDEKAWTALDSDPNKPMLNRALRETYPPGSTFKLVTAATAFETGKYQSPS  711\n",
      "             +P       ++ + A           M NRA+ + + PGST K  + + A  +G+++ PS\n",
      "Sbjct  226   NPNN----RRNLQPA----------AMRNRAMIDVFEPGSTVKPFSMSAALASGRWK-PS  270\n",
      "\n",
      "Query  712   DVTDTPDQYILPGTSTPLINASPTEDCGNATVQHAMDL------SCNTVFGKMGAELGGD  873\n",
      "             D+ D     + PGT    I      D    + Q  +DL      S N    K+  ++G +\n",
      "Sbjct  271   DIVD-----VYPGTLQ--IGRYTIRDVSRNSRQ--LDLTGILIKSSNVGISKIAFDIGAE  321\n",
      "\n",
      "Query  874   KLRAQAEKFGFNKEVTIPIRAD-ASHFPSGSK-PDG-TAMDAIGQHDTRATPLQMAMVAA  1044\n",
      "              + +  ++ G  ++  +    +   + P+  K P   TA  A G +    T +Q+A   A\n",
      "Sbjct  322   SIYSVMQQVGLGQDTGLGFPGERVGNLPNHRKWPKAETATLAYG-YGLSVTAIQLAHAYA  380\n",
      "\n",
      "Query  1045  AIANNGSLMEPYLVAQEGSAGNVISTHTEHQLSQAVSPATAQKLQQLMESVVQHGTGT-N  1221\n",
      "             A+AN+G  +   +   +     V          Q +SP  A  +Q +++ VV+   G   \n",
      "Sbjct  381   ALANDGKSVPLSMTRVDRVPDGV----------QVISPEVASTVQGMLQQVVEAQGGVFR  430\n",
      "\n",
      "Query  1222  AKIPGVTVGGKTGTAQH-------GQDNSGLPFAWFVSYAKGQDGKQVAVAVVVEDGSSE  1380\n",
      "             A++PG    GK+GTA+         ++N+    + F  +A   D  ++A+ VV+++ S  \n",
      "Sbjct  431   AQVPGYHAAGKSGTARKVSVGTKGYRENAYR--SLFAGFAPATD-PRIAMVVVIDEPSKA  487\n",
      "\n",
      "Query  1381  AQQISGGSLAAPIAKAMMQAAL  1446\n",
      "                  GG ++AP+   +M  AL\n",
      "Sbjct  488   G--YFGGLVSAPVFSKVMAGAL  507\n",
      "\n",
      "\n",
      ">4WEK_A Chain A, Crystal Structure Of Pseudomonas Aeruginosa Pbp3 With \n",
      "A R4 Substituted Vinyl Monocarbam\n",
      "Length=538\n",
      "\n",
      " Score = 77.8 bits (190),  Expect = 7e-15, Method: Compositional matrix adjust.\n",
      " Identities = 96/382 (25%), Positives = 170/382 (45%), Gaps = 64/382 (17%)\n",
      " Frame = +1\n",
      "\n",
      "Query  373   LTGKPKQGGDVVTTIDPKVQKAGFEGLGN-------KKGAAVAIDPKTGAILGLVSTPSY  531\n",
      "             +T   K G  +  +ID ++Q      L N       K G+ V +D KTG IL + + P+Y\n",
      "Sbjct  168   VTKNAKPGKTLALSIDLRLQYLAHRELRNALLENGAKAGSLVIMDVKTGEILAMTNQPTY  227\n",
      "\n",
      "Query  532   DPGTFAGGTKDDEKAWTALDSDPNKPMLNRALRETYPPGSTFKLVTAATAFETGKYQSPS  711\n",
      "             +P       ++ + A           M NRA+ + + PGST K  + + A  +G+++ PS\n",
      "Sbjct  228   NPNN----RRNLQPA----------AMRNRAMIDVFEPGSTVKPFSMSAALASGRWK-PS  272\n",
      "\n",
      "Query  712   DVTDTPDQYILPGTSTPLINASPTEDCGNATVQHAMDL------SCNTVFGKMGAELGGD  873\n",
      "             D+ D     + PGT    I      D    + Q  +DL      S N    K+  ++G +\n",
      "Sbjct  273   DIVD-----VYPGTLQ--IGRYTIRDVSRNSRQ--LDLTGILIKSSNVGISKIAFDIGAE  323\n",
      "\n",
      "Query  874   KLRAQAEKFGFNKEVTIPIRAD-ASHFPSGSK-PDG-TAMDAIGQHDTRATPLQMAMVAA  1044\n",
      "              + +  ++ G  ++  +    +   + P+  K P   TA  A G +    T +Q+A   A\n",
      "Sbjct  324   SIYSVMQQVGLGQDTGLGFPGERVGNLPNHRKWPKAETATLAYG-YGLSVTAIQLAHAYA  382\n",
      "\n",
      "Query  1045  AIANNGSLMEPYLVAQEGSAGNVISTHTEHQLSQAVSPATAQKLQQLMESVVQHGTGT-N  1221\n",
      "             A+AN+G  +   +   +     V          Q +SP  A  +Q +++ VV+   G   \n",
      "Sbjct  383   ALANDGKSVPLSMTRVDRVPDGV----------QVISPEVASTVQGMLQQVVEAQGGVFR  432\n",
      "\n",
      "Query  1222  AKIPGVTVGGKTGTAQH-------GQDNSGLPFAWFVSYAKGQDGKQVAVAVVVEDGSSE  1380\n",
      "             A++PG    GK+GTA+         ++N+    + F  +A   D  ++A+ VV+++ S  \n",
      "Sbjct  433   AQVPGYHAAGKSGTARKVSVGTKGYRENAYR--SLFAGFAPATD-PRIAMVVVIDEPSKA  489\n",
      "\n",
      "Query  1381  AQQISGGSLAAPIAKAMMQAAL  1446\n",
      "                  GG ++AP+   +M  AL\n",
      "Sbjct  490   G--YFGGLVSAPVFSKVMAGAL  509\n",
      "\n",
      "\n",
      ">3OC2_A Chain A, Crystal Structure Of Penicillin-binding Protein 3 From \n",
      "Pseudomonas Aeruginosa\n",
      " 3OCL_A Chain A, Crystal Structure Of Penicillin-binding Protein 3 From \n",
      "Pseudomonas Aeruginosa In Complex With Carbenicillin\n",
      " 3OCN_A Chain A, Crystal Structure Of Penicillin-binding Protein 3 From \n",
      "Pseudomonas Aeruginosa In Complex With Ceftazidime\n",
      " 4KQO_A Chain A, Crystal Structure Of Penicillin-binding Protein 3 From \n",
      "Pseudomonas Aeruginosa In Complex With Piperacillin\n",
      " 4KQO_B Chain B, Crystal Structure Of Penicillin-binding Protein 3 From \n",
      "Pseudomonas Aeruginosa In Complex With Piperacillin\n",
      " 4KQQ_A Chain A, Crystal Structure Of Penicillin-binding Protein 3 From \n",
      "Pseudomonas Aeruginosa In Complex With (5s)-penicilloic Acid\n",
      " 4KQQ_B Chain B, Crystal Structure Of Penicillin-binding Protein 3 From \n",
      "Pseudomonas Aeruginosa In Complex With (5s)-penicilloic Acid\n",
      " 4KQR_A Chain A, Crystal Structure Of Penicillin-binding Protein 3 From \n",
      "Pseudomonas Aeruginosa In Complex With (5s)-penicilloic Acid\n",
      " 4KQR_B Chain B, Crystal Structure Of Penicillin-binding Protein 3 From \n",
      "Pseudomonas Aeruginosa In Complex With (5s)-penicilloic Acid\n",
      " 5DF7_A Chain A, Crystal Structure Of Penicillin-binding Protein 3 From \n",
      "Pseudomonas Aeruginosa In Complex With Azlocillin\n",
      " 5DF7_B Chain B, Crystal Structure Of Penicillin-binding Protein 3 From \n",
      "Pseudomonas Aeruginosa In Complex With Azlocillin\n",
      " 5DF8_A Chain A, Crystal Structure Of Penicillin-binding Protein 3 From \n",
      "Pseudomonas Aeruginosa In Complex With Cefoperazone\n",
      " 5DF8_B Chain B, Crystal Structure Of Penicillin-binding Protein 3 From \n",
      "Pseudomonas Aeruginosa In Complex With Cefoperazone\n",
      " 5DF9_A Chain A, Crystal Structure Of Penicillin-binding Protein 3 In \n",
      "Complex With Deacylated Product Of Cefoperazone\n",
      "Length=564\n",
      "\n",
      " Score = 77.8 bits (190),  Expect = 8e-15, Method: Compositional matrix adjust.\n",
      " Identities = 96/382 (25%), Positives = 170/382 (45%), Gaps = 64/382 (17%)\n",
      " Frame = +1\n",
      "\n",
      "Query  373   LTGKPKQGGDVVTTIDPKVQKAGFEGLGN-------KKGAAVAIDPKTGAILGLVSTPSY  531\n",
      "             +T   K G  +  +ID ++Q      L N       K G+ V +D KTG IL + + P+Y\n",
      "Sbjct  194   VTKNAKPGKTLALSIDLRLQYLAHRELRNALLENGAKAGSLVIMDVKTGEILAMTNQPTY  253\n",
      "\n",
      "Query  532   DPGTFAGGTKDDEKAWTALDSDPNKPMLNRALRETYPPGSTFKLVTAATAFETGKYQSPS  711\n",
      "             +P       ++ + A           M NRA+ + + PGST K  + + A  +G+++ PS\n",
      "Sbjct  254   NPNN----RRNLQPA----------AMRNRAMIDVFEPGSTVKPFSMSAALASGRWK-PS  298\n",
      "\n",
      "Query  712   DVTDTPDQYILPGTSTPLINASPTEDCGNATVQHAMDL------SCNTVFGKMGAELGGD  873\n",
      "             D+ D     + PGT    I      D    + Q  +DL      S N    K+  ++G +\n",
      "Sbjct  299   DIVD-----VYPGTLQ--IGRYTIRDVSRNSRQ--LDLTGILIKSSNVGISKIAFDIGAE  349\n",
      "\n",
      "Query  874   KLRAQAEKFGFNKEVTIPIRAD-ASHFPSGSK-PDG-TAMDAIGQHDTRATPLQMAMVAA  1044\n",
      "              + +  ++ G  ++  +    +   + P+  K P   TA  A G +    T +Q+A   A\n",
      "Sbjct  350   SIYSVMQQVGLGQDTGLGFPGERVGNLPNHRKWPKAETATLAYG-YGLSVTAIQLAHAYA  408\n",
      "\n",
      "Query  1045  AIANNGSLMEPYLVAQEGSAGNVISTHTEHQLSQAVSPATAQKLQQLMESVVQHGTGT-N  1221\n",
      "             A+AN+G  +   +   +     V          Q +SP  A  +Q +++ VV+   G   \n",
      "Sbjct  409   ALANDGKSVPLSMTRVDRVPDGV----------QVISPEVASTVQGMLQQVVEAQGGVFR  458\n",
      "\n",
      "Query  1222  AKIPGVTVGGKTGTAQH-------GQDNSGLPFAWFVSYAKGQDGKQVAVAVVVEDGSSE  1380\n",
      "             A++PG    GK+GTA+         ++N+    + F  +A   D  ++A+ VV+++ S  \n",
      "Sbjct  459   AQVPGYHAAGKSGTARKVSVGTKGYRENAYR--SLFAGFAPATD-PRIAMVVVIDEPSKA  515\n",
      "\n",
      "Query  1381  AQQISGGSLAAPIAKAMMQAAL  1446\n",
      "                  GG ++AP+   +M  AL\n",
      "Sbjct  516   G--YFGGLVSAPVFSKVMAGAL  535\n",
      "\n",
      "\n",
      ">4ZTK_A Chain A, Transpeptidase Domain Of Ftsi4 D,d-transpeptidase From \n",
      "Legionella Pneumophila.\n",
      "Length=316\n",
      "\n",
      " Score = 64.3 bits (155),  Expect = 5e-11, Method: Compositional matrix adjust.\n",
      " Identities = 77/295 (26%), Positives = 126/295 (43%), Gaps = 53/295 (18%)\n",
      " Frame = +1\n",
      "\n",
      "Query  589   DSDPNKPMLNRALRETYPPGSTFKLVTAATAFETGKYQSPSDVTDTPDQYILPGTSTPLI  768\n",
      "             +S  NK + +   R  + PGST K   AA A  +G   S +   D    Y + G +    \n",
      "Sbjct  61    ESSKNKGLESWKSR-IFSPGSTIKPFIAAAAINSGN-SSETKNYDCHSPYYIAGKTFTNY  118\n",
      "\n",
      "Query  769   NASPTEDCGNATVQHAMDLSCNTVFGKMGAELGGDKLRAQAEKFGFNKEVTIPIRADASH  948\n",
      "             N+    +  +A++  A+  S N    ++  E G   +R +  +FGF          D + \n",
      "Sbjct  119   NS----NVESASLADAIAKSINVCLIRVSQEAGVPVIRKKLTEFGF----------DXNS  164\n",
      "\n",
      "Query  949   FPSGSKPDGT--AMDAIGQHDTRATPLQMAMVAAAIANNGSLMEPY-LVAQEGSA---GN  1110\n",
      "             +    + D    A  A+G++     P+ +           SL++ Y ++A +G +   GN\n",
      "Sbjct  165   WWQADQSDDLQLAXAALGEN----IPVTIE----------SLIKSYAILANKGHSFDRGN  210\n",
      "\n",
      "Query  1111  VISTHTEHQLSQAVSPATAQKLQQLMESVVQHGTGTNAKIPGVTVGGKTGTAQHGQDNSG  1290\n",
      "                       S  +S  +   +   +E+ V +GTG  A IPGV+V GKTGT     D   \n",
      "Sbjct  211   ----------SAIISETSTNSINHXLENAVTNGTGKLAVIPGVSVAGKTGTVIENNDKYL  260\n",
      "\n",
      "Query  1291  LPFAWFVSYAKGQDGKQVAVAVVVEDGSSEAQ---QISGGSLAAPIAKAMMQAAL  1446\n",
      "               FA +V      D  +  + VV+E+G         +SGG LAAP+ + +   AL\n",
      "Sbjct  261   ALFAGYVP----ADNPRYVLLVVIEEGYFSKNGKTLVSGGELAAPVFRNVAXDAL  311\n",
      "\n",
      "\n",
      ">4OVD_A Chain A, Crystal Structure Of A Putative Peptidoglycan Glycosyltransferase \n",
      "From Atopobium Parvulum Dsm 20469\n",
      "Length=564\n",
      "\n",
      " Score = 65.5 bits (158),  Expect = 6e-11, Method: Compositional matrix adjust.\n",
      " Identities = 102/396 (26%), Positives = 160/396 (40%), Gaps = 71/396 (18%)\n",
      " Frame = +1\n",
      "\n",
      "Query  199   FVNGLRYKYKRSWVDGPMYAPVTGY-SSQTYDASQLEKLEDGILSGTDSRLFFRNTLDML  375\n",
      "             FVN      KR +  G +   + G+ ++    AS LE   + IL+GT+            \n",
      "Sbjct  146   FVNNT----KRVYPYGNVGVQILGFVNADNEGASGLEYYYNDILAGTNGHXIVETGAG--  199\n",
      "\n",
      "Query  376   TGKPKQGG-----------DVVTTIDPKVQKAGFEGL--------GNKKGAAVAIDPKTG  498\n",
      "              G P  GG           D+V +ID ++QK   E L          + G+  A +P+TG\n",
      "Sbjct  200   -GTPIAGGTSNITEAQNGQDIVLSIDIELQKKAEEQLTAAVKDFEAKQGGSIXAXNPRTG  258\n",
      "\n",
      "Query  499   AILGLVSTPSYDPGTFAGGTKDDEKAWTALDSDPNKPMLN-RALRETYPPGSTFKLVTAA  675\n",
      "              I    S P  D  +               D+  +   LN + +   Y PGS FK++T +\n",
      "Sbjct  259   EIYAAASYPLPDFES---------------DNGVDYESLNLKLVSSIYEPGSVFKVITTS  303\n",
      "\n",
      "Query  676   TAFETGKYQSPSDVTDTPDQYILPGTSTPLINASPTEDCGNATVQHAMDLSCNTVFGKMG  855\n",
      "                +   +  P+   + P +  L      + +        + +V+  +  S N     + \n",
      "Sbjct  304   IGVDNNLF-GPNSTFNIPTE--LQVGDNKVTDDDWRTSAXDXSVREXLRRSSNVGXAFLE  360\n",
      "\n",
      "Query  856   AELGGDKLRAQA-EKFG--------FNKEVTIPIRA-DASHFPSGSKPDGTAMDAIGQHD  1005\n",
      "              +L GDK  A+  +KFG        F  E T  +R+ D    P+G         A GQ  \n",
      "Sbjct  361   TQLIGDKRFAEGIDKFGIGHTTGIDFPGEATGIVRSLDQYEGPTGGNX------AFGQG-  413\n",
      "\n",
      "Query  1006  TRATP-LQMAMVAAAIANNGSLMEPYLVAQEGSAGNVISTHTEHQLSQAVSPATAQKLQQ  1182\n",
      "               A P +Q+     A+AN G+ + P+    +G  G  +S  T+      +S +TA     \n",
      "Sbjct  414   -LAIPFIQVIRAYTAVANKGTXVTPHFXVSKG--GQEVSWPTK----DVISSSTASAELD  466\n",
      "\n",
      "Query  1183  LMESVVQHGTGTNAKIPGVTVGGKTGTAQHGQDNSG  1290\n",
      "                +VV+ GTG  A I G TV GKTGT Q   + +G\n",
      "Sbjct  467   XXTTVVKEGTGVRAGIYGYTVAGKTGTGQQVVEETG  502\n",
      "\n",
      "\n",
      ">4U3T_A Chain A, Crystal Structure Of The Transpeptidase Domain Of Neisseria \n",
      "Gonorrhoeae Penicillin-binding Protein 2 Derived From \n",
      "The Penicillin- Resistant Strain 6140\n",
      " 4U3T_B Chain B, Crystal Structure Of The Transpeptidase Domain Of Neisseria \n",
      "Gonorrhoeae Penicillin-binding Protein 2 Derived From \n",
      "The Penicillin- Resistant Strain 6140\n",
      "Length=330\n",
      "\n",
      " Score = 60.1 bits (144),  Expect = 2e-09, Method: Compositional matrix adjust.\n",
      " Identities = 91/374 (24%), Positives = 146/374 (39%), Gaps = 92/374 (25%)\n",
      " Frame = +1\n",
      "\n",
      "Query  412   TIDPKVQKAGFEGLGN-------KKGAAVAIDPKTGAILGLVSTPSYDPGTFAGGTKDDE  570\n",
      "             ++D ++Q   +E L         K G  V +D +TG IL L +TP               \n",
      "Sbjct  7     SLDQRIQTLAYEELNKAVEYHQAKAGTVVVLDARTGEILALANTPGR-------------  53\n",
      "\n",
      "Query  571   KAWTALDSDPNKPMLNRALRETYPPGSTFKLVTAATAFETGKYQSPSDVTDTPDQYILPG  750\n",
      "                            NRA+ +   PGS  K    A A + GK      +   P +     \n",
      "Sbjct  54    ---------------NRAVTDMIEPGSAIKPFVIAKALDAGKTDLNERLNTQPYK-----  93\n",
      "\n",
      "Query  751   TSTPLINASPTEDCGNAT----VQHAMDLSCNTVFGKMGAELGGDKLRAQAEKFGFNKEV  918\n",
      "                  I  SP  D  +      V+  M  S N    K+ A  G +      E + F  E+\n",
      "Sbjct  94    -----IGPSPVRDDTHVYPSLDVRGIMQKSSNVGTSKLSARFGAE------EMYDFYHEL  142\n",
      "\n",
      "Query  919   TIPIRADASHFPSGS----------KPDGTAMDAIGQHDTRATPLQMAMVAAAIANNGSL  1068\n",
      "              I +R   S FP  +          +P   A  + G +  + + LQ+A    A+ ++G L\n",
      "Sbjct  143   GIGVRMH-SGFPGETAGLLRNWRRWRPIEQATMSFG-YGLQLSLLQLARAYTALTHDGVL  200\n",
      "\n",
      "Query  1069  M----EPYLVAQEGSAGNVISTHTEHQLSQAVSPATAQKLQQLMESVVQ-HGTGTNAKIP  1233\n",
      "             +    E   VA +G               +    +TA++++ LM SV +  GTGT   + \n",
      "Sbjct  201   LPLSFEKQAVAPQGK--------------RIFKESTAREVRNLMVSVTEPGGTGTAGAVD  246\n",
      "\n",
      "Query  1234  GVTVGGKTGTAQ---HGQDNSGLPFAWFVSYAKGQDGKQVAVAVVVEDGSSEAQQISGGS  1404\n",
      "             G  VG KTGTA+   +G+         F+ +A  ++  +V VAV +++ +  A    GG \n",
      "Sbjct  247   GFDVGAKTGTARKLVNGRYVDNKHVGTFIGFAPAKN-PRVIVAVTIDEPT--AHGYYGGV  303\n",
      "\n",
      "Query  1405  LAAPIAKAMMQAAL  1446\n",
      "             +A    K +M  +L\n",
      "Sbjct  304   VAGSPFKKIMGGSL  317\n",
      "\n",
      "\n",
      ">3ZG8_B Chain B, Crystal Structure Of Penicillin Binding Protein 4 From \n",
      "Listeria Monocytogenes In The Ampicillin Bound Form\n",
      " 3ZGA_B Chain B, Crystal Structure Of Peniscillin-binding Protein 4 From \n",
      "Listeria Monocytogenes In The Carbenicillin Bound Form\n",
      " 3ZG9_B Chain B, Crystal Structure Of Penicillin-binding Protein 4 From \n",
      "Listeria Monocytogenes In The Cefuroxime Bound Form\n",
      "Length=537\n",
      "\n",
      " Score = 60.5 bits (145),  Expect = 2e-09, Method: Compositional matrix adjust.\n",
      " Identities = 79/285 (28%), Positives = 116/285 (41%), Gaps = 47/285 (16%)\n",
      " Frame = +1\n",
      "\n",
      "Query  472   AVAIDPKTGAILGLVSTPSYDPGTFAGGTKDDEKAWTALDSDPNKPMLNRALRETYPPGS  651\n",
      "             AV +DP TG I  LV         F G                     NRA +    PGS\n",
      "Sbjct  180   AVLMDPATGGIRALVG--GRGEHVFRG--------------------FNRATQMKAQPGS  217\n",
      "\n",
      "Query  652   TFK-LVTAATAFETGKYQSPSDVTDTPDQYILPGTSTPLINASPTEDCGNATVQHAMDLS  828\n",
      "             T K L     A ++G Y   S + D    Y   G  TP  N       G   +  A+  S\n",
      "Sbjct  218   TMKPLAVYTPALQSG-YDVDSMLKDEKITY--KGNYTP-TNVGGVY-SGEVPMYKAVANS  272\n",
      "\n",
      "Query  829   CNTVFGKMGAELGGDKLRAQAEKFGFNKEVTIPIRADASHFPSGSKPDGTAMDAIGQHDT  1008\n",
      "              N     +  ++G DK     EKFG    +T+P            + D T   A+G    \n",
      "Sbjct  273   INAPAVWLLDQIGIDKGVKSVEKFG----ITVP------------EKDRTLGLALGGMSK  316\n",
      "\n",
      "Query  1009  RATPLQMAMVAAAIANNGSLMEPYLVAQ-EGSAGNVISTHTEHQLSQAVSPATAQKLQQL  1185\n",
      "              A+P++MA   A  ANNG+  E +++ +    +GN +  +   +  Q +S   + ++  +\n",
      "Sbjct  317   GASPVEMATAYATFANNGAKPESHIITKIVDPSGNTVYENVP-KTKQIISETVSNEMTSM  375\n",
      "\n",
      "Query  1186  MESVVQHGTGTNAKIPGVTVGGKTGTAQ-HGQDNSGLPFAWFVSY  1317\n",
      "             +  V+  GTG +A + G  + GKTG+ Q    D SG    WFV Y\n",
      "Sbjct  376   LLDVINTGTGQSAAVSGHEMAGKTGSTQVPFDDTSGTKDQWFVGY  420\n",
      "\n",
      "\n",
      ">2WAF_A Chain A, Penicillin-binding Protein 2b (pbp-2b) From Streptococcus \n",
      "Pneumoniae (strain R6)\n",
      "Length=682\n",
      "\n",
      " Score = 58.2 bits (139),  Expect = 1e-08, Method: Compositional matrix adjust.\n",
      " Identities = 94/333 (28%), Positives = 140/333 (42%), Gaps = 64/333 (19%)\n",
      " Frame = +1\n",
      "\n",
      "Query  472   AVAIDPKTGAILGLVSTPSYDPGTFAGGTKDDEKAWTALDSDPNKPMLNRALRETYPPGS  651\n",
      "             AVA++PKTGA+L +             G K D K    L  D    + N      + PGS\n",
      "Sbjct  345   AVALNPKTGAVLSM------------SGIKHDLKTG-ELTPDSLGTVTN-----VFVPGS  386\n",
      "\n",
      "Query  652   TFKLVTAATAFETGKYQSPSDVTDTPDQYILPGTSTPLINASPTEDCGN--ATVQHAMDL  825\n",
      "               K  T ++ +E G       +TD   Q I+   S P IN+  T+  G+   T   A++ \n",
      "Sbjct  387   VVKAATISSGWENGVLSGNQTLTD---QSIVFQGSAP-INSWYTQAYGSFPITAVQALEY  442\n",
      "\n",
      "Query  826   SCNT-----VFGKMG-------------AELGGDKLRAQAEKFGFNKEVTIPIRADASHF  951\n",
      "             S NT       G MG              E   +KLR+   ++G      I +  +++ F\n",
      "Sbjct  443   SSNTYMVQTALGLMGQTYQPNMFVGTSNLESAMEKLRSTFGEYGLGTATGIDLPDESTGF  502\n",
      "\n",
      "Query  952   -PSGSKPDGTAMDAIGQHDTRATPLQMAMVAAAIANNGSLMEPYLVA------QEGSAGN  1110\n",
      "              P          +A GQ D   TP+Q+A   A IANNG  + P +V        +G  G+\n",
      "Sbjct  503   VPKEYSFANYITNAFGQFDNY-TPMQLAQYVATIANNGVRVAPRIVEGIYGNNDKGGLGD  561\n",
      "\n",
      "Query  1111  VISTHTEHQLSQA-VSPATAQKLQQLMESVVQHGTG--------TNAKIPGVTVGGKTGT  1263\n",
      "             +I      ++++  +S +    L Q    V  HGT         +N  +  V++ GKTGT\n",
      "Sbjct  562   LIQQLQPTEMNKVNISDSDMSILHQGFYQVA-HGTSGLTTGRAFSNGAL--VSISGKTGT  618\n",
      "\n",
      "Query  1264  AQ-HGQDNSGLPFAWFVSYAKGQDGKQVAVAVV  1359\n",
      "             A+ +  D         V+YA   D  Q+AVAVV\n",
      "Sbjct  619   AESYVADGQQATNTNAVAYAPS-DNPQIAVAVV  650\n",
      "\n",
      "\n",
      ">2V2F_F Chain F, Crystal Structure Of Pbp1a From Drug-resistant Strain \n",
      "5204 From Streptococcus Pneumoniae\n",
      "Length=390\n",
      "\n",
      " Score = 57.8 bits (138),  Expect = 1e-08, Method: Compositional matrix adjust.\n",
      " Identities = 71/251 (28%), Positives = 111/251 (44%), Gaps = 35/251 (14%)\n",
      " Frame = +1\n",
      "\n",
      "Query  616   NRALRETYPPGSTFKLVT-AATAFETGKYQSPSD-VTDTPDQYILPGTSTPLINASPTED  789\n",
      "             N+A+      GS  K +T  A A E G Y S +  V D P  Y  PGTSTP+ N      \n",
      "Sbjct  96    NQAVETNRDWGSAMKPITDYAPAIEYGVYDSTATMVNDIPYNY--PGTSTPVYNWDRAY-  152\n",
      "\n",
      "Query  790   CGNATVQHAMDLSCNTVFGKMGAELGGDKLRAQAEKFGFNKEVTIPIRADASHFPSGSKP  969\n",
      "              GN T+Q+A+  S N    +   ++G D+ +      G +        A +S+    +K \n",
      "Sbjct  153   FGNITLQYALQQSRNVTAVETLNKVGLDRAKTFLNGLGIDYPSMHYANAISSNTTESNKQ  212\n",
      "\n",
      "Query  970   DGTAMDAIGQHDTRATPLQMAMVAAAIANNGSLMEPYLVAQEGSAGNVISTHTEHQLS--  1143\n",
      "              G            A+  +MA   AA AN G   +P  + +      V S  ++ + S  \n",
      "Sbjct  213   YG------------ASSEKMAAAYAAFANGGIYHKPMYINKV-----VFSDGSKKEFSDV  255\n",
      "\n",
      "Query  1144  --QAVSPATAQKLQQLMESVVQHGTGTNAKIPGVTVGGKTGTAQHGQD-------NSGL-  1293\n",
      "               +A+   TA  + ++M++V+ +GTG  A +P +   GKTGT+ +  D       N+G  \n",
      "Sbjct  256   GTRAMKETTAYMMTEMMKTVLAYGTGRGAYLPWLAQAGKTGTSNYTDDEIEKHIKNTGYV  315\n",
      "\n",
      "Query  1294  -PFAWFVSYAK  1323\n",
      "              P   FV Y +\n",
      "Sbjct  316   APDEMFVGYTR  326\n",
      "\n",
      "\n",
      ">2C5W_B Chain B, Penicillin-Binding Protein 1a (Pbp-1a) Acyl-Enzyme Complex \n",
      "(Cefotaxime) From Streptococcus Pneumoniae\n",
      "Length=385\n",
      "\n",
      " Score = 55.8 bits (133),  Expect = 5e-08, Method: Compositional matrix adjust.\n",
      " Identities = 63/226 (28%), Positives = 105/226 (46%), Gaps = 26/226 (12%)\n",
      " Frame = +1\n",
      "\n",
      "Query  613   LNRALRETYPPGSTFKLVT-AATAFETGKYQSPSDVT-DTPDQYILPGTSTPLINASPTE  786\n",
      "             +N+A+      GST K +T  A A E G Y+S + +  D P  Y  PGT+TP+ N     \n",
      "Sbjct  93    INQAVETNRDWGSTMKPITDYAPALEYGVYESTATIVHDEPYNY--PGTNTPVYNWDRGY  150\n",
      "\n",
      "Query  787   DCGNATVQHAMDLSCNTVFGKMGAELGGDKLRAQAEKFGFNKEVTIPIRADASHFPSGSK  966\n",
      "               GN T+Q+A+  S N    +   ++G ++ +      G        I   + H+ +   \n",
      "Sbjct  151   -FGNITLQYALQQSRNVPAVETLNKVGLNRAKTFLNGLG--------IDYPSIHYSNAIS  201\n",
      "\n",
      "Query  967   PDGTAMDAIGQHDTRATPLQMAMVAAAIANNGSLMEPYLVAQEGSAGNVISTHTEHQLS-  1143\n",
      "              + T  D        A+  +MA   AA AN G+  +P  + +      V S  +E + S \n",
      "Sbjct  202   SNTTESDK----KYGASSEKMAAAYAAFANGGTYYKPMYIHKV-----VFSDGSEKEFSN  252\n",
      "\n",
      "Query  1144  ---QAVSPATAQKLQQLMESVVQHGTGTNAKIPGVTVGGKTGTAQH  1272\n",
      "                +A+   TA  +  +M++V+ +GTG NA +  +   GKTGT+ +\n",
      "Sbjct  253   VGTRAMKETTAYMMTDMMKTVLTYGTGRNAYLAWLPQAGKTGTSNY  298\n",
      "\n",
      "\n",
      ">2ZC5_B Chain B, Penicillin-Binding Protein 1a (Pbp 1a) Acyl-Enzyme Complex \n",
      "(Biapenem) From Streptococcus Pneumoniae\n",
      " 2ZC5_D Chain D, Penicillin-Binding Protein 1a (Pbp 1a) Acyl-Enzyme Complex \n",
      "(Biapenem) From Streptococcus Pneumoniae\n",
      " 2ZC6_B Chain B, Penicillin-Binding Protein 1a (Pbp 1a) Acyl-Enzyme Complex \n",
      "(Tebipenem) From Streptococcus Pneumoniae\n",
      " 2ZC6_D Chain D, Penicillin-Binding Protein 1a (Pbp 1a) Acyl-Enzyme Complex \n",
      "(Tebipenem) From Streptococcus Pneumoniae\n",
      "Length=390\n",
      "\n",
      " Score = 55.1 bits (131),  Expect = 9e-08, Method: Compositional matrix adjust.\n",
      " Identities = 63/226 (28%), Positives = 104/226 (46%), Gaps = 26/226 (12%)\n",
      " Frame = +1\n",
      "\n",
      "Query  613   LNRALRETYPPGSTFKLVT-AATAFETGKYQSPSDVT-DTPDQYILPGTSTPLINASPTE  786\n",
      "             +N+A+      GST K +T  A A E G Y S + +  D P  Y  PGT+TP+ N     \n",
      "Sbjct  95    INQAVETNRDWGSTMKPITDYAPALEYGVYDSTATIVHDEPYNY--PGTNTPVYNWDRGY  152\n",
      "\n",
      "Query  787   DCGNATVQHAMDLSCNTVFGKMGAELGGDKLRAQAEKFGFNKEVTIPIRADASHFPSGSK  966\n",
      "               GN T+Q+A+  S N    +   ++G ++ +      G        I   + H+ +   \n",
      "Sbjct  153   -FGNITLQYALQQSRNVPAVETLNKVGLNRAKTFLNGLG--------IDYPSIHYSNAIS  203\n",
      "\n",
      "Query  967   PDGTAMDAIGQHDTRATPLQMAMVAAAIANNGSLMEPYLVAQEGSAGNVISTHTEHQLS-  1143\n",
      "              + T  D        A+  +MA   AA AN G+  +P  + +      V S  +E + S \n",
      "Sbjct  204   SNTTESDK----KYGASSEKMAAAYAAFANGGTYYKPMYIHKV-----VFSDGSEKEFSN  254\n",
      "\n",
      "Query  1144  ---QAVSPATAQKLQQLMESVVQHGTGTNAKIPGVTVGGKTGTAQH  1272\n",
      "                +A+   TA  +  +M++V+ +GTG NA +  +   GKTGT+ +\n",
      "Sbjct  255   VGTRAMKETTAYMMTDMMKTVLSYGTGQNAYLAWLPQAGKTGTSNY  300\n",
      "\n",
      "\n",
      ">2C6W_B Chain B, Penicillin-Binding Protein 1a (Pbp-1a) From Streptococcus \n",
      "Pneumoniae\n",
      "Length=384\n",
      "\n",
      " Score = 54.7 bits (130),  Expect = 1e-07, Method: Compositional matrix adjust.\n",
      " Identities = 63/226 (28%), Positives = 104/226 (46%), Gaps = 26/226 (12%)\n",
      " Frame = +1\n",
      "\n",
      "Query  613   LNRALRETYPPGSTFKLVT-AATAFETGKYQSPSDVT-DTPDQYILPGTSTPLINASPTE  786\n",
      "             +N+A+      GST K +T  A A E G Y S + +  D P  Y  PGT+TP+ N     \n",
      "Sbjct  92    INQAVETNRDWGSTMKPITDYAPALEYGVYDSTATIVHDEPYNY--PGTNTPVYNWDRGY  149\n",
      "\n",
      "Query  787   DCGNATVQHAMDLSCNTVFGKMGAELGGDKLRAQAEKFGFNKEVTIPIRADASHFPSGSK  966\n",
      "               GN T+Q+A+  S N    +   ++G ++ +      G        I   + H+ +   \n",
      "Sbjct  150   -FGNITLQYALQQSRNVPAVETLNKVGLNRAKTFLNGLG--------IDYPSIHYSNAIS  200\n",
      "\n",
      "Query  967   PDGTAMDAIGQHDTRATPLQMAMVAAAIANNGSLMEPYLVAQEGSAGNVISTHTEHQLS-  1143\n",
      "              + T  D        A+  +MA   AA AN G+  +P  + +      V S  +E + S \n",
      "Sbjct  201   SNTTESDK----KYGASSEKMAAAYAAFANGGTYYKPMYIHKV-----VFSDGSEKEFSN  251\n",
      "\n",
      "Query  1144  ---QAVSPATAQKLQQLMESVVQHGTGTNAKIPGVTVGGKTGTAQH  1272\n",
      "                +A+   TA  +  +M++V+ +GTG NA +  +   GKTGT+ +\n",
      "Sbjct  252   VGTRAMKETTAYMMTDMMKTVLTYGTGRNAYLAWLPQAGKTGTSNY  297\n",
      "\n",
      "\n",
      ">3ZG7_B Chain B, Crystal Structure Of Penicillin-binding Protein 4 From \n",
      "Listeria Monocytogenes In The Apo Form\n",
      "Length=537\n",
      "\n",
      " Score = 53.9 bits (128),  Expect = 3e-07, Method: Compositional matrix adjust.\n",
      " Identities = 67/238 (28%), Positives = 99/238 (42%), Gaps = 25/238 (11%)\n",
      " Frame = +1\n",
      "\n",
      "Query  613   LNRALRETYPPGSTFK-LVTAATAFETGKYQSPSDVTDTPDQYILPGTSTPLINASPTED  789\n",
      "              NRA +    PGST K L     A ++G Y   S + D    Y   G  TP  N      \n",
      "Sbjct  205   FNRATQXKAQPGSTXKPLAVYTPALQSG-YDVDSXLKDEKITY--KGNYTP-TNVGGVY-  259\n",
      "\n",
      "Query  790   CGNATVQHAMDLSCNTVFGKMGAELGGDKLRAQAEKFGFNKEVTIPIRADASHFPSGSKP  969\n",
      "              G      A+  S N     +  ++G DK     EKFG    +T+P            + \n",
      "Sbjct  260   SGEVPXYKAVANSINAPAVWLLDQIGIDKGVKSVEKFG----ITVP------------EK  303\n",
      "\n",
      "Query  970   DGTAMDAIGQHDTRATPLQMAMVAAAIANNGSLMEPYLVAQ-EGSAGNVISTHTEHQLSQ  1146\n",
      "             D T   A+G     A+P++ A   A  ANNG+  E +++ +    +GN +  +   +  Q\n",
      "Sbjct  304   DRTLGLALGGXSKGASPVEXATAYATFANNGAKPESHIITKIVDPSGNTVYENVP-KTKQ  362\n",
      "\n",
      "Query  1147  AVSPATAQKLQQLMESVVQHGTGTNAKIPGVTVGGKTGTAQ-HGQDNSGLPFAWFVSY  1317\n",
      "              +S   + +    +  V+  GTG +A + G    GKTG+ Q    D SG    WFV Y\n",
      "Sbjct  363   IISETVSNEXTSXLLDVINTGTGQSAAVSGHEXAGKTGSTQVPFDDTSGTKDQWFVGY  420\n",
      "\n",
      "\n",
      ">2WAD_A Chain A, Penicillin-binding Protein 2b (pbp-2b) From Streptococcus \n",
      "Pneumoniae (strain 5204)\n",
      " 2WAD_B Chain B, Penicillin-binding Protein 2b (pbp-2b) From Streptococcus \n",
      "Pneumoniae (strain 5204)\n",
      " 2WAD_C Chain C, Penicillin-binding Protein 2b (pbp-2b) From Streptococcus \n",
      "Pneumoniae (strain 5204)\n",
      "Length=680\n",
      "\n",
      " Score = 53.5 bits (127),  Expect = 4e-07, Method: Compositional matrix adjust.\n",
      " Identities = 88/330 (27%), Positives = 133/330 (40%), Gaps = 58/330 (18%)\n",
      " Frame = +1\n",
      "\n",
      "Query  472   AVAIDPKTGAILGLVSTPSYDPGTFAGGTKDDEKAWTALDSDPNKPMLNRALRETYPPGS  651\n",
      "             AVA++P+TGA+L +             G K D K    L  D    + N      + PGS\n",
      "Sbjct  345   AVALNPQTGAVLSM------------SGLKHDLKTG-ELTPDSLGTVTN-----VFVPGS  386\n",
      "\n",
      "Query  652   TFKLVTAATAFETGKYQSPSDVTDTP----------DQYILPGTSTPLINASPTEDCGNA  801\n",
      "               K  T ++ +E G       +TD P            Y L   S P+      E   NA\n",
      "Sbjct  387   VVKAATISSGWENGVLSGNQTLTDQPIVFQGSAPIYSWYKLAYGSFPITAVEALEYSSNA  446\n",
      "\n",
      "Query  802   -TVQHAMDL-----SCNTVFGKMGAELGGDKLRAQAEKFGFNKEVTIPIRADASHF-PSG  960\n",
      "               VQ A+ +       N   G    E    KLR+   ++G      I +  +++   P  \n",
      "Sbjct  447   YVVQTALGIMGQTYQPNMFVGTSNLESAMGKLRSTFGEYGLGSATGIDLPDESTGLVPKE  506\n",
      "\n",
      "Query  961   SKPDGTAMDAIGQHDTRATPLQMAMVAAAIANNGSLMEPYLVA------QEGSAGNVIST  1122\n",
      "                     +A GQ D   TP+Q+A   A IANNG  + P++V        +G  G +I  \n",
      "Sbjct  507   YNFANFITNAFGQFDNY-TPMQLAQYVATIANNGVRLAPHIVEGIYDNNDKGGLGELIQA  565\n",
      "\n",
      "Query  1123  HTEHQLSQA-VSPATAQKLQQLMESVVQHGTG--TNAKI----PGVTVGGKTGTAQH---  1272\n",
      "                 ++++  +S +    L Q     V HGT   T  +       V++ GKTGT +    \n",
      "Sbjct  566   IDTKEINKVNISESDMAILHQGFYQ-VSHGTSPLTTGRAFSDGATVSISGKTGTGESYVA  624\n",
      "\n",
      "Query  1273  -GQDNSGLPFAWFVSYAKGQDGKQVAVAVV  1359\n",
      "              GQ+ +       V+YA  ++  Q+AVAVV\n",
      "Sbjct  625   GGQEATN---TNAVAYAPTEN-PQIAVAVV  650\n",
      "\n",
      "\n",
      ">2WAE_A Chain A, Penicillin-binding Protein 2b (pbp-2b) From Streptococcus \n",
      "Pneumoniae (strain 5204)\n",
      "Length=680\n",
      "\n",
      " Score = 53.1 bits (126),  Expect = 5e-07, Method: Compositional matrix adjust.\n",
      " Identities = 88/330 (27%), Positives = 133/330 (40%), Gaps = 58/330 (18%)\n",
      " Frame = +1\n",
      "\n",
      "Query  472   AVAIDPKTGAILGLVSTPSYDPGTFAGGTKDDEKAWTALDSDPNKPMLNRALRETYPPGS  651\n",
      "             AVA++P+TGA+L +             G K D K    L  D    + N      + PGS\n",
      "Sbjct  345   AVALNPQTGAVLSM------------SGLKHDLKTG-ELTPDSLGTVTN-----VFVPGS  386\n",
      "\n",
      "Query  652   TFKLVTAATAFETGKYQSPSDVTDTP----------DQYILPGTSTPLINASPTEDCGNA  801\n",
      "               K  T ++ +E G       +TD P            Y L   S P+      E   NA\n",
      "Sbjct  387   VVKAATISSGWENGVLSGNQTLTDQPIVFQGSAPIYSWYKLAYGSFPITAVEALEYSSNA  446\n",
      "\n",
      "Query  802   -TVQHAMDL-----SCNTVFGKMGAELGGDKLRAQAEKFGFNKEVTIPIRADASHF-PSG  960\n",
      "               VQ A+ +       N   G    E    KLR+   ++G      I +  +++   P  \n",
      "Sbjct  447   YVVQTALGIMGQTYQPNMFVGTSNLESAMGKLRSTFGEYGLGSATGIDLPDESTGLVPKE  506\n",
      "\n",
      "Query  961   SKPDGTAMDAIGQHDTRATPLQMAMVAAAIANNGSLMEPYLVA------QEGSAGNVIST  1122\n",
      "                     +A GQ D   TP+Q+A   A IANNG  + P++V        +G  G +I  \n",
      "Sbjct  507   YNFANFITNAFGQFDNY-TPMQLAQYVATIANNGVRLAPHIVEGIYDNNDKGGLGELIQA  565\n",
      "\n",
      "Query  1123  HTEHQLSQA-VSPATAQKLQQLMESVVQHGTG--TNAKI----PGVTVGGKTGTAQH---  1272\n",
      "                 ++++  +S +    L Q     V HGT   T  +       V++ GKTGT +    \n",
      "Sbjct  566   IDTKEINKVNISESDMAILHQGFYQ-VSHGTSPLTTGRAFSDGATVSISGKTGTGESYVA  624\n",
      "\n",
      "Query  1273  -GQDNSGLPFAWFVSYAKGQDGKQVAVAVV  1359\n",
      "              GQ+ +       V+YA  ++  Q+AVAVV\n",
      "Sbjct  625   GGQEANN---TNAVAYAPTEN-PQIAVAVV  650\n",
      "\n",
      "\n",
      ">3UDF_A Chain A, Crystal Structure Of Apo Pbp1a From Acinetobacter Baumannii\n",
      " 3UDF_B Chain B, Crystal Structure Of Apo Pbp1a From Acinetobacter Baumannii\n",
      " 3UDI_A Chain A, Crystal Structure Of Acinetobacter Baumannii Pbp1a In \n",
      "Complex With Penicillin G\n",
      " 3UDI_B Chain B, Crystal Structure Of Acinetobacter Baumannii Pbp1a In \n",
      "Complex With Penicillin G\n",
      " 3UDX_A Chain A, Crystal Structure Of Acinetobacter Baumannii Pbp1a In \n",
      "Complex With Imipenem\n",
      " 3UDX_B Chain B, Crystal Structure Of Acinetobacter Baumannii Pbp1a In \n",
      "Complex With Imipenem\n",
      " 3UE0_A Chain A, Crystal Structure Of Acinetobacter Baumannii Pbp1a In \n",
      "Complex With Aztreonam\n",
      " 3UE0_B Chain B, Crystal Structure Of Acinetobacter Baumannii Pbp1a In \n",
      "Complex With Aztreonam\n",
      " 3UE1_A Chain A, Crystal Strucuture Of Acinetobacter Baumanni Pbp1a In \n",
      "Complex With Mc- 1\n",
      " 3UE1_B Chain B, Crystal Strucuture Of Acinetobacter Baumanni Pbp1a In \n",
      "Complex With Mc- 1\n",
      "Length=731\n",
      "\n",
      " Score = 50.8 bits (120),  Expect = 3e-06, Method: Compositional matrix adjust.\n",
      " Identities = 103/421 (24%), Positives = 153/421 (36%), Gaps = 130/421 (31%)\n",
      " Frame = +1\n",
      "\n",
      "Query  367   DMLTGKPKQGGDVVTTID-PKVQKAGFEGLGNKKGAAVAIDPKTGAILGLVSTPSYDPGT  543\n",
      "             D++  +P +     + +  PKVQ           G  +AI+P  G+I  +V   ++    \n",
      "Sbjct  365   DIVRLRPNEAKTAWSLVQVPKVQ-----------GQLIAINPNDGSIEAIVGGYNFYQSK  413\n",
      "\n",
      "Query  544   FAGGTKDDEKAWTALDSDPNKPMLNRALRETYPPGSTFKLVTAATAFETGKYQSP-SDVT  720\n",
      "             F                       NRAL+    PGST K    A A E G   +P S V \n",
      "Sbjct  414   F-----------------------NRALQGWRQPGSTIKPFLYALALERG--MTPYSMVN  448\n",
      "\n",
      "Query  721   DTPDQYILPGTSTPLINASPTEDCGNATVQHAMDLSCNTVFGKMGAELGGDKLRAQAEKF  900\n",
      "             D+P   I  G  TP    S     G   ++ A+ LS NTV  ++   +G ++ R     F\n",
      "Sbjct  449   DSP---ITIGKWTP--KNSDGRYLGMIPLRRALYLSRNTVSVRLLQTVGIERTRQLFMDF  503\n",
      "\n",
      "Query  901   GFNKEVTIPIRADASHFPSGSKPDGTAMDAIGQHDTRATPLQMAMVAAAIANNGSLMEPY  1080\n",
      "             G  +E  IP                     I     +  P+QMA   A  AN G  ++P+\n",
      "Sbjct  504   GL-QEDQIPRNY-----------------TIALGTPQVLPIQMATGYATFANGGYRVQPH  545\n",
      "\n",
      "Query  1081  LVAQ-EGSAGNVIS-----------------------THTEHQLSQAVSPATAQK-----  1173\n",
      "              + + E + G VI                        T  + Q+ +  +    QK     \n",
      "Sbjct  546   FIQRIEDAYGKVIYEAKPEYACIPCINAPETTDDAQVTTPDDQVVEVTNKELEQKEKTTK  605\n",
      "\n",
      "Query  1174  ----------------LQQLMES------------VVQHGTGTNA-KIPGVTVGGKTGTA  1266\n",
      "                              Q++++S            V++HGTG  A KI    +GGKTGT \n",
      "Sbjct  606   QLNLKQTDKNNSQYRQAQRILKSSSAYDMANILRDVIEHGTGRAALKIGRSDLGGKTGTT  665\n",
      "\n",
      "Query  1267  QHGQDNSGLPFAWFVSYAKGQDGKQVAVAVV-VEDGSSEAQQISGGSLAAPIAKAMMQAA  1443\n",
      "                +D      AWF     G +GK V V  V  +  ++  ++  GG  A PI    M  A\n",
      "Sbjct  666   NDAKD------AWFA----GFNGKLVTVTWVGFDQPTTLGRREYGGIAALPIWINFMGQA  715\n",
      "\n",
      "Query  1444  L  1446\n",
      "             L\n",
      "Sbjct  716   L  716\n",
      "\n",
      "\n",
      ">1K25_A Chain A, Pbp2x From A Highly Penicillin-Resistant Streptococcus \n",
      "Pneumoniae Clinical Isolate\n",
      " 1K25_B Chain B, Pbp2x From A Highly Penicillin-Resistant Streptococcus \n",
      "Pneumoniae Clinical Isolate\n",
      " 1K25_C Chain C, Pbp2x From A Highly Penicillin-Resistant Streptococcus \n",
      "Pneumoniae Clinical Isolate\n",
      " 1K25_D Chain D, Pbp2x From A Highly Penicillin-Resistant Streptococcus \n",
      "Pneumoniae Clinical Isolate\n",
      "Length=685\n",
      "\n",
      " Score = 49.7 bits (117),  Expect = 5e-06, Method: Compositional matrix adjust.\n",
      " Identities = 88/379 (23%), Positives = 141/379 (37%), Gaps = 83/379 (22%)\n",
      " Frame = +1\n",
      "\n",
      "Query  295   SQLEKLEDGILSGTDSRLFFRNT--------LDMLTGKPKQGGDVVTTIDPKVQK-----  435\n",
      "             S +E   + IL+GTD  + +            ++++ +   G DV TT+   +Q      \n",
      "Sbjct  152   SGMESSLNSILAGTDGIITYEKDRVGNIVPGTELVSQQTVDGKDVYTTLSSPLQSFMETQ  211\n",
      "\n",
      "Query  436   --AGFEGLGNKKGAAVAIDPKTGAILGLVSTPSYDPGTFAGGTKDDEKAWTALDSDPNKP  609\n",
      "               A  E +  K   A  +  KTG IL     P+++  T  G T+D    W  +    N  \n",
      "Sbjct  212   MDAFLEKVKGKYMTATLVSAKTGEILATTQRPTFNADTKEGITED--FVWRDILYQSN--  267\n",
      "\n",
      "Query  610   MLNRALRETYPPGSTFKLVTAATA-----FETGKYQSPSDV----TDTPDQYILPGTSTP  762\n",
      "                      Y PGS  K++T A++     F +G+Y + S++      T D  +  G +T \n",
      "Sbjct  268   ---------YEPGSAMKVMTLASSIDNNTFPSGEYFNSSELKIADATTRDWDVNEGLTT-  317\n",
      "\n",
      "Query  763   LINASPTEDCGNATVQHAMDLSCNTVFGKMGAELGGDKLRAQAEKFGFNKEVTIPIRADA  942\n",
      "                       G  T       S N     +  ++G        ++F F     +P R   \n",
      "Sbjct  318   ---------GGMMTFLQGFAHSSNVGMSLLEQKMGDATWLDYLKRFKFG----VPTRFGL  364\n",
      "\n",
      "Query  943   SHFPSGSKPD----GTAMDAIGQHDTRATPLQMAMVAAAIANNGSLMEPYLV--------  1086\n",
      "             +   +G  P       A  + GQ     T  QM     AIAN+G ++EP  +        \n",
      "Sbjct  365   TDEYAGQLPADNIVSIAQSSFGQ-GISVTQTQMLRAFTAIANDGVMLEPKFISAIYDTNN  423\n",
      "\n",
      "Query  1087  -----AQEGSAGNVI-----STHTEHQLSQAVSPATAQKLQQLMESVVQHGTGT-NAKIP  1233\n",
      "                  +Q+   GN +     ST   H +     P        L  ++  H TG     +P\n",
      "Sbjct  424   QSVRKSQKEIVGNPVSKEAASTTRNHMILVGTDP--------LYGTMYNHYTGKPIITVP  475\n",
      "\n",
      "Query  1234  GVTVGGKTGTAQHGQDNSG  1290\n",
      "             G  V  K+GTAQ   + +G\n",
      "Sbjct  476   GQNVAVKSGTAQIADEKNG  494\n",
      "\n",
      "\n",
      ">5U47_A Chain A, 1.95 Angstrom Resolution Crystal Structure Of Penicillin \n",
      "Binding Protein 2x From Streptococcus Thermophilus\n",
      "Length=707\n",
      "\n",
      " Score = 48.9 bits (115),  Expect = 1e-05, Method: Compositional matrix adjust.\n",
      " Identities = 91/364 (25%), Positives = 141/364 (39%), Gaps = 56/364 (15%)\n",
      " Frame = +1\n",
      "\n",
      "Query  301   LEKLEDGILSGTDSRLFFR-----NTL---DMLTGKPKQGGDVVTTIDPKVQK-------  435\n",
      "             LE+  + +L+GTD    +      NTL   + +T +   G ++ TT+   +Q        \n",
      "Sbjct  176   LERSLNSLLTGTDGEAIYEKDKDGNTLLGTETITKEAIDGKNIYTTLSAPLQTFLETQMD  235\n",
      "\n",
      "Query  436   AGFEGLGNKKGAAVAIDPKTGAILGLVSTPSYDPGTFAGGTKDDEKAWTALDSDPNKPML  615\n",
      "                E       +A  ++ KTG IL     P+Y+  T  G  K     W           +\n",
      "Sbjct  236   TFMEQTKGINASATVVNAKTGEILATTQRPTYNSDTLEGQAKKGYD-W-----------V  283\n",
      "\n",
      "Query  616   NRALRETYPPGSTFKLVTAATAFETGKYQSPSDVTDTPDQYILPGTSTPLINASPTEDCG  795\n",
      "             NR     Y PGST K++  + A   G +    + T +    I  G    + + S  E   \n",
      "Sbjct  284   NRLYEAQYEPGSTMKVMLLSAAINNGSFN--PNATYSNANGIKVG-DVEINDWSINEGIS  340\n",
      "\n",
      "Query  796   NA-TVQHAMDLSCNTVFG-KMGAELGGDKLRAQ---AEKFGFNKEVTIPIRADASHFPSG  960\n",
      "                T+  A   S ++  G  M  +  GDK+ +      KFG      IP R       SG\n",
      "Sbjct  341   KGRTMSFAQGFSYSSNVGMTMLEQAMGDKVWSNYLSLYKFG------IPTRFGMVGESSG  394\n",
      "\n",
      "Query  961   SKPDGT---AMDAIGQHDTRATPLQMAMVAAAIANNGSLMEPYLVAQEGSAGN-VISTHT  1128\n",
      "                  +   A  + GQ     T +QM     AI+NNG ++EP  + Q        + T  \n",
      "Sbjct  395   IVSQNSVNIAQSSFGQG-ISVTQVQMLRAFTAISNNGIMLEPQFIKQVADTNKGTVRTAK  453\n",
      "\n",
      "Query  1129  EHQLSQAVSPATAQKLQQLMESV---VQHGTGTNAK--IPGVTVGG-----KTGTAQHGQ  1278\n",
      "             +  + + VS   A + +  M SV    + GT  N     P + VG      K+GTAQ   \n",
      "Sbjct  454   KEVIGKPVSKQAASETRNYMISVGTDPEFGTLYNKSEGSPIIQVGNNDVTVKSGTAQVPD  513\n",
      "\n",
      "Query  1279  DNSG  1290\n",
      "             + +G\n",
      "Sbjct  514   EKTG  517\n",
      "\n",
      "\n",
      ">1RP5_A Chain A, Pbp2x From Streptococcus Pneumoniae Strain 5259 With \n",
      "Reduced Susceptibility To Beta-Lactam Antibiotics\n",
      " 1RP5_B Chain B, Pbp2x From Streptococcus Pneumoniae Strain 5259 With \n",
      "Reduced Susceptibility To Beta-Lactam Antibiotics\n",
      "Length=702\n",
      "\n",
      " Score = 47.4 bits (111),  Expect = 3e-05, Method: Compositional matrix adjust.\n",
      " Identities = 89/374 (24%), Positives = 138/374 (37%), Gaps = 73/374 (20%)\n",
      " Frame = +1\n",
      "\n",
      "Query  295   SQLEKLEDGILSGTDSRLFFRNT--------LDMLTGKPKQGGDVVTTIDPKVQK-----  435\n",
      "             S +E   + IL+GTD  + +            + ++ +   G DV TTI   +Q      \n",
      "Sbjct  169   SGMESSLNSILAGTDGIITYEKDRLGNIVPGTEQVSQQTVDGKDVYTTISSPLQSFMETQ  228\n",
      "\n",
      "Query  436   --AGFEGLGNKKGAAVAIDPKTGAILGLVSTPSYDPGTFAGGTKDDEKAWTALDSDPNKP  609\n",
      "               A  E +  K   A  +  KTG IL     P++D  T  G T+D    W  +    N  \n",
      "Sbjct  229   MDAFQEKVKGKYMTATLVSAKTGEILATTQRPTFDADTKEGITED--FVWRDILYQSN--  284\n",
      "\n",
      "Query  610   MLNRALRETYPPGSTFKLVTAATAFETGKYQSPSDVTDTPDQYILPGTSTPL-INASPTE  786\n",
      "                      Y PGST K++  A A +   +    +V ++ +  I   T     +N   T \n",
      "Sbjct  285   ---------YEPGSTMKVMMLAAAIDNNTFPG-GEVFNSSELKIADATIRDWDVNEGLTG  334\n",
      "\n",
      "Query  787   DCGNATVQ---HAMDLSCNTVFGKMGAELGGDKLRAQAEKFGFNKEVTIPIRADASHFPS  957\n",
      "                    Q   H+ ++    +  KMG     D L     +F F     +P R   +   +\n",
      "Sbjct  335   GRMMTFSQGFAHSSNVGMTLLEQKMGDATWLDYLN----RFKFG----VPTRFGLTDEYA  386\n",
      "\n",
      "Query  958   GSKPD----GTAMDAIGQHDTRATPLQMAMVAAAIANNGSLMEPYLV-------------  1086\n",
      "             G  P       AM A GQ     T  QM     AIAN+G ++EP  +             \n",
      "Sbjct  387   GQLPADNIVNIAMSAFGQ-GISVTQTQMLRAFTAIANDGVMLEPKFISALYDPNDQSVRK  445\n",
      "\n",
      "Query  1087  AQEGSAGNVISTHT-----EHQLSQAVSPATAQKLQQLMESVVQHGTG-TNAKIPGVTVG  1248\n",
      "             +Q+   GN +S        +H +     P           ++  H TG     +PG  V \n",
      "Sbjct  446   SQKEIVGNPVSKEAASVTRDHMVMVGTDPTYG--------TMYNHSTGKATVNVPGQNVA  497\n",
      "\n",
      "Query  1249  GKTGTAQHGQDNSG  1290\n",
      "              K+GTA+   + +G\n",
      "Sbjct  498   LKSGTAEIADEKNG  511\n",
      "\n",
      "\n",
      ">4OON_A Chain A, Crystal Structure Of Pbp1a In Complex With Compound \n",
      "17 ((4z,8s,11e, 14s)-5-(2-amino-1,3-thiazol-4-yl)-14-(5,6-dihydroxy-1,3-dioxo-1,3- \n",
      "Dihydro-2h-isoindol-2-yl)-8-formyl-2-methyl-6-oxo-3,10-dioxa-4,7,11- \n",
      "Triazatetradeca-4,11-diene-2,12,14-tricarboxylic \n",
      "Acid)\n",
      "Length=795\n",
      "\n",
      " Score = 46.6 bits (109),  Expect = 6e-05, Method: Compositional matrix adjust.\n",
      " Identities = 94/401 (23%), Positives = 150/401 (37%), Gaps = 87/401 (22%)\n",
      " Frame = +1\n",
      "\n",
      "Query  385   PKQGGDVVTTIDP-KVQKA-----GFEGLGNKKGAAVAIDPKTGAILGLVSTPSYDPGTF  546\n",
      "             P+Q  DV    D  +VQ+       F  +   + A +++DPK GAI  LV   S++   +\n",
      "Sbjct  363   PRQPADVAQAGDQIRVQRQEDGTLRFVQIPAAQSALISLDPKDGAIRSLVGGFSFEQSNY  422\n",
      "\n",
      "Query  547   AGGTKDDEKAWTALDSDPNKPMLNRALRETYPPGSTFKLVTAATAFETGKYQSPSDVTDT  726\n",
      "                                    NRA++    PGS+FK    + A + G + + S V D \n",
      "Sbjct  423   -----------------------NRAIQAKRQPGSSFKPFIYSAALDNG-FTAASLVNDA  458\n",
      "\n",
      "Query  727   P----DQYI----LPGTSTPLINASPTEDCGNATVQHAMDLSCNTVFGKMGAELGGDKLR  882\n",
      "             P    D+Y+     P   T           G   ++ A+  S N V  ++   LG ++  \n",
      "Sbjct  459   PIVFVDEYLDKVWRPKNDTNTF-------LGPIPLREALYKSRNMVSIRVLQGLGIERAI  511\n",
      "\n",
      "Query  883   AQAEKFGFNKE---------------VTIPIRADASHFPSGS-KPDGTAMDAI----GQH  1002\n",
      "             +   KFGF ++                 + I    S F +G  K +   ++ I    GQ \n",
      "Sbjct  512   SYITKFGFQRDELPRNFSLALGTATVTPMEIAGAWSVFANGGYKVNPYVIERIESRDGQV  571\n",
      "\n",
      "Query  1003  DTRATP------LQMAMVAAAIANNGSLMEPYLVAQEGS------AGNVISTHTEHQLSQ  1146\n",
      "               +A P       Q+A  A    N G    P     EGS      A    +T       +\n",
      "Sbjct  572   LYQANPPRVPVEEQVAADAEDAGNPGDPEHPESAEGEGSIEAQQVAAKAQTTFEPTPAER  631\n",
      "\n",
      "Query  1147  AVSPATAQKLQQLMESVVQHGTGTNA-KIPGVTVGGKTGTAQHGQDNSGLPFAWFVSYAK  1323\n",
      "              +   TA  +  +++ V++ GTG  A  +    + GKTGT    +D       WF  Y  \n",
      "Sbjct  632   IIDARTAYIMTSMLQDVIKRGTGRRALALKRTDLAGKTGTTNDSKD------GWFSGY--  683\n",
      "\n",
      "Query  1324  GQDGKQVAVAVVVEDGSSEAQQISGGSLAAPIAKAMMQAAL  1446\n",
      "                    +V V  +   +  ++  GG++A PI    M  AL\n",
      "Sbjct  684   -NSDYVTSVWVGFDQPETLGRREYGGTVALPIWIRYMGFAL  723\n",
      "\n",
      "\n",
      ">1PYY_A Chain A, Double Mutant Pbp2x T338aM339F FROM STREPTOCOCCUS PNEUMONIAE \n",
      "STRAIN R6 At 2.4 A Resolution\n",
      "Length=702\n",
      "\n",
      " Score = 43.9 bits (102),  Expect = 3e-04, Method: Compositional matrix adjust.\n",
      " Identities = 86/366 (23%), Positives = 138/366 (38%), Gaps = 57/366 (16%)\n",
      " Frame = +1\n",
      "\n",
      "Query  295   SQLEKLEDGILSGTDSRLFFRNT--------LDMLTGKPKQGGDVVTTIDPKVQK-----  435\n",
      "             S +E   + IL+GTD  + +            + ++ +   G DV TTI   +Q      \n",
      "Sbjct  169   SGMESSLNSILAGTDGIITYEKDRLGNIVPGTEQVSQRTMDGKDVYTTISSPLQSFMETQ  228\n",
      "\n",
      "Query  436   --AGFEGLGNKKGAAVAIDPKTGAILGLVSTPSYDPGTFAGGTKDDEKAWTALDSDPNKP  609\n",
      "               A  E +  K   A  +  KTG IL     P++D  T  G T+D    W  +    N  \n",
      "Sbjct  229   MDAFQEKVKGKYMTATLVSAKTGEILATTQRPTFDADTKEGITED--FVWRDILYQSN--  284\n",
      "\n",
      "Query  610   MLNRALRETYPPGSTFKLVTAATAFETGKYQSPSDVTDTPDQYILPGTSTPL-INASPTE  786\n",
      "                      Y PGS FK++  A A +   +    +V ++ +  I   T     +N   T \n",
      "Sbjct  285   ---------YEPGSAFKVMMLAAAIDNNTFPG-GEVFNSSELKIADATIRDWDVNEGLTG  334\n",
      "\n",
      "Query  787   DCGNATVQ---HAMDLSCNTVFGKMGAELGGDKLRAQAEKFGFNKEVTIPIRADASHFPS  957\n",
      "                    Q   H+ ++    +  KMG     D L     +F F     +P R   +   +\n",
      "Sbjct  335   GRMMTFSQGFAHSSNVGMTLLEQKMGDATWLDYLN----RFKFG----VPTRFGLTDEYA  386\n",
      "\n",
      "Query  958   GSKPD----GTAMDAIGQHDTRATPLQMAMVAAAIANNGSLMEPYLVAQEGSAGNVISTH  1125\n",
      "             G  P       A  + GQ     T  QM     AIAN+G ++EP  ++      +  +  \n",
      "Sbjct  387   GQLPADNIVNIAQSSFGQ-GISVTQTQMIRAFTAIANDGVMLEPKFISAIYDPNDQTARK  445\n",
      "\n",
      "Query  1126  TEHQL-SQAVSPATAQKLQQLME---------SVVQHGTGT-NAKIPGVTVGGKTGTAQH  1272\n",
      "             ++ ++    VS   A   +  M          ++  H TG     +PG  V  K+GTAQ \n",
      "Sbjct  446   SQKEIVGNPVSKDAASLTRTNMVLVGTDPVYGTMYNHSTGKPTVTVPGQNVALKSGTAQI  505\n",
      "\n",
      "Query  1273  GQDNSG  1290\n",
      "               + +G\n",
      "Sbjct  506   ADEKNG  511\n",
      "\n",
      "\n",
      ">1PMD_A Chain A, Penicillin-Binding Protein 2x (Pbp-2x)\n",
      "Length=675\n",
      "\n",
      " Score = 43.5 bits (101),  Expect = 5e-04, Method: Compositional matrix adjust.\n",
      " Identities = 86/366 (23%), Positives = 138/366 (38%), Gaps = 57/366 (16%)\n",
      " Frame = +1\n",
      "\n",
      "Query  295   SQLEKLEDGILSGTDSRLFFRNT--------LDMLTGKPKQGGDVVTTIDPKVQK-----  435\n",
      "             S +E   + IL+GTD  + +            + ++ +   G DV TTI   +Q      \n",
      "Sbjct  142   SGMESSLNSILAGTDGIITYEKDRLGNIVPGTEQVSQRTMDGKDVYTTISSPLQSFMETQ  201\n",
      "\n",
      "Query  436   --AGFEGLGNKKGAAVAIDPKTGAILGLVSTPSYDPGTFAGGTKDDEKAWTALDSDPNKP  609\n",
      "               A  E +  K   A  +  KTG IL     P++D  T  G T+D    W  +    N  \n",
      "Sbjct  202   MDAFQEKVKGKYMTATLVSAKTGEILATTQRPTFDADTKEGITED--FVWRDILYQSN--  257\n",
      "\n",
      "Query  610   MLNRALRETYPPGSTFKLVTAATAFETGKYQSPSDVTDTPDQYILPGTSTPL-INASPTE  786\n",
      "                      Y PGST K++  A A +   +    +V ++ +  I   T     +N   T \n",
      "Sbjct  258   ---------YEPGSTMKVMMLAAAIDNNTFPG-GEVFNSSELKIADATIRDWDVNEGLTG  307\n",
      "\n",
      "Query  787   DCGNATVQ---HAMDLSCNTVFGKMGAELGGDKLRAQAEKFGFNKEVTIPIRADASHFPS  957\n",
      "                    Q   H+ ++    +  KMG     D L     +F F     +P R   +   +\n",
      "Sbjct  308   GRMMTFSQGFAHSSNVGMTLLEQKMGDATWLDYLN----RFKFG----VPTRFGLTDEYA  359\n",
      "\n",
      "Query  958   GSKPD----GTAMDAIGQHDTRATPLQMAMVAAAIANNGSLMEPYLVAQEGSAGNVISTH  1125\n",
      "             G  P       A  + GQ     T  QM     AIAN+G ++EP  ++      +  +  \n",
      "Sbjct  360   GQLPADNIVNIAQSSFGQG-ISVTQTQMIRAFTAIANDGVMLEPKFISAIYDPNDQTARK  418\n",
      "\n",
      "Query  1126  TEHQL-SQAVSPATAQKLQQLME---------SVVQHGTGT-NAKIPGVTVGGKTGTAQH  1272\n",
      "             ++ ++    VS   A   +  M          ++  H TG     +PG  V  K+GTAQ \n",
      "Sbjct  419   SQKEIVGNPVSKDAASLTRTNMVLVGTDPVYGTMYNHSTGKPTVTVPGQNVALKSGTAQI  478\n",
      "\n",
      "Query  1273  GQDNSG  1290\n",
      "               + +G\n",
      "Sbjct  479   ADEKNG  484\n",
      "\n",
      "\n",
      ">1QME_A Chain A, Penicillin-binding Protein 2x (pbp-2x)\n",
      " 1QMF_A Chain A, Penicillin-binding Protein 2x (pbp-2x) Acyl-enzyme Complex\n",
      "Length=702\n",
      "\n",
      " Score = 43.1 bits (100),  Expect = 6e-04, Method: Compositional matrix adjust.\n",
      " Identities = 86/366 (23%), Positives = 138/366 (38%), Gaps = 57/366 (16%)\n",
      " Frame = +1\n",
      "\n",
      "Query  295   SQLEKLEDGILSGTDSRLFFRNT--------LDMLTGKPKQGGDVVTTIDPKVQK-----  435\n",
      "             S +E   + IL+GTD  + +            + ++ +   G DV TTI   +Q      \n",
      "Sbjct  169   SGMESSLNSILAGTDGIITYEKDRLGNIVPGTEQVSQRTMDGKDVYTTISSPLQSFMETQ  228\n",
      "\n",
      "Query  436   --AGFEGLGNKKGAAVAIDPKTGAILGLVSTPSYDPGTFAGGTKDDEKAWTALDSDPNKP  609\n",
      "               A  E +  K   A  +  KTG IL     P++D  T  G T+D    W  +    N  \n",
      "Sbjct  229   MDAFQEKVKGKYMTATLVSAKTGEILATTQRPTFDADTKEGITED--FVWRDILYQSN--  284\n",
      "\n",
      "Query  610   MLNRALRETYPPGSTFKLVTAATAFETGKYQSPSDVTDTPDQYILPGTSTPL-INASPTE  786\n",
      "                      Y PGST K++  A A +   +    +V ++ +  I   T     +N   T \n",
      "Sbjct  285   ---------YEPGSTMKVMMLAAAIDNNTFPG-GEVFNSSELKIADATIRDWDVNEGLTG  334\n",
      "\n",
      "Query  787   DCGNATVQ---HAMDLSCNTVFGKMGAELGGDKLRAQAEKFGFNKEVTIPIRADASHFPS  957\n",
      "                    Q   H+ ++    +  KMG     D L     +F F     +P R   +   +\n",
      "Sbjct  335   GRMMTFSQGFAHSSNVGMTLLEQKMGDATWLDYLN----RFKFG----VPTRFGLTDEYA  386\n",
      "\n",
      "Query  958   GSKPD----GTAMDAIGQHDTRATPLQMAMVAAAIANNGSLMEPYLVAQEGSAGNVISTH  1125\n",
      "             G  P       A  + GQ     T  QM     AIAN+G ++EP  ++      +  +  \n",
      "Sbjct  387   GQLPADNIVNIAQSSFGQ-GISVTQTQMIRAFTAIANDGVMLEPKFISAIYDPNDQTARK  445\n",
      "\n",
      "Query  1126  TEHQL-SQAVSPATAQKLQQLME---------SVVQHGTGT-NAKIPGVTVGGKTGTAQH  1272\n",
      "             ++ ++    VS   A   +  M          ++  H TG     +PG  V  K+GTAQ \n",
      "Sbjct  446   SQKEIVGNPVSKDAASLTRTNMVLVGTDPVYGTMYNHSTGKPTVTVPGQNVALKSGTAQI  505\n",
      "\n",
      "Query  1273  GQDNSG  1290\n",
      "               + +G\n",
      "Sbjct  506   ADEKNG  511\n",
      "\n",
      "\n",
      ">4YE5_A Chain A, The Crystal Structure Of A Peptidoglycan Synthetase \n",
      "From Bifidobacterium Adolescentis Atcc 15703\n",
      " 4YE5_B Chain B, The Crystal Structure Of A Peptidoglycan Synthetase \n",
      "From Bifidobacterium Adolescentis Atcc 15703\n",
      "Length=566\n",
      "\n",
      " Score = 42.7 bits (99),  Expect = 7e-04, Method: Compositional matrix adjust.\n",
      " Identities = 85/405 (21%), Positives = 158/405 (39%), Gaps = 55/405 (14%)\n",
      " Frame = +1\n",
      "\n",
      "Query  301   LEKLEDGILSGTDSRLFFR--NTLDMLTGKPKQ------GGDVVTTIDPKVQKAGFEGLG  456\n",
      "             +E+ E+  L+G D    ++  N+   + G   +      G DV  TID  VQ    + L \n",
      "Sbjct  173   IEQXENKTLTGRDGYQVYQQGNSGVEIPGTXTESKDAVNGSDVTLTIDRDVQWYTEKVLS  232\n",
      "\n",
      "Query  457   NKK-------GAAVAIDPKTGAILGLVSTPSYDPGTFAGGTKDDEKAWTALDSDPNKPML  615\n",
      "             + +       G A   D ++G IL L  + + + G                 SD  K   \n",
      "Sbjct  233   DSENKYHSAWGIAXVQDVQSGDILALADSDTTEAG-----------------SDQAKXGA  275\n",
      "\n",
      "Query  616   NRALRETYPPGSTFKLVTAATAFETGKYQSPSDVTDTPDQYILPGTSTPLINASPTEDCG  795\n",
      "             +RA+ ET+ PGS  K++  +   + G ++         D++ +P T T  +     +D  \n",
      "Sbjct  276   SRAVSETFEPGSIGKVLAXSGXLQLGLHK-------IDDKFTVPNTVT--VEGQTYKDAV  326\n",
      "\n",
      "Query  796   NATVQHAMDLSCNTVFGKMGAELGGDKLRAQAE-----KFGFNKE--VTIPIRADASHFP  954\n",
      "             +   +H            +G  + GDK   +       KFG  +   + +P  ++    P\n",
      "Sbjct  327   DHGNEHWTLAGILEQSSNVGXVIAGDKXTNEQRYNFISKFGIGQATGLNLPGESEGVLHP  386\n",
      "\n",
      "Query  955   SGSKPDGTAMDAIGQHDTRATPLQMAMVAAAIANNGSLMEPYLVAQE--GSAGNVISTHT  1128\n",
      "             S S    T    +          Q+    + IAN G + +P  + +    +AG+V    +\n",
      "Sbjct  387   SDSWDRRTRNTVLFGQGYTVNVXQLTNAISVIANKG-VKKPQRIIKSITDTAGHVEEQQS  445\n",
      "\n",
      "Query  1129  EHQLSQAVSPATAQKLQQLMESVVQHGTGTNAKIPGVTVGGKTGTAQHGQDNSGLPFAWF  1308\n",
      "             + + ++ +  + A +     ES  +H   T  K+ G     K+GTA+    N  L  +  \n",
      "Sbjct  446   KGEATRVIDESVASQXLNAXESSAEH-YNTFVKVDGYRXAAKSGTAEVAGANGQLT-SII  503\n",
      "\n",
      "Query  1309  VSYAK--GQDGKQVAVAVVVEDGSSEAQQISGGSLAAPIAKAMMQ  1437\n",
      "               Y+     D  +  + VV++D       ++ G + A I + + Q\n",
      "Sbjct  504   SDYSTIIPADNPRFVITVVLKDPQGSFGGLTAGPVTAEIGEFLXQ  548\n",
      "\n",
      "\n",
      ">3DWK_A Chain A, Identification Of Dynamic Structural Motifs Involved \n",
      "In Peptidoglycan Glycosyltransfer\n",
      " 3DWK_B Chain B, Identification Of Dynamic Structural Motifs Involved \n",
      "In Peptidoglycan Glycosyltransfer\n",
      " 3DWK_C Chain C, Identification Of Dynamic Structural Motifs Involved \n",
      "In Peptidoglycan Glycosyltransfer\n",
      " 3DWK_D Chain D, Identification Of Dynamic Structural Motifs Involved \n",
      "In Peptidoglycan Glycosyltransfer\n",
      "Length=625\n",
      "\n",
      " Score = 42.4 bits (98),  Expect = 0.001, Method: Compositional matrix adjust.\n",
      " Identities = 69/303 (23%), Positives = 120/303 (40%), Gaps = 61/303 (20%)\n",
      " Frame = +1\n",
      "\n",
      "Query  400   DVVTTIDPKVQKAGFEGLGNKKGAAVAIDPKTGAILGLVSTPSYDPGTFAGGTKDDEKAW  579\n",
      "             DV  T+   V    F    +++  A  +D KTG ++ +             G +D     \n",
      "Sbjct  271   DVQKTLQNDVDNGSFYKNKDQQVGATILDSKTGGLVAI------------SGGRD-----  313\n",
      "\n",
      "Query  580   TALDSDPNKPMLNR-ALRETYPPGSTFK-LVTAATAFETGKYQSPSDVTDTPDQYILPGT  753\n",
      "                     K ++NR    + +P GS+ K  +    A E  K+ +   + D    Y + G+\n",
      "Sbjct  314   -------FKDVVNRNQATDPHPTGSSLKPFLAYGPAIENMKWATNHAIQDES-SYQVDGS  365\n",
      "\n",
      "Query  754   STPLINASPTEDCGNATVQHAMDLSCNT----VFGKMGAELGGDKLRAQAEKFGFNKEVT  921\n",
      "             +        T+  G  ++  A+  S N      +  +    G D  +  A K G N E  \n",
      "Sbjct  366   T---FRNYDTKSHGTVSIYDALRQSFNIPALKAWQSVKQNAGNDAPKKFAAKLGLNYEGD  422\n",
      "\n",
      "Query  922   IPIRADASHFPSGSKPDGTAMDAIGQHDTRATPLQMAMVAAAIANNGSLMEPY----LVA  1089\n",
      "             I         PS         + +G   +  +P Q+A   AAIAN G+    +    +V \n",
      "Sbjct  423   IG--------PS---------EVLGGSASEFSPTQLASAFAAIANGGTYNNAHSIQKVVT  465\n",
      "\n",
      "Query  1090  QEGSAGNVISTHTEHQLSQAVSPATAQKLQQLMESVVQ-HGTGTNAKIPGVTVGGKTGTA  1266\n",
      "             ++G    +   HT H   +A+S  TA  L ++++   + +G+     + GV +G KTGT \n",
      "Sbjct  466   RDGET--IEYDHTSH---KAMSDYTAYMLAEMLKGTFKPYGSAYGHGVSGVNMGAKTGTG  520\n",
      "\n",
      "Query  1267  QHG  1275\n",
      "              +G\n",
      "Sbjct  521   TYG  523\n",
      "\n",
      "\n",
      ">2Z2L_B Chain B, Penicillin-Binding Protein 2x (Pbp2x) From Streptococcus \n",
      "Pneumoniae\n",
      " 2Z2L_E Chain E, Penicillin-Binding Protein 2x (Pbp2x) From Streptococcus \n",
      "Pneumoniae\n",
      " 2Z2M_B Chain B, Cefditoren-Acylated Penicillin-Binding Protein 2x (Pbp2x) \n",
      "From Streptococcus Pneumoniae\n",
      " 2Z2M_E Chain E, Cefditoren-Acylated Penicillin-Binding Protein 2x (Pbp2x) \n",
      "From Streptococcus Pneumoniae\n",
      " 2ZC3_B Chain B, Penicillin-Binding Protein 2x (Pbp 2x) Acyl-Enzyme Complex \n",
      "(Biapenem) From Streptococcus Pneumoniae\n",
      " 2ZC3_E Chain E, Penicillin-Binding Protein 2x (Pbp 2x) Acyl-Enzyme Complex \n",
      "(Biapenem) From Streptococcus Pneumoniae\n",
      " 2ZC4_B Chain B, Penicillin-Binding Protein 2x (Pbp 2x) Acyl-Enzyme Complex \n",
      "(Tebipenem) From Streptococcus Pneumoniae\n",
      " 2ZC4_E Chain E, Penicillin-Binding Protein 2x (Pbp 2x) Acyl-Enzyme Complex \n",
      "(Tebipenem) From Streptococcus Pneumoniae\n",
      "Length=385\n",
      "\n",
      " Score = 40.4 bits (93),  Expect = 0.004, Method: Compositional matrix adjust.\n",
      " Identities = 79/334 (24%), Positives = 126/334 (38%), Gaps = 49/334 (15%)\n",
      " Frame = +1\n",
      "\n",
      "Query  367   DMLTGKPKQGGDVVTTIDPKVQK-------AGFEGLGNKKGAAVAIDPKTGAILGLVSTP  525\n",
      "             + ++ +   G DV TTI   +Q        A  E +  K   A  +  KTG IL     P\n",
      "Sbjct  9     EQVSQRTMDGKDVYTTISSPLQSFMETQMDAFQEKVKGKYMTATLVSAKTGEILATTQRP  68\n",
      "\n",
      "Query  526   SYDPGTFAGGTKDDEKAWTALDSDPNKPMLNRALRETYPPGSTFKLVTAATAFETGKYQS  705\n",
      "             ++D  T  G T+D    W  +    N           Y PGST K++  A A +   +  \n",
      "Sbjct  69    TFDADTKEGITED--FVWRDILYQSN-----------YEPGSTMKVMMLAAAIDNNTFPG  115\n",
      "\n",
      "Query  706   PSDVTDTPDQYILPGTSTPL-INASPTEDCGNATVQ---HAMDLSCNTVFGKMGAELGGD  873\n",
      "               +V ++ +  I   T     +N   T        Q   H+ ++    +  KMG     D\n",
      "Sbjct  116   -GEVFNSSELKIADATIRDWDVNEGLTGGRMMTFSQGFAHSSNVGMTLLEQKMGDATWLD  174\n",
      "\n",
      "Query  874   KLRAQAEKFGFNKEVTIPIRADASHFPSGSKPD----GTAMDAIGQHDTRATPLQMAMVA  1041\n",
      "              L     +F F     +P R   +   +G  P       A  + GQ     T  QM    \n",
      "Sbjct  175   YLN----RFKFG----VPTRFGLTDEYAGQLPADNIVNIAQSSFGQG-ISVTQTQMIRAF  225\n",
      "\n",
      "Query  1042  AAIANNGSLMEPYLVAQEGSAGNVISTHTEHQL-SQAVSPATAQKLQQLME---------  1191\n",
      "              AIAN+G ++EP  ++      +  +  ++ ++    VS   A   +  M          \n",
      "Sbjct  226   TAIANDGVMLEPKFISAIYDPNDQTARKSQKEIVGNPVSKDAASLTRTNMVLVGTDPVYG  285\n",
      "\n",
      "Query  1192  SVVQHGTGT-NAKIPGVTVGGKTGTAQHGQDNSG  1290\n",
      "             ++  H TG     +PG  V  K+GTAQ   + +G\n",
      "Sbjct  286   TMYNHSTGKPTVTVPGQNVALKSGTAQIADEKNG  319\n",
      "\n",
      "\n",
      ">2OLU_A Chain A, Structural Insight Into The Transglycosylation Step \n",
      "Of Bacterial Cell Wall Biosynthesis : Apoenzyme\n",
      " 2OLV_A Chain A, Structural Insight Into The Transglycosylation Step \n",
      "Of Bacterial Cell Wall Biosynthesis : Donor Ligand Complex\n",
      " 2OLV_B Chain B, Structural Insight Into The Transglycosylation Step \n",
      "Of Bacterial Cell Wall Biosynthesis : Donor Ligand Complex\n",
      "Length=669\n",
      "\n",
      " Score = 40.0 bits (92),  Expect = 0.005, Method: Compositional matrix adjust.\n",
      " Identities = 69/303 (23%), Positives = 117/303 (39%), Gaps = 61/303 (20%)\n",
      " Frame = +1\n",
      "\n",
      "Query  400   DVVTTIDPKVQKAGFEGLGNKKGAAVAIDPKTGAILGLVSTPSYDPGTFAGGTKDDEKAW  579\n",
      "             DV  T+   V    F    +++  A  +D KTG ++ +             G +D     \n",
      "Sbjct  280   DVQKTLQNDVDNGSFYKNKDQQVGATILDSKTGGLVAI------------SGGRD-----  322\n",
      "\n",
      "Query  580   TALDSDPNKPMLNR-ALRETYPPGSTFK-LVTAATAFETGKYQSPSDVTDTPDQYILPGT  753\n",
      "                     K ++NR    + +P GS+ K  +    A E  K+ +   + D    Y + G+\n",
      "Sbjct  323   -------FKDVVNRNQATDPHPTGSSLKPFLAYGPAIENXKWATNHAIQDES-SYQVDGS  374\n",
      "\n",
      "Query  754   STPLINASPTEDCGNATVQHAMDLSCN----TVFGKMGAELGGDKLRAQAEKFGFNKEVT  921\n",
      "             +        T+  G  ++  A+  S N      +  +    G D  +  A K G N E  \n",
      "Sbjct  375   T---FRNYDTKSHGTVSIYDALRQSFNIPALKAWQSVKQNAGNDAPKKFAAKLGLNYEGD  431\n",
      "\n",
      "Query  922   IPIRADASHFPSGSKPDGTAMDAIGQHDTRATPLQMAMVAAAIANNGSLMEPY----LVA  1089\n",
      "             I         PS         + +G   +  +P Q+A   AAIAN G+    +    +V \n",
      "Sbjct  432   IG--------PS---------EVLGGSASEFSPTQLASAFAAIANGGTYNNAHSIQKVVT  474\n",
      "\n",
      "Query  1090  QEGSAGNVISTHTEHQLSQAVSPATAQKLQQLMESVVQ-HGTGTNAKIPGVTVGGKTGTA  1266\n",
      "             ++G    +   HT H   +A S  TA  L + ++   + +G+     + GV  G KTGT \n",
      "Sbjct  475   RDGET--IEYDHTSH---KAXSDYTAYXLAEXLKGTFKPYGSAYGHGVSGVNXGAKTGTG  529\n",
      "\n",
      "Query  1267  QHG  1275\n",
      "              +G\n",
      "Sbjct  530   TYG  532\n",
      "\n",
      "\n",
      ">2Y2G_A Chain A, Penicillin-Binding Protein 1b (Pbp-1b) In Complex With \n",
      "An Alkyl Boronate (A01)\n",
      " 2Y2G_B Chain B, Penicillin-Binding Protein 1b (Pbp-1b) In Complex With \n",
      "An Alkyl Boronate (A01)\n",
      " 2Y2H_A Chain A, Penicillin-Binding Protein 1b (Pbp-1b) In Complex With \n",
      "An Alkyl Boronate (Za2)\n",
      " 2Y2H_B Chain B, Penicillin-Binding Protein 1b (Pbp-1b) In Complex With \n",
      "An Alkyl Boronate (Za2)\n",
      " 2Y2I_A Chain A, Penicillin-Binding Protein 1b (Pbp-1b) In Complex With \n",
      "An Alkyl Boronate (Za3)\n",
      " 2Y2J_A Chain A, Penicillin-Binding Protein 1b (Pbp-1b) In Complex With \n",
      "An Alkyl Boronate (Za4)\n",
      " 2Y2K_A Chain A, Penicillin-Binding Protein 1b (Pbp-1b) In Complex With \n",
      "An Alkyl Boronate (Za5)\n",
      " 2Y2L_A Chain A, Penicillin-binding Protein 1b (pbp-1b) In Complex With \n",
      "An Alkyl Boronate (e06)\n",
      " 2Y2L_B Chain B, Penicillin-binding Protein 1b (pbp-1b) In Complex With \n",
      "An Alkyl Boronate (e06)\n",
      " 2Y2M_A Chain A, Penicillin-Binding Protein 1b (Pbp-1b) In Complex With \n",
      "An Alkyl Boronate (E08)\n",
      " 2Y2N_A Chain A, Penicillin-Binding Protein 1b (Pbp-1b) In Complex With \n",
      "An Alkyl Boronate (E07)\n",
      " 2Y2O_A Chain A, Penicillin-binding Protein 1b (pbp-1b) In Complex With \n",
      "An Alkyl Boronate (eo9)\n",
      " 2Y2P_A Chain A, Penicillin-Binding Protein 1b (Pbp-1b) In Complex With \n",
      "An Alkyl Boronate (Z10)\n",
      " 2Y2Q_A Chain A, Penicillin-Binding Protein 1b (Pbp-1b) In Complex With \n",
      "An Alkyl Boronate (Z06)\n",
      " 2Y2Q_B Chain B, Penicillin-Binding Protein 1b (Pbp-1b) In Complex With \n",
      "An Alkyl Boronate (Z06)\n",
      "Length=494\n",
      "\n",
      " Score = 39.7 bits (91),  Expect = 0.008, Method: Compositional matrix adjust.\n",
      " Identities = 84/338 (25%), Positives = 134/338 (40%), Gaps = 80/338 (24%)\n",
      " Frame = +1\n",
      "\n",
      "Query  349   FFRNTLDMLTGKPKQGG-DVVTTIDPKVQKA--------GF---EGLGNKKGAAVAIDPK  492\n",
      "             F+R   D+   + + GG  + TTID K+  A        G+   +G G  +   V +D +\n",
      "Sbjct  77    FYR---DLAAKEIENGGYKITTTIDQKIHSAMQSAVADYGYLLDDGTGRVEVGNVLMDNQ  133\n",
      "\n",
      "Query  493   TGAILGLVSTPSYDPGTFAGGTKDDEKAWTALDSDPNKPMLNRALRETYPPGSTFK-LVT  669\n",
      "             TGAILG           F GG    E               N A      P ST K L+ \n",
      "Sbjct  134   TGAILG-----------FVGGRNYQEN------------QNNHAFDTKRSPASTTKPLLA  170\n",
      "\n",
      "Query  670   AATAFETGKYQSPSDVTDTPDQYILPGTSTPLINASPTEDCGNATVQHAMDLSCNT---V  840\n",
      "                A + G   S + +++ P  +       P++ A+ ++  G  T+  A++ S N     \n",
      "Sbjct  171   YGIAIDQGLMGSETILSNYPTNF---ANGNPIMYAN-SKGTGMMTLGEALNYSWNIPAYW  226\n",
      "\n",
      "Query  841   FGKMGAELGGDKLRAQAEKFGFNKEVTIPIRADASHFPSGSKPDGTAMDA-IGQHDTRAT  1017\n",
      "               +M  E G D ++   EK G+     IP       +   S P G  ++  + QH     \n",
      "Sbjct  227   TYRMLRENGVD-VKGYMEKMGYE----IP------EYGIESLPMGGGIEVTVAQHTNGYQ  275\n",
      "\n",
      "Query  1018  PLQMAMVAAAIANNGSLMEPYLVAQ-EGSAGNVISTHTEHQLSQAVSPATAQKLQQLMES  1194\n",
      "              L         ANNG   + +++++ E + G V+  + +  + Q  S ATA  +Q L+  \n",
      "Sbjct  276   TL---------ANNGVYHQKHVISKIEAADGRVVYEYQDKPV-QVYSKATATIMQGLLRE  325\n",
      "\n",
      "Query  1195  VV---------QHGTGTNAKIPGVTVGGKTGTAQHGQD  1281\n",
      "             V+          + T  N  +      GKTGT   GQD\n",
      "Sbjct  326   VLSSRVTTTFKSNLTSLNPTLANADWIGKTGTT--GQD  361\n",
      "\n",
      "\n",
      ">5FGZ_A Chain A, E. Coli Pbp1b In Complex With Fpi-1465\n",
      " 5HL9_A Chain A, E. Coli Pbp1b In Complex With Acyl-ampicillin And Moenomycin\n",
      " 5HLA_A Chain A, E. Coli Pbp1b In Complex With Acyl-cephalexin And Moenomycin\n",
      " 5HLB_A Chain A, E. Coli Pbp1b In Complex With Acyl-aztreonam And Moenomycin\n",
      " 5HLD_A Chain A, E. Coli Pbp1b In Complex With Acyl-centa And Moenomycin\n",
      "Length=747\n",
      "\n",
      " Score = 39.7 bits (91),  Expect = 0.009, Method: Compositional matrix adjust.\n",
      " Identities = 71/286 (25%), Positives = 110/286 (38%), Gaps = 44/286 (15%)\n",
      " Frame = +1\n",
      "\n",
      "Query  592   SDPNKPMLNRALRETYPPGSTFKLVTAATAFETGK-YQSPSDVTDTPDQYILPG--TSTP  762\n",
      "             S+P     NRA++     GS  K  T  TA    K Y+  + + D P     P     +P\n",
      "Sbjct  434   SEPQFAGYNRAMQARRSIGSLAKPATYLTALSQPKIYRLNTWIADAPIALRQPNGQVWSP  493\n",
      "\n",
      "Query  763   LINASPTEDCGNATVQHAMDLSCNTVFGKMGAELGGDKLRAQAEKFGFNKEVTIPIRADA  942\n",
      "               +     + G   +  A+  S N     +G  LG   +     K G  K+   P+ A  \n",
      "Sbjct  494   QNDDRRYSESGRVMLVDALTRSMNVPTVNLGMALGLPAVTETWIKLGVPKDQLHPVPA--  551\n",
      "\n",
      "Query  943   SHFPSGSKPDGTAMDAIGQHDTRATPLQMAMVAAAIANNGSLMEPY-----LVAQEGSAG  1107\n",
      "                          + A+       TP+++A     IA+ G+   P      ++A++G   \n",
      "Sbjct  552   -----------MLLGALN-----LTPIEVAQAFQTIASGGN-RAPLSALRSVIAEDGK--  592\n",
      "\n",
      "Query  1108  NVISTHTEHQLSQAVSPATAQKLQQLMESVVQHGTGTN--AKIPGVTVGGKTGTAQHGQD  1281\n",
      "               +   +  Q  +AV    A      M+ VVQ GTG    AK P + + GKTGT  +  D\n",
      "Sbjct  593   --VLYQSFPQAERAVPAQAAYLTLWTMQQVVQRGTGRQLGAKYPNLHLAGKTGTTNNNVD  650\n",
      "\n",
      "Query  1282  NSGLPFAWFVSYAKGQDGKQVAVAVVVEDGSSEAQQISGGSLAAPI  1419\n",
      "                    WF     G DG  V +  V  D +++  ++ G S A  I\n",
      "Sbjct  651   T------WFA----GIDGSTVTITWVGRD-NNQPTKLYGASGAMSI  685\n",
      "\n",
      "\n",
      "  Database: PDB protein database\n",
      "    Posted date:  Feb 26, 2017  11:31 AM\n",
      "  Number of letters in database: 22,414,899\n",
      "  Number of sequences in database:  89,751\n",
      "\n",
      "Lambda      K        H\n",
      "   0.314    0.131    0.383 \n",
      "Gapped\n",
      "Lambda      K        H\n",
      "   0.267   0.0410    0.140 \n",
      "Matrix: BLOSUM62\n",
      "Gap Penalties: Existence: 11, Extension: 1\n",
      "Number of Sequences: 89751\n",
      "Number of Hits to DB: 7829287\n",
      "Number of extensions: 214620\n",
      "Number of successful extensions: 891\n",
      "Number of sequences better than 10: 21\n",
      "Number of HSP's better than 10 without gapping: 0\n",
      "Number of HSP's gapped: 856\n",
      "Number of HSP's successfully gapped: 23\n",
      "Length of query: 1455\n",
      "Length of database: 22414899\n",
      "Length adjustment: 106\n",
      "Effective length of query: 1349\n",
      "Effective length of database: 12901293\n",
      "Effective search space: 4889590047\n",
      "Effective search space used: 4889590047\n",
      "T: 12\n",
      "A: 40\n",
      "X1: 16 (7.3 bits)\n",
      "X2: 38 (14.6 bits)\n",
      "X3: 64 (24.7 bits)\n",
      "S1: 41 (20.4 bits)\n",
      "S2: 63 (28.9 bits)\n",
      "ka-blk-alpha gapped: 1.9\n",
      "ka-blk-alpha ungapped: 0.7916\n",
      "ka-blk-alpha_v gapped: 42.6028\n",
      "ka-blk-alpha_v ungapped: 4.96466\n",
      "ka-blk-sigma gapped: 43.6362\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Read the contents of the handle, and print to the cell\n",
    "output = result_handle.read()\n",
    "print(output)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "deletable": true,
    "editable": true
   },
   "source": [
    "<div class=\"alert-warning\">\n",
    "<h3>QUESTIONS</h3>\n",
    "<ol>\n",
    "<li> How is this output different to that from command-line and web-interface `BLAST` results?\n",
    "<li> How many hits are there?\n",
    "<li> What is the \"best hit\" to the query? Do you think it is a 'good' hit?\n",
    "<li> At what point do you think the matches start to become less reliable? Why do you think this? (*HINT:* inspect the alignments)\n",
    "</ol>\n",
    "</div>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "deletable": true,
    "editable": true
   },
   "source": [
    "<a id=\"save\"></a>\n",
    "### Save `BLASTX` results to file\n",
    "\n",
    "The results you obtianed above are human-readable, and similar to the default output type you saw from command-line/terminal `BLAST` in [notebook 02](02-blast_at_terminal.ipynb). But, for now, they exist only in the variable called `output`. If we want to come back to these results at some time in the future, we will need to save them to a file somewhere.\n",
    "\n",
    "This is a common operation in programmatic approaches to bioinformatics: once a result is obtained, we usually want to save it to a file. Most programs will do this for you, but here you need to save it explicitly, yourself.\n",
    "\n",
    "The Python code for saving the contents of `output` to the file `output/kitasatospora/remote_blastx_query_01.txt` is given in the cell below:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [],
   "source": [
    "#Make directory to save output, if it doesn't exist\n",
    "outdir = os.path.join(\"output\", \"kitasatospora\")\n",
    "os.makedirs(outdir, exist_ok=True)\n",
    "\n",
    "# Save output to file\n",
    "outfilename = os.path.join(outdir, \"remote_blastx_query_01.txt\")\n",
    "with open(outfilename, 'w') as outfh:\n",
    "    outfh.write(output)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "deletable": true,
    "editable": true
   },
   "source": [
    "This code does three main things:\n",
    "\n",
    "1. It creates a variable called `outfilename`, with the path to the file we want to write\n",
    "2. It opens that file, ready for writing, as a *handle* called `outfh`\n",
    "3. It writes the contents of `output` into the `outfh` *handle*\n",
    "\n",
    "When this is done, the `BLAST` search results we got from NCBI are written to the file `output/kitasatospora/remote_blastx_query_01.txt`, just as though we did the search locally. You can inspect the contents of that file at the terminal using a command like:\n",
    "\n",
    "```bash\n",
    "less output/kitasatospora/remote_blastx_query_01.txt\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "deletable": true,
    "editable": true
   },
   "source": [
    "<a id=\"entrez\"></a>\n",
    "### Using Entrez queries to modify the remote database\n",
    "\n",
    "The remote NCBI `BLAST+` service provides large, comprehensive databases such as `nr`, `nt` and `refseq`, but it doesn't provide very many specialised databases. Searches against large databases - where you don't care about most of the searches, and most of the sequences are unlikely to match - can be wasteful and take longer than necessary to get the useful answer you're looking for. But a smaller, specialised organism-centric database - which would be much quicker to search - may not exist separately.\n",
    "\n",
    "<br></br>\n",
    "<div class=\"alert-success\">\n",
    "<b>NCBI provide a service called `Entrez E-Utilities` which allows for complex text-based searches to be defined. A detailed consideration of this very powerful tool is beyond the scope of this course, but we will use the service to replicate the `Organism` field of the NCBI web browser interface that allowed us to restrict the `nr` database only to <i>Kitasatospora</i> sequences.</b>\n",
    "</div>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "deletable": true,
    "editable": true
   },
   "source": [
    "In [notebook 01](01-blast_at_NCBI_website.ipynb) you used the `Organism` field in the web browser interface to specify `Kitasatospora (taxid: 2063)` when querying against the complete `nr` database. This had the effect of only querying against sequences in `nr` that originated from *Kitasatospora*, reducing the overall search time and giving you the relevant hits you needed without producing a very large set of output.\n",
    "\n",
    "To perform a similar filtering of the search using `NCBIWWW.qblast()`, you specify the argument `entrez_query`. The value to be passed is a string that describes the search field you want to filter on. There are many search field options (see the [list here](https://www.ncbi.nlm.nih.gov/books/NBK49540/)), but we will use only one in this lesson: the `organism` field."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "deletable": true,
    "editable": true
   },
   "source": [
    "#### The `organism` field `[ORGN]`\n",
    "\n",
    "The syntax for filtering on organism is `<value>[ORGN]` where `<value>` is the term you want to filter on. This can be the name of the organism, or an identifier such as the NCBI taxon ID.\n",
    "\n",
    "In the cell below, the remote search is conducted on the `nr` database, but restricted only to the set of sequences originating from *Kitasatospora*, by specififying `entrez_query=\"kitasatospora[ORGN]\"`:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [],
   "source": [
    "# Repeat the BLAST search on the nr database, restricted to Kitasatospora spp.\n",
    "result_handle = NCBIWWW.qblast(\"blastx\", \"nr\", seq.format(\"fasta\"),\n",
    "                               entrez_query=\"kitasatospora[ORGN]\",\n",
    "                               format_type=\"Text\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [],
   "source": [
    "# Show human-readable results\n",
    "output = result_handle.read()\n",
    "print(output)\n",
    "\n",
    "# Save output to file\n",
    "outfilename = os.path.join(outdir, \"remote_blastx_query_02.txt\")\n",
    "with open(outfilename, 'w') as outfh:\n",
    "    outfh.write(output)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "deletable": true,
    "editable": true
   },
   "source": [
    "<div class=\"alert-warning\">\n",
    "<h3>QUESTIONS</h3>\n",
    "<ol>\n",
    "<li> How does this output differ from that of the same search conducted through the web browser interface?\n",
    "</ol>\n",
    "</div>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "deletable": true,
    "editable": true
   },
   "source": [
    "<img src=\"images/exercise.png\" style=\"width: 100px; float: left;\">\n",
    "<a id=\"ex01\"></a>\n",
    "### Exercise 01 (10min)\n",
    "\n",
    "Using the sequence in the file `data/kitasatospora/lantibiotic.fasta` as a query, can you do the following?\n",
    "\n",
    "<br><div class=\"alert-danger\">\n",
    "<ul>\n",
    "<li> Conduct a remote `BLASTX` query against the `nr` database, restricted to <i>Kitasatospora</i> proteins\n",
    "<li> Write the output to a new file called `lantibiotic_blastx_nr_kitasatospora.txt` in the output directory `output/kitasatospora`.\n",
    "<li> Compare the results to the output from Exercise 01 in notebook 03.\n",
    "</ul>\n",
    "</div></br>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true,
    "deletable": true,
    "editable": true
   },
   "outputs": [],
   "source": [
    "# SOLUTION - EXERCISE 01\n",
    "\n",
    "# Load sequence of lantibiotic synthesis protein, and inspect the information\n",
    "seq = SeqIO.read(\"data/kitasatospora/lantibiotic.fasta\", \"fasta\")\n",
    "\n",
    "# Run the BLASTX search on the nr database, restricted to Kitasatospora spp.\n",
    "result_handle = NCBIWWW.qblast(\"blastx\", \"nr\", seq.format(\"fasta\"),\n",
    "                               entrez_query=\"kitasatospora[ORGN]\",\n",
    "                               format_type=\"Text\")\n",
    "\n",
    "# Save output to file\n",
    "outfilename = os.path.join('output', 'kitasatospora', 'lantibiotic_blastx_nr_kitasatospora.txt')\n",
    "with open(outfilename, 'w') as outfh:\n",
    "    outfh.write(result_handle.read())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true,
    "deletable": true,
    "editable": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (IBioIC)",
   "language": "python",
   "name": "python3_ibioic_course"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
