{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "deletable": true,
    "editable": true
   },
   "source": [
    "<img src=\"../images/JHI_STRAP_Web.png\" style=\"width: 150px; float: right;\">\n",
    "# 10 - `Ensembl`"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "deletable": true,
    "editable": true
   },
   "source": [
    "## Table of Contents\n",
    "\n",
    "1. [Introduction](#introduction)\n",
    "2. [The `EnsemblBacteria` portal](#ensembl)\n",
    "  1. [Using `Ensembl` Genomes](#genomes)\n",
    "  2. [The Genome Browser](#browser)\n",
    "  3. [Gene view](#genes)\n",
    "  4. [Transcript view](#transcripts)\n",
    "3. [Comparative Genomics in `Ensembl`](#comparative)\n",
    "4. [Exercise 01](#ex01)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "deletable": true,
    "editable": true
   },
   "source": [
    "<a id=\"introduction\"></a>\n",
    "## Introduction\n",
    "\n",
    "<p></p>\n",
    "<div class=\"alert-success\">\n",
    "<b>`Ensembl` is a collection of browsers covering the major domains of life, that supports research in comparative genomics, evolution, sequence variation and transcriptional regulation. `Ensembl` contains genome annotations, computes multiple alignments, predicts regulatory function and collects disease data. `Ensembl` tools include BLAST, BLAT, BioMart and the Variant Effect Predictor (VEP) for all supported species. The service is provided by the European Bioinformatics Institute (EBI)</b>\n",
    "</div>\n",
    "\n",
    "`Ensembl` has a number of entrypoints or *portal*s, divided broadly by type of organism:\n",
    "\n",
    "* [`Ensembl` (vertebrates)](http://www.ensembl.org/index.html)\n",
    "* [`EnsemblBacteria`](http://bacteria.ensembl.org/index.html)\n",
    "* [`EnsemblFungi`](http://fungi.ensembl.org/index.html)\n",
    "* [`EnsemblPlants`](http://plants.ensembl.org/index.html)\n",
    "* [`EnsemblProtists`](http://protists.ensembl.org/index.html)\n",
    "* [`EnsemblMetazoa`](http://metazoa.ensembl.org/index.html)\n",
    "\n",
    "Each of these resources is organised in a very similar way, and provides consistent browser, programmatic and pipeline-based routes to access their curated, integrated data. Not all features are available in each resource, however.\n",
    "\n",
    "\n",
    "### Resources\n",
    "* [EMBL-EBI](http://www.ebi.ac.uk/)\n",
    "* [`Ensembl` online training](https://www.ebi.ac.uk/training/online/course/ensembl-browser-webinar-series-2016/introduction-ensembl)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "deletable": true,
    "editable": true
   },
   "source": [
    "<a id=\"ensembl\"></a>\n",
    "## The `EnsemblBacteria` portal\n",
    "\n",
    "<p></p>\n",
    "<div class=\"alert-success\">\n",
    "<b>You'll be using the `EnsemblBacteria` portal in this lesson. `EnsemblBacteria` provides similar access to the other `Ensembl*` portals, and what you learn here will be transferable to the other sites without much modification.</b>\n",
    "</div>\n",
    "\n",
    "`EnsemblBacteria` provides access to over 40,000 bacterial genomes through a common genome browser interface. This data can also be accessed programmatically through a `REST` interface (see later lessons), and downloaded directly. Additionally, over 100 bacterial genomes are covered in the pan-taxonomic `Compara` tool. Bacterial genes from all the genomes are classified into families with `HAMAP` and `PANTHER` tools.\n",
    "\n",
    "Given all this power behind the scenes, the landing page is deceptively plain:\n",
    "\n",
    "![EnsemblBacteria landing page](images/10-01_landing.png)\n",
    "\n",
    "<p></p>\n",
    "<div class=\"alert-warning\">\n",
    "One thing to note is in the lower left corner: the *release number*. In the image above, it shows we are using release 34. `Ensembl`'s release policy is to release complete, updated versions of the entire database, rather than to provide incrementally-updated versions of each individual database entry (which is what `UniProt` does). It is important to cite the `Ensembl` version number used, in any publication that derives from or uses its data.\n",
    "</div>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "deletable": true,
    "editable": true
   },
   "source": [
    "<a id=\"genomes\"></a>\n",
    "### Using `Ensembl` Genomes\n",
    "\n",
    "<p></p>\n",
    "<div class=\"alert-success\">\n",
    "<b>The way we interact with `EnsemblBacteria` is similar to that for all the other `Ensembl` portals. In this section, you'll use the portal to find and browse a *Pectobacterium* genome.</b>\n",
    "</div>\n",
    "\n",
    "As `EnsemblBacteria` provides over 40,000 genomes, it can be hard to navigate the [full list](http://bacteria.ensembl.org/species.html), even though you can link to it from the landing page. It's much simpler to start typing the name of the species you want in the **Search for a genome** box, as for the term *Pectobacterium* below:\n",
    "\n",
    "![EnsemblBacteria genome search page](images/10-02_genomesearch.png)\n",
    "\n",
    "* **Start typing *Pectobacterium* in the search field**\n",
    "\n",
    "This produces a drop-down list, which we can scroll through until we find our organism of interest. In this case, we're going for *Pectobacterium atrosepticum* SCRI 1043\n",
    "\n",
    "* **Click on the entry for *Pectobacterium atrosepticum* SCRI 1043**\n",
    "\n",
    "This brings up the [genome's homepage](http://bacteria.ensembl.org/Pectobacterium_atrosepticum_scri1043/Info/Index), which offers several useful links for information and statistics, comparative genomics, downloading data and, at the top, another search bar so you can search for features of interest:\n",
    "\n",
    "#### Downloading data\n",
    "\n",
    "![Pba SCRI1043 homepage](images/10-03_pba_homepage.png)\n",
    "\n",
    "<p></p>\n",
    "<div class=\"alert-warning\">\n",
    "<b>To download files for this genome, click on the `Download DNA sequence (FASTA)` link. This will take you to an FTP page, and clicking on any of those links will download the corresponding file.</b>\n",
    "</div>\n",
    "\n",
    "\n",
    "![Pba SCRI1043 homepage](images/10-03a_pba_download.png)\n",
    "\n",
    "On the right hand side of the page, there are links to download gene and protein sequences in `FASTA` or `GFF3` (usable with `Artemis` or `Tablet`) format. These also take you to an FTP site from where you can download the corresponding data."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "deletable": true,
    "editable": true
   },
   "source": [
    "<a id=\"browser\"></a>\n",
    "### The Genome Browser\n",
    "\n",
    "<p></p>\n",
    "<div class=\"alert-success\">\n",
    "<b>On each genome home page, there's a link to a region of interest, in this case: [`Chromosome:2198014-2201221`](http://bacteria.ensembl.org/Pectobacterium_atrosepticum_scri1043/psychic?q=Chromosome:2198014-2201221;site=ensemblthis), just below the search bar. We'll use this to get a first look at the genome browser.</b>\n",
    "</div>\n",
    "\n",
    "* **Click on the region of interest link to `Chromosome:2198014-2201221`**\n",
    "\n",
    "![Pba region of interest 1](images/10-04_pba_roi.png)\n",
    "\n",
    "<p></p>\n",
    "<div class=\"alert-warning\">\n",
    "<b>NOTE:</b> the `Ensembl` browser interface puts up tabs at the top of the page to indicate which view you are currently in - this is a good way to take shortcuts between search outputs.\n",
    "</div>\n",
    "\n",
    "At the top of this page, there's a circular overview of the entire chromosome, with features and GC content/skew indicated, and the region of interest marked with a red wedge. Below this there's a linear view of the region in detail.\n",
    "\n",
    "On the circular view, there are yellow and blue *handles*. You can click and drag these to modify the region of interest, and if you click on this new region, the linear view beneath the circle will update (if the region is not too large to view).\n",
    "\n",
    "* **Use the handles to zoom in on a region approximately 1,644434-1,688,332**\n",
    "\n",
    "![Pba region of interest 2](images/10-05_pba_roi_moved.png)\n",
    "\n",
    "The lower region shows a view of the features annotated on the genome, coloured by type. In the example above, all the features are protein-coding genes. If you click on one of these, a small popup window appears, giving a little more information.\n",
    "\n",
    "* **Click on one of the features in the lower view**\n",
    "\n",
    "![Pop-up window](images/10-06_feature_popup.png)\n",
    "\n",
    "<p></p>\n",
    "<div class=\"alert-warning\">\n",
    "<b>You can change which features show in the browser view by clicking on the `Configure this page` option in the left-hand menu. For now, we will persist with the default view.</b>\n",
    "</div>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "deletable": true,
    "editable": true
   },
   "source": [
    "<a id=\"genes\"></a>\n",
    "### Gene view\n",
    "\n",
    "In the popup you saw above, you will have noticed several distinct identifiers - for the transcript, the gene, and the protein product. These are live links, and will take you to a new page centring the browser on that feature, with an overview window that shows the *gene-based display*. For instance, clicking on the [`ECA1453`](http://bacteria.ensembl.org/Pectobacterium_atrosepticum_scri1043/Gene/Summary?db=core;g=ECA1453;r=Chromosome:1649698-1650771;t=CAG74363) gene link above brings up the gene view:\n",
    "\n",
    "* **Click on the ECA1453 link**\n",
    "\n",
    "![ECA1453 gene view](images/10-07_gene_view.png)\n",
    "\n",
    "You will see that the content of the left-hand menu has changed, giving links out to sequence annotations such as GO terms, and external references.\n",
    "\n",
    "<p></p>\n",
    "<div class=\"alert-warning\">\n",
    "<b>NOTE:</b> in general the annotations in Ensembl may not be identical to, or contain the same detailed information as, those found in other resources. This genome, <i>Pba</i> SCRI 1043 was carefully manually annotated by a team of six people over a period of six months; that complete annotation can be found in the GenBank record for this genome.\n",
    "</div>\n",
    "\n",
    "To obtain an account of the links to other databases for this feature, you can use the `External references` link in the left-hand menu. This will bring up an account of other entries for this sequence.\n",
    "\n",
    "* **Click on the External references menu item**\n",
    "\n",
    "![ECA1453 external references](images/10-08_references.png)\n",
    "\n",
    "<p></p>\n",
    "<div class=\"alert-warning\">\n",
    "<b>QUESTIONS</b>\n",
    "<ol>\n",
    "<li> Can you find the corresponding entry for this feature in NCBI/GenBank, using your browser?\n",
    "</ol>\n",
    "</div>\n",
    "\n",
    "You can search directly for a gene of interest in the the gene search box, and we will do this for the virulence-related protein VgrG, below.\n",
    "\n",
    "* **Click on the tab for the whole genome home page**\n",
    "* **Enter vgrG into the search field and click `Go` or hit `Return`**\n",
    "\n",
    "![ECA1453 vgrG search results](images/10-09_vgrg_search.png)\n",
    "\n",
    "<div class=\"alert-warning\">\n",
    "<b>QUESTIONS</b>\n",
    "<ol>\n",
    "<li> How many genes does the search return?\n",
    "<li> Are all of these results likely to be active coding genes?\n",
    "<li> How many genes in the GenBank record for this genome are annotated as `VgrG` <b>HINT: the genome record can be found <a href=\"https://www.ncbi.nlm.nih.gov/nuccore/49609491\">here</a></b>.\n",
    "</ol>\n",
    "</div>\n",
    "\n",
    "* **Go to the gene page for the first hit, and click on `Sequence` in the left-hand menu**\n",
    "\n",
    "This will present a marked-up view of the genomic sequence for, and flanking, that `vgrG` search result. There are buttons that will allow you to download the sequence in a range of formats, or submit the sequence as a `BLAST` search, and the exons are highlighted, with the current gene of interest shown in red.\n",
    "\n",
    "![Ensembl sequence markup](images/10-10_markup.png)\n",
    "\n",
    "The FASTA header shows the location of the region covered by the sequence.\n",
    "\n",
    "<p></p>\n",
    "<div class=\"alert-warning\">\n",
    "<b>QUESTIONS</b>\n",
    "<ol>\n",
    "<li> What does the FASTA sequence represent?\n",
    "<li> Is this the same sequence you would expect from `UniProt`, `KEGG` or `NCBI`? <b>HINT: <a href=\"http://www.uniprot.org/uniprot/Q6D772\">`UniProt`</a>, <a href=\"http://www.genome.jp/dbget-bin/www_bget?eca:ECA1453\">`KEGG`</a> and <a href=\"https://www.ncbi.nlm.nih.gov/gene/?term=ECA1453\">`NCBI`</a> records for ECA1453.</b>\n",
    "</ol>\n",
    "</div>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "deletable": true,
    "editable": true
   },
   "source": [
    "<a id=\"transcripts\"></a>\n",
    "### Transcript view\n",
    "\n",
    "<p></p>\n",
    "<div class=\"alert-success\">\n",
    "<b>When you clicked through to reach the gene page for your search result, `Ensembl` kindly also opened up a page for the corresponding transcript (on the right hand side of the tab list), which you should be able to see in the top tab.</b>\n",
    "</div>\n",
    "\n",
    "* **Click on the transcript tab**\n",
    "\n",
    "This will bring you to the transcript summary view:\n",
    "\n",
    "![Ensembl transcript page](images/10-11_transcript.png)\n",
    "\n",
    "* **Click on the `cDNA` menu item on the left-hand side**\n",
    "\n",
    "This will show the transcript coding sequence.\n",
    "\n",
    "![Ensembl cDNA sequence](images/10-12_cdna.png)\n",
    "\n",
    "* **Click on the `General identifiers` menu item**\n",
    "\n",
    "This shows the external database links, to find more information about this sequence. In particular, the link out to `UniProtKB` indicates the percentage sequence identity to one of `UniProt`'s protein sequence entries. Here it is, as would be expected, a 100% identity match."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "deletable": true,
    "editable": true
   },
   "source": [
    "<a id=\"comparative\"></a>\n",
    "## Comparative Genomics in `Ensembl`\n",
    "\n",
    "<p></p>\n",
    "<div class=\"alert-success\">\n",
    "<b>For some genomes, `EnsemblBacteria` provides gene trees and lists of homologues in a range of other species. To explore this, you will look at the comparative genomics of the *E. coli* K-12 gene `pfkA`.</b>\n",
    "</div>\n",
    "\n",
    "* **Go to the page for *E. coli* K-12 MG1655**\n",
    "\n",
    "* [*E. coli* K-12 MG1655 at `Ensembl`](http://bacteria.ensembl.org/Escherichia_coli_str_k_12_substr_mg1655/Info/Index)\n",
    "\n",
    "![E. coli K-12 MG1655 homepage](images/10-13_k12.png)\n",
    "\n",
    "* **Search for the gene *pfkA*.**\n",
    "* **Click on the link for `b3916` to get the gene view**\n",
    "\n",
    "![pfkA gene page](images/10-14_b3916.png)\n",
    "\n",
    "Note that the menu on the left now has the `Pan-taxonomic Compara`, `Gene Tree` and `Orthologues` links available. These are the entrypoints to `Ensembl`'s comparative genomics tools. \n",
    "\n",
    "* **Click on the `Gene Tree` menu option**\n",
    "\n",
    "![pfkA gene tree](images/10-15_genetree.png)\n",
    "\n",
    "The browser now shows a tree locating this *E. coli* *pfkA* in the context of orthologues from other organisms. The *E. coli* gene of interest is highlighted in red, and a set of figurative protein alignments is shown to the right of the tree - these alignments indicate regions of the sequence and their extent of sequence identity. On the tree itself, several branches are 'collapsed' into shaded triangles, and more information about these collapsed sequence sets can be obtained by clicking on the grey triangle, to bring up a pop-up window.\n",
    "\n",
    "![pfkA gene tree popup](images/10-16_popup.png)\n",
    "\n",
    "The pop-up has a link (`expand this sub-tree`) that allows you to see the members of the collapsed branches in full.\n",
    "\n",
    "* **Click on the `Orthologues` link in the left-hand menu**\n",
    "\n",
    "This will present the set of orthologues to *pfkA* in the `Pan-taxonomic Compara` as a table:\n",
    "\n",
    "![pfkA orthologue table](images/10-17_orthologues.png)\n",
    "\n",
    "<p></p>\n",
    "<div class=\"alert-warning\">\n",
    "<b>NOTE:</b> the `Ensembl` definition of <i>orthologue</i> is different to how it might be used in the literature. In `Ensembl` an \"orthologue\" may refer to a `1-to-1` orthologue (as in the original, strict definition), but it may also be `1-to-many` or `many-to-many`, which are in phylogenetic terms not orthologues by the standard definition.\n",
    "</div>\n",
    "\n",
    "* How `Ensembl` calculates and describes orthologues: [Compara homology methods](http://www.ensembl.org/info/genome/compara/homology_method.html)\n",
    "\n",
    "The table is interactive and will allow you to live-filter on organism type by selecting `Show details` for any of the offered groups. Also, each individual row in the table presents links to view the corresponding gene page, and the sequence alignment.\n",
    "\n",
    "* **Click on `Show details` to restrict the table to bacterial orthologues**\n",
    "* **Enter `Myco` into the table filter to restrict the table to two rows**\n",
    "* **Click on the link to view sequence alignments in the first row**\n",
    "\n",
    "Clicking on the sequence alignment link will bring up a pop-up menu asking if you want the protein or cDNA alignment.\n",
    "\n",
    "* **Click on `View Protein Alignment`**\n",
    "\n",
    "![pfkA orthologue alignment](images/10-18_alignment.png)\n",
    "\n",
    "This gives the pairwise alignment used in constructing the orthologue set for your query sequence, in `ClustalW` format. There is information given on sequence percentage identity and coverage for this alignment."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "deletable": true,
    "editable": true
   },
   "source": [
    "<img src=\"../images/exercise.png\" style=\"width: 100px; float: left;\">\n",
    "<a id=\"ex01\"></a>\n",
    "### Exercise 01 (10min)\n",
    "\n",
    "Using the `EnsemblBacteria` tools, starting from `EnsemblBacteria`, can you:\n",
    "\n",
    "<br></br>\n",
    "<div class=\"alert-danger\">\n",
    "<ul>\n",
    "<li> Go to the home page for the *Kitasatospora setae* KM-6054 genome\n",
    "<li> Find how many coding genes are annotated in this genome? \n",
    "<li> Download the FASTA file describing all predicted protein gene products?\n",
    "<li> Find how many putative penicillin-binding proteins are annotated in the genome?\n",
    "<li> Examine the feature `KSE_59840` in the genome browser, and find the GO term for its molecular function? What is the evidence for this functional annotation?\n",
    "<li> What is the `UniProt` accession for this protein?\n",
    "</ul>\n",
    "</div>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true,
    "deletable": true,
    "editable": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (IBioIC)",
   "language": "python",
   "name": "python3_ibioic_course"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
